--- misc/build/libwpd-0.7.99/src/lib/WPDocument.cpp	1 Aug 2004 20:10:52 -0000	1.23
+++ misc/build/libwpd-0.7.99/src/lib/WPDocument.cpp	7 Dec 2004 12:45:50 -0000
@@ -109,7 +109,8 @@
 					confidence = WPD_CONFIDENCE_EXCELLENT;
 					break;
 				case 0x04: // WP Mac 3.5e
-					confidence = WPD_CONFIDENCE_NONE; // We currently don't have a parser for this, since we don't have the file format documentation
+					confidence = WPD_CONFIDENCE_EXCELLENT;
+//					confidence = WPD_CONFIDENCE_NONE; // We currently don't have a parser for this, since we don't have the file format documentation
 					break;
 				default:
 					// unhandled file format
@@ -204,7 +205,10 @@
 					parser->parse(listenerImpl);
 					break;
 				case 0x04: // WP Mac 3.5e
-					WPD_DEBUG_MSG(("WordPerfect: WP Mac 3.5e documents are currently unsupported since we have no file format documentation.\n"));
+//					WPD_DEBUG_MSG(("WordPerfect: WP Mac 3.5e documents are currently unsupported since we have no file format documentation.\n"));
+					WPD_DEBUG_MSG(("WordPerfect: Using the WP3 parser.\n"));
+					parser = new WP3Parser(document, header);
+					parser->parse(listenerImpl);
 					break;				
 				default:
 					// unhandled file format
