--- misc/build/libwpd-0.8.3/src/lib/WP6HLContentListener.cpp	25 Aug 2005 14:34:42 -0000	1.81
+++ misc/build/libwpd-0.8.3/src/lib/WP6HLContentListener.cpp	31 Aug 2005 16:17:50 -0000
@@ -1037,6 +1037,7 @@
 		// save a reference to the text PID, we want to parse
 		// the packet after we're through with the footnote ref.
 		m_parseState->m_noteTextPID = textPID;
+		m_ps->m_isNote = true;
 	}
 }
 
@@ -1064,6 +1065,7 @@
 			m_listenerImpl->closeFootnote();
 		else
 			m_listenerImpl->closeEndnote();
+		m_ps->m_isNote = false;
 	}
 }
 
@@ -1253,7 +1255,7 @@
 //
 void WP6HLContentListener::_flushText()
 {
-	if (m_ps->m_isListElementOpened)
+	if (m_ps->m_isListElementOpened || m_ps->m_isNote)
 	{
 		m_parseState->m_textBeforeNumber.clear();
 		m_parseState->m_textBeforeDisplayReference.clear();
--- misc/build/libwpd-0.8.3/src/lib/WPXHLListener.cpp	25 Aug 2005 14:34:42 -0000	1.62
+++ misc/build/libwpd-0.8.3/src/lib/WPXHLListener.cpp	31 Aug 2005 16:17:50 -0000
@@ -883,6 +883,7 @@
 	m_ps->m_pageMarginLeft = oldPS->m_pageMarginLeft;
 	m_ps->m_pageMarginRight = oldPS->m_pageMarginRight;
 	m_ps->m_subDocumentTextPIDs = oldPS->m_subDocumentTextPIDs;
+	m_ps->m_isNote = oldPS->m_isNote;
 	// END: copy page properties into the new parsing state
 	m_ps->m_inSubDocument = true;
 	// Check whether the document is calling its own TextPID
