--- MainController.m	2009-11-16 02:59:23.000000000 -0800
+++ MainController.m	2010-09-01 15:37:36.000000000 -0700
@@ -19,11 +19,9 @@
 }
 
 - (void)applicationWillBecomeActive:(NSNotification *)aNotification {
-	NSLog(@"Application will become active - Using remote controls");
 	[remoteControl startListening: self];
 }
 - (void)applicationWillResignActive:(NSNotification *)aNotification {
-	NSLog(@"Application will resign active - Releasing remote controls");
 	[remoteControl stopListening: self];
 }
 
@@ -112,10 +110,6 @@
 	// delegate to view
 	[feedbackView remoteButton:buttonIdentifier pressedDown:pressedDown clickCount: clickCount];
 	
-	// print out events
-	NSLog(@"%@", feedbackString);
-	if (pressedDown == NO) printf("\n");
-	
 }
 
-@end
\ No newline at end of file
+@end
--- RemoteControlFramework.xcodeproj/project.pbxproj	2010-01-11 07:16:49.000000000 -0800
+++ RemoteControlFramework.xcodeproj/project.pbxproj	2012-07-13 11:19:53.000000000 -0700
@@ -243,7 +243,6 @@
 				);
 				COPY_PHASE_STRIP = NO;
 				GCC_ENABLE_OBJC_EXCEPTIONS = YES;
-				SDKROOT = /Developer/SDKs/MacOSX10.4u.sdk;
 			};
 			name = Debug;
 		};
@@ -256,7 +255,6 @@
 				);
 				COPY_PHASE_STRIP = YES;
 				GCC_ENABLE_OBJC_EXCEPTIONS = YES;
-				SDKROOT = /Developer/SDKs/MacOSX10.4u.sdk;
 			};
 			name = Release;
 		};
