--- source/quicktime/framegrabber.hxx.bak	2008-04-10 11:21:13.000000000 -0700
+++ source/quicktime/framegrabber.hxx	2014-04-28 09:41:36.000000000 -0700
@@ -67,7 +67,7 @@
     ::rtl::OUString                                                                     maURL;
 	QTMovie*                                                                            mpMovie; 
     sal_Bool																			mbInitialized;
-    long																				mnVersion;
+    SInt32																				mnVersion;
 };
 
 } // namespace quicktime
--- source/quicktime/player.hxx.bak	2008-04-10 11:23:51.000000000 -0700
+++ source/quicktime/player.hxx	2014-04-28 09:42:19.000000000 -0700
@@ -105,7 +105,7 @@
     int                     mnWidth;
     int                     mnHeight;
 
-    long                    mnVersion;
+    SInt32                  mnVersion;
     oslCondition            maSizeCondition;
 };
 
