? unxmacxp.pro.oo
Index: source/app/appwin.hxx
===================================================================
RCS file: /cvs/script/basic/source/app/appwin.hxx,v
retrieving revision 1.2
diff -u -r1.2 appwin.hxx
--- source/app/appwin.hxx	28 Mar 2002 14:43:03 -0000	1.2
+++ source/app/appwin.hxx	27 Sep 2005 05:30:44 -0000
@@ -101,7 +101,7 @@
 
 class AppWin : public DockingWindow, public SfxListener		// Dokumentfenster
 {
-	friend MsgEdit;
+	friend class MsgEdit;
 protected:
 	static short nNumber;			// fortlaufende Nummer
 	static short nCount;			// Anzahl Editfenster
