--- basic/source/comp/codegen.cxx	2012-07-12 17:33:11.000000000 -0700
+++ basic/source/comp/codegen.cxx	2012-09-25 20:23:32.000000000 -0700
@@ -533,6 +533,9 @@
 	m_nCnvtdSize = static_cast<S>( aTrnsfrmer.buffer().GetSize() );
 }
 
+template class PCodeBuffConvertor< UINT16, UINT32 >;
+template class PCodeBuffConvertor< UINT32, UINT16 >;
+
 void NeverRunsEver()
 {
 	// force instatiation of templates... I dunno why, but I have to do
