--- com/sun/star/beans/makefile.mk	2008-04-10 09:14:53.000000000 -0700
+++ com/sun/star/beans/makefile.mk	2011-12-20 10:39:20.000000000 -0800
@@ -37,12 +37,12 @@
 # --- Settings -----------------------------------------------------
 .INCLUDE: settings.mk
 
-.IF "$(OS)"=="MACOSX"
+.IF "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 dummy:
 	@echo "Nothing to build for OS $(OS)"
 
-.ELSE		# "$(OS)"=="MACOSX"
+.ELSE		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 JARFILES 		= sandbox.jar ridl.jar unoil.jar jurt.jar juh.jar
 
@@ -60,7 +60,7 @@
 
 JAVACLASSFILES= $(foreach,i,$(JAVAFILES) $(CLASSDIR)$/$(PACKAGE)$/$(i:b).class)
 
-.ENDIF
+.ENDIF		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 # --- Targets ------------------------------------------------------
 
--- com/sun/star/comp/beans/makefile.mk	2008-04-10 09:21:28.000000000 -0700
+++ com/sun/star/comp/beans/makefile.mk	2011-12-20 10:39:24.000000000 -0800
@@ -37,12 +37,12 @@
 # --- Settings -----------------------------------------------------
 .INCLUDE: settings.mk
 
-.IF "$(OS)"=="MACOSX"
+.IF "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 dummy:
 	@echo "Nothing to build for OS $(OS)"
 
-.ELSE		# "$(OS)"=="MACOSX"
+.ELSE		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 JARFILES 		= sandbox.jar ridl.jar unoil.jar jurt.jar juh.jar
 
@@ -71,7 +71,7 @@
 
 JAVACLASSFILES= $(foreach,i,$(JAVAFILES) $(CLASSDIR)$/$(PACKAGE)$/$(i:b).class)
 
-.ENDIF
+.ENDIF		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
 
 # --- Targets ------------------------------------------------------
 
--- native/win32/makefile.mk	2007-06-26 10:45:58.000000000 -0700
+++ native/win32/makefile.mk	2011-12-20 10:39:52.000000000 -0800
@@ -11,6 +11,13 @@
 
 .INCLUDE :  settings.mk
 
+.IF "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
+
+dummy:
+	@echo "Nothing to build for OS $(OS)"
+
+.ELSE		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
+
 # --- Files --------------------------------------------------------
 
 SLOFILES = \
@@ -28,6 +35,8 @@
 DEF1EXPORTFILE=	$(TARGET).dxp
 DEF1DES=officebean
 
+.ENDIF		# "$(OS)"=="MACOSX" || "$(OS)"=="WNT"
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
