Index: makefile.mk
===================================================================
RCS file: /cvs/external/berkeleydb/makefile.mk,v
retrieving revision 1.30
diff -u -r1.30 makefile.mk
--- makefile.mk	13 Jan 2006 16:23:55 -0000	1.30
+++ makefile.mk	11 Apr 2006 17:18:14 -0000
@@ -138,6 +138,11 @@
 
 # --- Targets ------------------------------------------------------
 
+.IF "$(OS)"=="MACOSX"
+ALL: ALLTAR
+	install_name_tool -id @executable_path/$(DLLPRE)db-4.2$(DLLPOST) $(LB)$/$(DLLPRE)db-4.2$(DLLPOST)
+.ENDIF
+
 .INCLUDE : set_ext.mk
 .INCLUDE :	target.mk
 .INCLUDE :	tg_ext.mk
