Index: makefile.mk
===================================================================
RCS file: /cvs/external/berkeleydb/makefile.mk,v
retrieving revision 1.13.12.2
diff -u -r1.13.12.2 makefile.mk
--- makefile.mk	28 Oct 2004 11:54:48 -0000	1.13.12.2
+++ makefile.mk	28 Dec 2004 18:11:31 -0000
@@ -93,6 +93,9 @@
 .ENDIF
 
 OUT2LIB=$(BUILD_DIR)$/.libs$/libdb*$(DLLPOST)
+.IF "$(OS)"=="MACOSX"
+OUT2LIB+=$(BUILD_DIR)$/.libs$/libdb*.jnilib
+.ENDIF
 .IF "$(SOLAR_JAVA)"!=""
 OUT2BIN=$(BUILD_DIR)$/db.jar
 OUT2CLASS=$(BUILD_DIR)$/db.jar
Index: prj/d.lst
===================================================================
RCS file: /cvs/external/berkeleydb/prj/d.lst,v
retrieving revision 1.4.6.1
diff -u -r1.4.6.1 d.lst
--- prj/d.lst	28 Oct 2004 11:55:58 -0000	1.4.6.1
+++ prj/d.lst	28 Dec 2004 18:11:31 -0000
@@ -13,6 +13,7 @@
 
 ..\%__SRC%\lib\libdb-4.2.dylib %_DEST%\lib%_EXT%\libdb-4.2.dylib
 ..\%__SRC%\lib\libdb_java-4.2.dylib %_DEST%\lib%_EXT%\libdb_java-4.2.dylib
+..\%__SRC%\lib\libdb_java-4.2.jnilib %_DEST%\lib%_EXT%\libdb_java-4.2.jnilib
 ..\%__SRC%\lib\libdb_cxx-4.2.dylib %_DEST%\lib%_EXT%\libdb_cxx-4.2.dylib
 
 ..\%__SRC%\bin\libdb42.dll %_DEST%\bin%_EXT%\libdb42.dll
