Index: bf_forms/util/makefile.mk
===================================================================
RCS file: /cvs/framework/binfilter/bf_forms/util/makefile.mk,v
retrieving revision 1.7
diff -u -r1.7 makefile.mk
--- bf_forms/util/makefile.mk	19 Jul 2006 13:14:05 -0000	1.7
+++ bf_forms/util/makefile.mk	21 Nov 2006 17:45:39 -0000
@@ -103,19 +103,10 @@
 RESLIB1SRSFILES=$(RES1FILELIST)
 
 .IF "$(GUI)"=="UNX"
-
-# [ed] 6/19/02 Only add in libraries for X11 OS X builds
-
-.IF "$(OS)"=="MACOSX"
-.IF "$(GUIBASE)"=="unx"
-SHL1STDLIBS +=\
-        -lX11 -lXt -lXmu
-.ENDIF
-.ELSE
+.IF "$(OS)"!="MACOSX"
 SHL1STDLIBS +=\
 	-lX11
-.ENDIF # OS == MACOSX
-
+.ENDIF
 .ENDIF
 
 # --- Targets ----------------------------------
