--- binfilter/bf_sc/source/ui/unoobj/sc_notesuno.cxx	2008-04-10 03:00:03.000000000 -0700
+++ binfilter/bf_sc/source/ui/unoobj/sc_notesuno.cxx	2012-09-25 21:12:09.000000000 -0700
@@ -131,9 +131,9 @@
 	//!	existierendes Objekt finden und zurueckgeben ???
 
 	if (pDocShell)
-		return (cppu::OWeakObject*)new ScCellObj( pDocShell, aCellPos );
+		return uno::Reference<uno::XInterface>((cppu::OWeakObject*)new ScCellObj( pDocShell, aCellPos ));
 
-	return NULL;
+	return uno::Reference<uno::XInterface>(NULL);
 }
 
 void SAL_CALL ScAnnotationObj::setParent( const uno::Reference<uno::XInterface>& Parent )
