--- source/cpp_uno/gcc3_macosx_intel/makefile.mk	2013-08-13 04:42:39.000000000 -0700
+++ source/cpp_uno/gcc3_macosx_intel/makefile.mk	2014-04-26 16:30:19.000000000 -0700
@@ -24,7 +24,11 @@
 PRJ=..$/..$/..
 
 PRJNAME=bridges
+.IF "$(UPD)" == "310"
+TARGET=gcc3_uno
+.ELSE	# "$(UPD)" == "310"
 TARGET=cxx_uno
+.ENDIF	# "$(UPD)" == "310"
 LIBTARGET=no
 ENABLE_EXCEPTIONS=TRUE
 
@@ -34,7 +38,7 @@
 
 # --- Files --------------------------------------------------------
 
-.IF "$(OS)$(CPU)" == "MACOSXX"
+.IF "$(OS)$(CPU)" == "MACOSXX" || ("$(OS)$(CPU)" == "MACOSXI" || "$(UPD)" == "310")
 
 .IF "$(cppu_no_leak)" == ""
 CFLAGS += -DLEAK_STATIC_DATA
