--- configure.in	2014-04-22 16:31:37.000000000 -0700
+++ configure.in	2014-04-22 16:59:18.000000000 -0700
@@ -2559,7 +2559,7 @@
       LDFLAGS=$sharedlink_ldflags_save
    fi
 
-   if test "$HAVE_GCC_VISIBILITY_FEATURE" = "TRUE"; then
+   if test "$_os" != "Darwin" -a test "$HAVE_GCC_VISIBILITY_FEATURE" = "TRUE"; then
       AC_MSG_CHECKING([if gcc has a visibility bug with class-level attributes (GCC bug 26905)])
       cat >visibility.cxx <<_ACEOF
 #pragma GCC visibility push(hidden)
@@ -4305,9 +4305,11 @@
 
 if test "$BUILD_MOZAB" = "TRUE"; then
    if test -z "$MOZILLA_VERSION"; then
-      MOZILLA_VERSION=1.7.5
+      MOZILLA_VERSION=3.16
    fi
-   if test "$MOZILLA_VERSION" = "1.7b" ; then
+   if test "$MOZILLA_VERSION" = "3.16" ; then
+      MOZILLA_SOURCE_VERSION="nss-3.16-with-nspr-4.10.4"
+   elif test "$MOZILLA_VERSION" = "1.7b" ; then
       MOZILLA_SOURCE_VERSION="mozilla-source-1.7b-source"
    else
       MOZILLA_SOURCE_VERSION="mozilla-source-${MOZILLA_VERSION}"
@@ -4346,11 +4348,6 @@
       if test "$MOZILLA_TOOLKIT" = "gtk2"; then
          AC_MSG_NOTICE([checking whether mozilla can be built...])
          PKG_CHECK_MODULES(MOZGTK2, gtk+-2.0 >= 2.4 libIDL-2.0 >= 0.8, AC_MSG_NOTICE([OK - can build mozilla]), AC_MSG_ERROR([Prerequisites to build mozilla not met. Either use the precompiled mozilla binaries or install the missing packages]))
-      else
-         PKG_CHECK_MODULES(MOZLIBREQ, libIDL >= 0.6.3 libIDL <= 0.6.8, MOZIDL="TRUE", MOZIDL="")
-         if test -z "$MOZIDL"; then
-            AC_MSG_ERROR([libIDL 0.6.3 - 0.6.8 is needed when not using GTK2 to build mozilla.])
-         fi
       fi
    else
       # Generic Unix/Linux section
