--- configure.in	2007-05-06 21:54:21.000000000 -0700
+++ configure.in	2007-05-06 21:54:34.000000000 -0700
@@ -3793,11 +3793,6 @@
    enable_build_mozilla=1
    AC_MSG_RESULT([$MOZILLA_TOOLKIT])
 fi
-if test "$_os" = "Darwin" && test "$MOZILLA_TOOLKIT" != "gtk2"; then
-   #only gtk2 toolkit supported - xlib or cocoa nees glib1 and libIDL1 - the latter is not
-   #available using fink, mac (carbon) doesn't work because xcode installs conflicting headers
-   AC_MSG_ERROR([Only gtk2 toolkit supported on Mac, sorry.])
-fi
 
 AC_SUBST(MOZILLA_TOOLKIT)
 
@@ -3851,10 +3851,6 @@
       else
          AC_MSG_RESULT([ok])
       fi
-   elif test "$_os" = "Darwin"; then
-      # only gtk2 supported - see above
-      AC_MSG_NOTICE([checking whether mozilla can be built...])
-      PKG_CHECK_MODULES(MOZGTK2, gtk+-2.0 >= 2.4 libIDL-2.0 >= 0.8, AC_MSG_NOTICE([OK - can build mozilla]), AC_MSG_ERROR([Prerequisites to build mozilla not met. Either use the precompiled mozilla binaries or install the missing packages]))
    else
       # Generic Unix/Linux section
       if test "$MOZILLA_TOOLKIT" = "gtk2"; then
@@ -3862,7 +3862,7 @@
          if test -z "$MOZIDL"; then
             AC_MSG_ERROR([libIDL >= 0.8.0 is needed when using GTK2 to build mozilla.])
          fi 
-      else
+      elif test "$MOZILLA_TOOLKIT" != "xlib"; then
          PKG_CHECK_MODULES(MOZLIBREQ, gtk+ >= 1.2.3, MOZGTK="TRUE", MOZGTK="")
          if test -z "$MOZGTK"; then
             AC_MSG_ERROR([gtk 1.2 is needed when not using GTK2 to build mozilla.])
