--- source/drivers/file/FResultSet.cxx	2009-02-06 03:35:13.000000000 -0800
+++ source/drivers/file/FResultSet.cxx	2012-09-24 13:16:52.000000000 -0700
@@ -475,7 +475,7 @@
 	checkDisposed(OResultSet_BASE::rBHelper.bDisposed);
 
 
-	return m_xStatement;
+	return ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface >(m_xStatement);
 }
 // -------------------------------------------------------------------------
 
--- source/drivers/mozab/makefile.mk	2009-02-17 01:08:51.000000000 -0800
+++ source/drivers/mozab/makefile.mk	2014-04-28 07:26:50.000000000 -0700
@@ -34,7 +34,7 @@
 TARGET=mozab
 TARGET2=$(TARGET)drv
 
-.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || ( "$(OS)" == "MACOSX" )
+.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || ( "$(OS)" == "MACOSX" ) || "$(UPD)"=="310"
 all: 
 	@echo "    Not building the mozilla address book driver"
 	@echo "    dependency to Mozilla developer snapshots not feasable at the moment"
--- source/drivers/mozab/mozillasrc/makefile.mk	2008-12-30 06:32:01.000000000 -0800
+++ source/drivers/mozab/mozillasrc/makefile.mk	2014-04-28 07:25:56.000000000 -0700
@@ -52,7 +52,7 @@
 
 .INCLUDE : settings.mk
 
-.IF  ("$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO"
+.IF  ("$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || "$(UPD)"=="310"
 dummy:
 	@echo "		Not building the mozillasrc stuff in OpenOffice.org build"
 	@echo "		dependency to Mozilla developer snapshots not feasable at the moment"
--- source/drivers/mozab/bootstrap/makefile.mk	2009-02-17 01:08:51.000000000 -0800
+++ source/drivers/mozab/bootstrap/makefile.mk	2014-04-28 07:49:37.000000000 -0700
@@ -64,7 +64,7 @@
 	$(SLO)$/MMozillaBootstrap.obj \
 	$(SLO)$/MNSFolders.obj
 
-.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES" ) || ( "$(WITH_MOZILLA)" == "NO" ) || ( "$(OS)" == "MACOSX" )
+.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES" ) || ( "$(WITH_MOZILLA)" == "NO" ) || ( "$(OS)" == "MACOSX" ) || "$(UPD)" == "310"
 CDEFS+=-DMINIMAL_PROFILEDISCOVER
 
 SHL1TARGET=$(TARGET)
