--- source/drivers/file/FResultSet.cxx	2009-02-06 03:35:13.000000000 -0800
+++ source/drivers/file/FResultSet.cxx	2012-09-24 13:16:52.000000000 -0700
@@ -475,7 +475,7 @@
 	checkDisposed(OResultSet_BASE::rBHelper.bDisposed);
 
 
-	return m_xStatement;
+	return ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface >(m_xStatement);
 }
 // -------------------------------------------------------------------------
 
--- source/drivers/mozab/makefile.mk	2009-02-17 01:08:51.000000000 -0800
+++ source/drivers/mozab/makefile.mk	2014-04-28 07:26:50.000000000 -0700
@@ -34,7 +34,7 @@
 TARGET=mozab
 TARGET2=$(TARGET)drv
 
-.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || ( "$(OS)" == "MACOSX" )
+.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || ( "$(OS)" == "MACOSX" ) || "$(UPD)"=="310"
 all: 
 	@echo "    Not building the mozilla address book driver"
 	@echo "    dependency to Mozilla developer snapshots not feasable at the moment"
--- source/drivers/mozab/mozillasrc/makefile.mk	2008-12-30 06:32:01.000000000 -0800
+++ source/drivers/mozab/mozillasrc/makefile.mk	2014-04-28 07:25:56.000000000 -0700
@@ -52,7 +52,7 @@
 
 .INCLUDE : settings.mk
 
-.IF  ("$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO"
+.IF  ("$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES") || "$(WITH_MOZILLA)" == "NO" || "$(UPD)"=="310"
 dummy:
 	@echo "		Not building the mozillasrc stuff in OpenOffice.org build"
 	@echo "		dependency to Mozilla developer snapshots not feasable at the moment"
--- source/drivers/mozab/bootstrap/makefile.mk	2009-02-17 01:08:51.000000000 -0800
+++ source/drivers/mozab/bootstrap/makefile.mk	2014-04-28 07:49:37.000000000 -0700
@@ -64,7 +64,7 @@
 	$(SLO)$/MMozillaBootstrap.obj \
 	$(SLO)$/MNSFolders.obj
 
-.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES" ) || ( "$(WITH_MOZILLA)" == "NO" ) || ( "$(OS)" == "MACOSX" )
+.IF ( "$(SYSTEM_MOZILLA)" == "YES" && "$(WITH_MOZILLA)" == "YES" ) || ( "$(WITH_MOZILLA)" == "NO" ) || ( "$(OS)" == "MACOSX" ) || "$(UPD)" == "310"
 CDEFS+=-DMINIMAL_PROFILEDISCOVER
 
 SHL1TARGET=$(TARGET)
--- source/drivers/odbcbase/OPreparedStatement.cxx	2009-01-08 06:32:54.000000000 -0800
+++ source/drivers/odbcbase/OPreparedStatement.cxx	2014-05-16 12:00:03.000000000 -0700
@@ -446,7 +446,7 @@
 	checkParameterIndex(parameterIndex);
 
 	sal_Int8* lenBuf = getLengthBuf (parameterIndex);
-	*(SDWORD*)lenBuf = SQL_NULL_DATA;
+	*(SQLINTEGER*)lenBuf = SQL_NULL_DATA;
 
 
 	SQLINTEGER	prec = 0;
@@ -476,7 +476,7 @@
 											nDecimalDigits,
 											NULL,
 											prec,
-											(SDWORD*)lenBuf
+											(SQLINTEGER*)lenBuf
 											);
 	OTools::ThrowException(m_pConnection,nReturn,m_aStatementHandle,SQL_HANDLE_STMT,*this);
 }
@@ -879,7 +879,7 @@
 						0,
 						dataBuf,
 						sizeof(ParameterIndex),
-						(SDWORD*)lenBuf);
+						(SQLINTEGER*)lenBuf);
 
 	// Save the input stream
 	boundParams[ParameterIndex - 1].setInputStream (x, length);
--- source/drivers/odbcbase/OTools.cxx	2009-01-13 05:34:17.000000000 -0800
+++ source/drivers/odbcbase/OTools.cxx	2014-05-16 12:12:25.000000000 -0700
@@ -444,7 +444,7 @@
     RTL_LOGFILE_CONTEXT_AUTHOR( aLogger, "odbc", "Ocke.Janssen@sun.com", "OTools::ThrowException" );
 
 	SDB_ODBC_CHAR szSqlState[5];
-	SDWORD pfNativeError;
+	SQLINTEGER pfNativeError;
 	SDB_ODBC_CHAR szErrorMessage[SQL_MAX_MESSAGE_LENGTH];
 	szErrorMessage[0] = '\0';
 	SWORD pcbErrorMsg = 0;
@@ -622,7 +622,7 @@
 			if(_bWasNull)
 				return ::rtl::OUString();
 
-			SQLINTEGER nLen = pcbValue != SQL_NO_TOTAL ? std::min(pcbValue, nMaxLen) : (nMaxLen-1);
+			SQLINTEGER nLen = pcbValue != SQL_NO_TOTAL ? std::min(pcbValue, (SQLINTEGER)nMaxLen) : (nMaxLen-1);
 			aCharArray[nLen] = 0;
 			if ( ((pcbValue == SQL_NO_TOTAL) || pcbValue > nMaxLen) && aCharArray[nLen-1] == 0 && nLen > 0 )
 				--nLen;
@@ -643,7 +643,7 @@
 												(SQLINTEGER)nMaxLen,
 												&pcbValue),
 									_aStatementHandle,SQL_HANDLE_STMT,_xInterface);
-				nLen = pcbValue != SQL_NO_TOTAL ? std::min(pcbValue, nMaxLen) : (nMaxLen-1);
+				nLen = pcbValue != SQL_NO_TOTAL ? std::min(pcbValue, (SQLINTEGER)nMaxLen) : (nMaxLen-1);
 				if ( ((pcbValue == SQL_NO_TOTAL) || pcbValue > nMaxLen) && aCharArray[nLen-1] == 0 && nLen > 0 )
 					--nLen;
 				aCharArray[nLen] = 0;
--- source/inc/odbc/OFunctiondefs.hxx	2008-04-10 04:43:33.000000000 -0700
+++ source/inc/odbc/OFunctiondefs.hxx	2014-05-16 12:20:09.000000000 -0700
@@ -136,7 +136,13 @@
 #define CALLBACK
 #define EXPORT
 #ifdef SYSTEM_ODBC_HEADERS
+#ifdef MACOSX
+#include <premac.h>
+#endif
 #include <sqlext.h>
+#ifdef MACOSX
+#include <postmac.h>
+#endif
 #else
 #include <odbc/sqlext.h>
 #endif
--- source/drivers/jdbc/ResultSet.cxx	2009-02-16 09:52:44.000000000 -0800
+++ source/drivers/jdbc/ResultSet.cxx	2015-06-23 19:08:25.000000000 -0700
@@ -953,7 +953,7 @@
 Reference< XInterface > SAL_CALL java_sql_ResultSet::getStatement(  ) throw(SQLException, RuntimeException)
 {
     RTL_LOGFILE_CONTEXT_AUTHOR( aLogger, "jdbc", "Ocke.Janssen@sun.com", "java_sql_ResultSet::getStatement" );
-	return m_xStatement;
+	return Reference< XInterface >( m_xStatement );
 }
 // -------------------------------------------------------------------------
 
