--- connectivity/source/drivers/file/FResultSet.cxx	2009-02-06 03:35:13.000000000 -0800
+++ connectivity/source/drivers/file/FResultSet.cxx	2012-09-24 13:16:52.000000000 -0700
@@ -475,7 +475,7 @@
 	checkDisposed(OResultSet_BASE::rBHelper.bDisposed);
 
 
-	return m_xStatement;
+	return ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface >(m_xStatement);
 }
 // -------------------------------------------------------------------------
 
--- connectivity/source/drivers/mozab/MResultSet.cxx	2008-12-30 06:32:01.000000000 -0800
+++ connectivity/source/drivers/mozab/MResultSet.cxx	2012-09-29 17:38:29.000000000 -0700
@@ -551,7 +551,7 @@
     ResultSetEntryGuard aGuard( *this );
 
     OSL_TRACE("In/Out: OResultSet::getStatement" );
-	return m_xStatement;
+	return Reference< XInterface >( m_xStatement );
 }
 // -------------------------------------------------------------------------
 
