Index: source/makefile.mk
===================================================================
RCS file: /cvs/udk/cppuhelper/source/makefile.mk,v
retrieving revision 1.54
diff -u -r1.54 makefile.mk
--- source/makefile.mk	9 Jun 2008 12:24:19 -0000	1.54
+++ source/makefile.mk	26 Oct 2008 23:16:59 -0000
@@ -49,6 +49,10 @@
 
 .INCLUDE :  settings.mk
 
+.IF "$(OS)" == "MACOSX"
+CDEFS+=-g -O0
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
