Index: source/makefile.mk
===================================================================
RCS file: /cvs/udk/cppuhelper/source/makefile.mk,v
retrieving revision 1.54
diff -u -r1.54 makefile.mk
--- source/makefile.mk	9 Jun 2008 12:24:19 -0000	1.54
+++ source/makefile.mk	26 Oct 2008 23:16:59 -0000
@@ -49,6 +49,10 @@
 
 .INCLUDE :  settings.mk
 
+.IF "$(OS)" == "MACOSX"
+CDEFS+=-O0
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
--- inc/cppuhelper/propertysetmixin.hxx	2008-04-11 04:21:36.000000000 -0700
+++ inc/cppuhelper/propertysetmixin.hxx	2013-08-13 04:49:36.000000000 -0700
@@ -164,8 +164,8 @@
         void notify() const;
 
     private:
-        BoundListeners(BoundListeners &); // not defined
-        void operator =(BoundListeners); // not defined
+        BoundListeners( const BoundListeners&); // not defined
+        void operator=( const BoundListeners&); // not defined
 
         class Impl;
         Impl * m_impl;
@@ -386,8 +379,8 @@
             com::sun::star::uno::RuntimeException);
 
 private:
-    PropertySetMixinImpl(PropertySetMixinImpl &); // not defined
-    void operator =(PropertySetMixinImpl &); // not defined
+    PropertySetMixinImpl( const PropertySetMixinImpl&); // not defined
+    void operator=( const PropertySetMixinImpl&); // not defined
 
     PropertySetMixinImpl(
         com::sun::star::uno::Reference< com::sun::star::uno::XComponentContext >
@@ -478,8 +471,8 @@
     ~PropertySetMixin() {}
 
 private:
-    PropertySetMixin(PropertySetMixin &); // not defined
-    void operator =(PropertySetMixin); // not defined
+    PropertySetMixin( const PropertySetMixin&); // not defined
+    void operator=( const PropertySetMixin&); // not defined
 };
 
 #if defined _MSC_VER
