Index: source/unoexe/makefile.mk
===================================================================
RCS file: /cvs/udk/cpputools/source/unoexe/makefile.mk,v
retrieving revision 1.15
diff -u -r1.15 makefile.mk
--- source/unoexe/makefile.mk	11 Apr 2008 11:49:39 -0000	1.15
+++ source/unoexe/makefile.mk	26 Oct 2008 23:17:52 -0000
@@ -39,6 +39,10 @@
 
 .INCLUDE :  settings.mk
 
+.IF "$(OS)" == "MACOSX"
+CDEFS+=-O0
+.ENDIF
+
 UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb 
 UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
 
