--- source/core/api/definitioncolumn.cxx	2008-04-10 05:32:11.000000000 -0700
+++ source/core/api/definitioncolumn.cxx	2012-09-25 22:04:45.000000000 -0700
@@ -338,7 +338,7 @@
 Reference< XInterface > SAL_CALL OTableColumnDescriptor::getParent(  ) throw (RuntimeException)
 {
 	::osl::MutexGuard aGuard(m_aMutex);
-	return m_xParent;
+	return Reference< XInterface >(m_xParent);
 }
 // -----------------------------------------------------------------------------
 void SAL_CALL OTableColumnDescriptor::setParent( const Reference< XInterface >& _xParent ) throw (NoSupportException, RuntimeException)
--- source/core/api/RowSetBase.cxx	2009-02-16 09:52:44.000000000 -0800
+++ source/core/api/RowSetBase.cxx	2012-09-25 22:07:28.000000000 -0700
@@ -1153,7 +1153,7 @@
 //------------------------------------------------------------------------------
 Reference< XInterface >  ORowSetBase::getStatement(void) throw( SQLException, RuntimeException )
 {
-	return NULL;
+	return Reference< XInterface >(NULL);
 }
 // -------------------------------------------------------------------------
 void SAL_CALL ORowSetBase::refreshRow(  ) throw(SQLException, RuntimeException)
--- source/core/dataaccess/bookmarkcontainer.cxx	2008-04-10 05:41:17.000000000 -0700
+++ source/core/dataaccess/bookmarkcontainer.cxx	2012-09-25 22:06:00.000000000 -0700
@@ -410,7 +410,7 @@
 //--------------------------------------------------------------------------
 Reference< XInterface > SAL_CALL OBookmarkContainer::getParent(  ) throw (RuntimeException)
 {
-	return m_rParent;
+	return Reference< XInterface >(m_rParent);
 }
 
 //--------------------------------------------------------------------------
--- source/ui/dlg/odbcconfig.cxx	2008-05-13 07:03:03.000000000 -0700
+++ source/ui/dlg/odbcconfig.cxx	2014-05-16 15:52:05.000000000 -0700
@@ -98,7 +98,13 @@
 #endif
 
 #ifdef SYSTEM_ODBC_HEADERS
+#ifdef MACOSX
+#include <premac.h>
+#endif
 #include <sqlext.h>
+#ifdef MACOSX
+#include <postmac.h>
+#endif
 #else
 #ifndef __SQLEXT_H
 #include <odbc/sqlext.h>
