Index: dlcompat.pat
===================================================================
RCS file: /cvs/external/dlcompat/dlcompat.pat,v
retrieving revision 1.1
diff -u -r1.1 dlcompat.pat
--- dlcompat.pat	22 Mar 2004 15:45:19 -0000	1.1
+++ dlcompat.pat	25 May 2004 22:10:49 -0000
@@ -1,4 +1,19 @@
-115c115
-< 	$(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@ $(VERSTRING)
----
-> 	$(CC) -dynamiclib -o $@ $(OBJS) -install_name @executable_path/$@ $(VERSTRING)
+*** misc/dlcompat-20030629/Makefile.in	Sat Jun 28 09:50:34 2003
+--- misc/build/dlcompat-20030629/Makefile.in	Tue May 25 14:44:12 2004
+***************
+*** 112,118 ****
+  	ln -sf libdl.$(LIBVER).dylib $@
+  
+  libdl.$(LIBVER).dylib: $(OBJS)
+! 	$(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@ $(VERSTRING)
+  
+  clean:
+  	rm -f $(OBJS) libdl.* *~ *.o dlopen.3
+--- 112,118 ----
+  	ln -sf libdl.$(LIBVER).dylib $@
+  
+  libdl.$(LIBVER).dylib: $(OBJS)
+! 	$(CC) -dynamiclib -o $@ $(OBJS) -install_name @executable_path/$@ $(VERSTRING)
+  
+  clean:
+  	rm -f $(OBJS) libdl.* *~ *.o dlopen.3
Index: makefile.mk
===================================================================
RCS file: /cvs/external/dlcompat/makefile.mk,v
retrieving revision 1.1
diff -u -r1.1 makefile.mk
--- makefile.mk	22 Mar 2004 15:45:19 -0000	1.1
+++ makefile.mk	25 May 2004 22:10:49 -0000
@@ -77,13 +77,18 @@
 .IF "$(BUILD_OS_MINOR)" == "2" 
 
 TARFILE_NAME=dlcompat-20030629
-PATCH_FILE_NAME=dlcompat.patch
+PATCH_FILE_NAME=dlcompat.pat
 
-BUILD_DIR=dlcompat-20030629
+CONFIGURE_DIR=.
+CONFIGURE_ACTION=./configure
+BUILD_DIR=$(CONFIGURE_DIR)
 BUILD_ACTION=make -f Makefile clean all
 
 OUT2INC= \
-    dlcompat-20030629/dlfcn.h
+    dlfcn.h
+
+OUT2LIB= \
+	libdl*$(DLLPOST)
 
 .ELSE
 verbose:
