--- source/aqua/DataFlavorMapping.hxx	2008-04-11 00:20:19.000000000 -0700
+++ source/aqua/DataFlavorMapping.hxx	2012-07-11 21:43:17.000000000 -0700
@@ -125,7 +125,7 @@
 
   /* Returns an NSArray containing all pasteboard types supported by OOo
    */
-  NSArray* DataFlavorMapper::getAllSupportedPboardTypes() const;
+  NSArray* getAllSupportedPboardTypes() const;
 
 private:
   /* Determines if the provided Mime content type is valid.
