--- source/plugin/base/nfuncs.cxx	2008-04-11 03:22:47.000000000 -0700
+++ source/plugin/base/nfuncs.cxx	2012-09-26 23:41:12.000000000 -0700
@@ -102,8 +102,8 @@
 	NPN_MemFree,
 	NPN_MemFlush,
 	NPN_ReloadPlugins,
-	NPN_GetJavaEnv,
-	NPN_GetJavaPeer,
+	NULL,
+	NULL,
 	NPN_GetURLNotify,
 	NPN_PostURLNotify,
 	NPN_GetValue,
--- source/config/ldap/makefile.mk	2008-04-11 02:48:07.000000000 -0700
+++ source/config/ldap/makefile.mk	2014-04-28 11:55:39.000000000 -0700
@@ -44,7 +44,7 @@
 .INCLUDE : settings.mk
 DLLPRE =
 
-.IF "$(WITH_LDAP)" != "YES"
+.IF "$(WITH_LDAP)" != "YES" || "$(UPD)" == "310"
 @all:
 	@echo "LDAP disabled."
 .ENDIF
