Index: source/plugin/base/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/base/makefile.mk,v
retrieving revision 1.9
diff -u -r1.9 makefile.mk
--- source/plugin/base/makefile.mk	20 Jan 2006 12:58:03 -0000	1.9
+++ source/plugin/base/makefile.mk	11 Mar 2006 08:38:52 -0000
@@ -48,6 +48,8 @@
 CDEFS+=-DOJI
 .ENDIF
 
+.IF "$(WITH_MOZILLA)" != "NO"
+
 SLOFILES=		\
 				$(SLO)$/plctrl.obj		\
 				$(SLO)$/service.obj		\
@@ -60,6 +62,8 @@
 				$(SLO)$/multiplx.obj    \
 				$(SLO)$/plmodel.obj
 
+.ENDIF # $(WITH_MOZILLA) != "NO"
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
Index: source/plugin/unx/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/unx/makefile.mk,v
retrieving revision 1.9
diff -u -r1.9 makefile.mk
--- source/plugin/unx/makefile.mk	20 Jan 2006 12:59:15 -0000	1.9
+++ source/plugin/unx/makefile.mk	11 Mar 2006 08:38:52 -0000
@@ -49,6 +49,8 @@
 CDEFS+=-DOJI
 .ENDIF
 
+.IF "$(WITH_MOZILLA)" != "NO"
+
 SLOFILES=\
 	$(SLO)$/nppapi.obj		\
 	$(SLO)$/sysplug.obj		\
@@ -80,6 +82,7 @@
 
 APP1DEF=	$(MISC)$/$(TARGET).def
 
+.ENDIF # $(WITH_MOZILLA) != "NO"
 
 # --- Targets ------------------------------------------------------
 
Index: source/plugin/util/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/util/makefile.mk,v
retrieving revision 1.12
diff -u -r1.12 makefile.mk
--- source/plugin/util/makefile.mk	8 Sep 2005 19:57:10 -0000	1.12
+++ source/plugin/util/makefile.mk	11 Mar 2006 08:38:52 -0000
@@ -43,6 +43,9 @@
 .INCLUDE :  settings.mk
 
 # --- Files --------------------------------------------------------
+
+.IF "$(WITH_MOZILLA)" != "NO"
+
 LIB1TARGET = $(SLB)$/plall.lib
 LIB1FILES  = \
 	$(SLB)$/plbase.lib	\
@@ -97,6 +100,8 @@
 
 SHL1STDLIBS+=$(SHL1OWNLIBS)
 
+.ENDIF # $(WITH_MOZILLA) != "NO"
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
Index: source/plugin/win/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/win/makefile.mk,v
retrieving revision 1.5
diff -u -r1.5 makefile.mk
--- source/plugin/win/makefile.mk	20 Jan 2006 12:59:58 -0000	1.5
+++ source/plugin/win/makefile.mk	11 Mar 2006 08:38:53 -0000
@@ -53,6 +53,8 @@
 
 # --- Types -------------------------------------
 
+.IF "$(WITH_MOZILLA)" != "NO"
+
 #UNOUCROUT=$(OUT)$/inc$/$(PRJNAME)
 #INCPRE+=$(UNOUCROUT)
 
@@ -61,7 +63,9 @@
 SLOFILES =	$(SLO)$/winmgr.obj	\
 			$(SLO)$/sysplug.obj
 
+.ENDIF # $(WITH_MOZILLA) != "NO"
 .ENDIF
+
 # ------------------------------------------------------------------
 
 
