Index: prj/build.lst
===================================================================
RCS file: /cvs/oo/util/external/prj/build.lst,v
retrieving revision 1.5.2.1.2.1
diff -u -r1.5.2.1.2.1 build.lst
--- prj/build.lst	30 Apr 2003 06:28:22 -0000	1.5.2.1.2.1
+++ prj/build.lst	13 Jun 2003 16:32:30 -0000
@@ -3,6 +3,8 @@
 es	external\ado		nmake	-	w	es_ado NULL
 es	external\atl		nmake	-	w	es_atl NULL
 es	external\dt		nmake	-	u	es_dt NULL
+es	external\dlcompat	nmake	-	u	es_dl NULL
+es	external\dlcompat\rtufiles	nmake	-	u	es_dlrtu es_dl.u NULL
 es	external\glibc		nmake	-	all	es_glibc NULL
 es	external\glibc\rtufiles	nmake	-	all	es_glibcrtu es_glibc NULL
 es	external\gpc		nmake	-	u	es_gpc NULL
Index: prj/d.lst
===================================================================
RCS file: /cvs/oo/util/external/prj/d.lst,v
retrieving revision 1.15.12.1
diff -u -r1.15.12.1 d.lst
--- prj/d.lst	30 Apr 2003 06:28:22 -0000	1.15.12.1
+++ prj/d.lst	13 Jun 2003 16:32:31 -0000
@@ -34,6 +34,9 @@
 ..\atl\rtufiles\*.idl %_DEST%\inc%_EXT%\external\atl\*.idl
 ..\atl\rtufiles\*.cpp %_DEST%\inc%_EXT%\external\atl\*.cpp
 ..\dt\rtufiles\*.h %_DEST%\inc%_EXT%\external\dt\*.h
+..\dlcompat\rtufiles\dlfcn.h %_DEST%\inc%_EXT%\external\dlfcn.h
+..\dlcompat\rtufiles\libdl*.a %_DEST%\lib%_EXT%\libdl*.a
+..\dlcompat\rtufiles\libdl*.dylib %_DEST%\lib%_EXT%\libdl*.dylib
 ..\glibc\rtufiles\config.h %_DEST%\inc%_EXT%\external\glibc\config.h
 ..\glibc\rtufiles\getopt.h %_DEST%\inc%_EXT%\external\glibc\getopt.h
 ..\gpc\gpc.h %_DEST%\inc%_EXT%\external\gpc\gpc.h
Index: dlcompat/dlcompatpatch.sh
diff -r -c dlcompat/dlcompatpatch.sh dlcompat/dlcompatpatch.sh
*** dlcompat/dlcompatpatch.sh	Fri Sep 27 20:06:26 2002
--- dlcompat/dlcompatpatch.sh	Fri Sep 27 20:03:57 2002
***************
*** 0 ****
--- 1,113 ----
+ #!/bin/sh
+ ##########################################################################
+ #
+ #   $RCSfile$
+ #
+ #   $Revision$
+ #
+ #   last change: $Author$ $Date$
+ #
+ #   The Contents of this file are made available subject to the terms of
+ #   either of the following licenses
+ #
+ #          - GNU General Public License Version 2.1
+ #
+ #   Sun Microsystems Inc., October, 2000
+ #
+ #   GNU General Public License Version 2.1
+ #   =============================================
+ #   Copyright 2000 by Sun Microsystems, Inc.
+ #   901 San Antonio Road, Palo Alto, CA 94303, USA
+ #
+ #   This library is free software; you can redistribute it and/or
+ #   modify it under the terms of the GNU General Public
+ #   License version 2.1, as published by the Free Software Foundation.
+ #
+ #   This library is distributed in the hope that it will be useful,
+ #   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ #   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ #   General Public License for more details.
+ #
+ #   You should have received a copy of the GNU General Public
+ #   License along with this library; if not, write to the Free Software
+ #   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ #   MA  02111-1307  USA
+ #
+ #   =================================================
+ #   Modified June 2003 by Patrick Luby. SISSL Removed. NeoOffice is
+ #   distributed under GPL only under modification term 3 of the LGPL.
+ #
+ #   Contributor(s): _______________________________________
+ #
+ ##########################################################################
+ 
+ tarball="../download/dlcompat-20020709.tar.gz"
+ unziptarball="../download/dlcompat-20020709.tar.gz"
+ 
+ if test -s "$tarball"; then
+   if test "$OS" = "MACOSX"; then
+ #Only MACOSX 
+     tar -x -z -f $tarball `cat flist.cp`
+     tar -x -z -f dlcompat.pat.tar.gz
+   fi
+ 
+ #
+ # convert the files into unix format
+ #
+ 
+   for CPNAME in `cat flist.cp`
+   do
+      if test -s "$CPNAME"; then
+       mv -f $CPNAME ./pat.tmp
+       sed -e 's/$//' ./pat.tmp > $CPNAME
+       rm -f pat.tmp
+      fi
+   done
+ 
+ #
+ # move the files into the final place
+ #
+ 
+   for FNAME in `cat flist.cp`
+   do
+     cp $FNAME rtufiles/
+   done
+   rm -rf dlcompat-20020709
+ 
+ #
+ # apply the patches
+ #
+ 
+   for PNAME in `cat flist.pat`
+   do
+       patch -i pat/$PNAME.pat rtufiles/$PNAME
+   done
+ 
+ #
+ # copy the SD files into rtufiles directory
+ #
+ 
+   for SDNAME in `cat flist.sd`
+   do
+     SDDIR=`dirname $SDNAME`
+     if test ! -d "rtufiles/$SDDIR"; then
+       mkdir -p rtufiles/$SDDIR
+     fi
+     cp sdfiles/$SDNAME rtufiles/$SDNAME
+   done
+ 
+   echo Patch done. > patchinf
+   echo Patch done.
+   rm -f patcherror
+   rm -rf pat
+ 
+ else
+ 
+ #
+ # error message
+ #
+ 
+   echo "Can't find the $tarball file. See patcherror file."
+   echo "Source tarball is missing: $tarball " > patcherror
+   echo "Copy it into <OPENOFFICE>/external/download directory. " >> patcherror
+ fi
Index: dlcompat/flist.cp
diff -r -c dlcompat/flist.cp dlcompat/flist.cp
*** dlcompat/flist.cp	Fri Sep 27 20:03:59 2002
--- dlcompat/flist.cp	Fri Sep 27 20:03:57 2002
***************
*** 0 ****
--- 1,3 ----
+ dlcompat-20020709/dlfcn.c
+ dlcompat-20020709/dlfcn.h
+ dlcompat-20020709/Makefile
Index: dlcompat/flist.pat
diff -r -c dlcompat/flist.pat dlcompat/flist.pat
*** dlcompat/flist.pat	Fri Sep 27 20:04:01 2002
--- dlcompat/flist.pat	Fri Sep 27 20:03:57 2002
***************
*** 0 ****
--- 1 ----
+ Makefile
Index: dlcompat/flist.sd
diff -r -c dlcompat/flist.sd dlcompat/flist.sd
*** dlcompat/flist.sd	Fri Sep 27 20:04:02 2002
--- dlcompat/flist.sd	Fri Sep 27 20:03:32 2002
***************
*** 0 ****
--- 1 ----
+ makefile.mk
Index: dlcompat/makefile.mk
diff -r -c dlcompat/makefile.mk dlcompat/makefile.mk
*** dlcompat/makefile.mk	Fri Sep 27 20:04:08 2002
--- dlcompat/makefile.mk	Fri Sep 27 20:03:57 2002
***************
*** 0 ****
--- 1,61 ----
+ ##########################################################################
+ #
+ #   $RCSfile$
+ #
+ #   $Revision$
+ #
+ #   last change: $Author$ $Date$
+ #
+ #   The Contents of this file are made available subject to the terms of
+ #   either of the following licenses
+ #
+ #          - GNU General Public License Version 2.1
+ #
+ #   Sun Microsystems Inc., October, 2000
+ #
+ #   GNU General Public License Version 2.1
+ #   =============================================
+ #   Copyright 2000 by Sun Microsystems, Inc.
+ #   901 San Antonio Road, Palo Alto, CA 94303, USA
+ #
+ #   This library is free software; you can redistribute it and/or
+ #   modify it under the terms of the GNU General Public
+ #   License version 2.1, as published by the Free Software Foundation.
+ #
+ #   This library is distributed in the hope that it will be useful,
+ #   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ #   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ #   General Public License for more details.
+ #
+ #   You should have received a copy of the GNU General Public
+ #   License along with this library; if not, write to the Free Software
+ #   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ #   MA  02111-1307  USA
+ #
+ #   =================================================
+ #   Modified June 2003 by Patrick Luby. SISSL Removed. NeoOffice is
+ #   distributed under GPL only under modification term 3 of the LGPL.
+ #
+ #   Contributor(s): _______________________________________
+ #
+ ##########################################################################
+ 
+ MAKEFILERC=yes
+ 
+ .IF "$(OS)"=="MACOSX"
+ 
+ patchsuffix=sh
+ 
+ ALL: patchinf \
+ 	ALLTAR
+ 
+ patchinf:
+ 	@+dlcompatpatch.$(patchsuffix)
+ 
+ clean:
+ 	@+rm -f patchinf
+ 	@+rm -f patcherror
+ 
+ .ENDIF	# "$(OS)"=="MACOSX"
+ 
+ .INCLUDE :  target.mk
Index: dlcompat/rtufiles/dummy
diff -r -c dlcompat/rtufiles/dummy dlcompat/rtufiles/dummy
*** dlcompat/rtufiles/dummy	Fri Sep 27 20:04:18 2002
--- dlcompat/rtufiles/dummy	Fri Sep 27 20:03:57 2002
***************
*** 0 ****
--- 1 ----
+ dummy for keeping this directory alive
Index: dlcompat/sdfiles/makefile.mk
diff -r -c dlcompat/sdfiles/makefile.mk dlcompat/sdfiles/makefile.mk
*** dlcompat/sdfiles/makefile.mk	Fri Sep 27 20:04:44 2002
--- dlcompat/sdfiles/makefile.mk	Fri Sep 27 20:03:32 2002
***************
*** 0 ****
--- 1,69 ----
+ ##########################################################################
+ #
+ #   $RCSfile$
+ #
+ #   $Revision$
+ #
+ #   last change: $Author$ $Date$
+ #
+ #   The Contents of this file are made available subject to the terms of
+ #   either of the following licenses
+ #
+ #          - GNU General Public License Version 2.1
+ #
+ #   Sun Microsystems Inc., October, 2000
+ #
+ #   GNU General Public License Version 2.1
+ #   =============================================
+ #   Copyright 2000 by Sun Microsystems, Inc.
+ #   901 San Antonio Road, Palo Alto, CA 94303, USA
+ #
+ #   This library is free software; you can redistribute it and/or
+ #   modify it under the terms of the GNU General Public
+ #   License version 2.1, as published by the Free Software Foundation.
+ #
+ #   This library is distributed in the hope that it will be useful,
+ #   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ #   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ #   General Public License for more details.
+ #
+ #   You should have received a copy of the GNU General Public
+ #   License along with this library; if not, write to the Free Software
+ #   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ #   MA  02111-1307  USA
+ #
+ #   =================================================
+ #   Modified June 2003 by Patrick Luby. SISSL Removed. NeoOffice is
+ #   distributed under GPL only under modification term 3 of the LGPL.
+ #
+ #   Contributor(s): _______________________________________
+ #
+ ##########################################################################
+ 
+ PRJ=..$/..
+ 
+ PRJNAME=external
+ TARGET=dl
+ LIBTARGET=NO
+ 
+ # --- Settings -----------------------------------------------------
+ 
+ .INCLUDE :  svpre.mk
+ .INCLUDE :  settings.mk
+ .INCLUDE :  sv.mk
+ 
+ # --- Files --------------------------------------------------------
+ 
+ .IF "$(OS)"=="MACOSX"
+ 
+ ALL: dlcompatbuild \
+     ALLTAR 
+  
+ dlcompatbuild:
+     @+make -f Makefile clean all
+ 
+ .ENDIF	# "$(OS)"=="MACOSX"
+ 
+ # --- Targets ------------------------------------------------------
+ 
+ .INCLUDE :  target.mk
