Index: prj/build.lst
===================================================================
RCS file: /cvs/util/external/prj/build.lst,v
retrieving revision 1.12
diff -u -r1.12 build.lst
--- prj/build.lst	18 Jun 2003 12:44:11 -0000	1.12
+++ prj/build.lst	25 May 2004 22:11:48 -0000
@@ -1,4 +1,4 @@
-es      external :       soltools NULL
+es      external :       soltools dlcompat NULL
 es	external					usr1 - all es_mkout NULL
 es	external\ado		nmake	-	w	es_ado NULL
 es	external\glibc		nmake	-	all	es_glibc NULL
