Index: source/scanner/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/scanner/makefile.mk,v
retrieving revision 1.6.30.1
diff -u -r1.6.30.1 makefile.mk
--- source/scanner/makefile.mk	28 Jan 2004 12:06:04 -0000	1.6.30.1
+++ source/scanner/makefile.mk	2 Jun 2004 18:51:50 -0000
@@ -92,10 +92,7 @@
 	com.sun.star.scanner.ScannerException						
 
 # --- Files --------------------------------------------------------
-.IF "$(OS)"=="MACOSX"
-dummy:
-	@echo "Nothing to build for Mac OS X"
-.ELSE
+
 SLOFILES=\
 				$(SLO)$/scnserv.obj		\
 				$(SLO)$/scanner.obj
@@ -150,8 +147,6 @@
 DEF1NAME=$(SHL1TARGET)
 DEF1EXPORTFILE=exports.dxp
 
-.ENDIF		# "$(OS)"=="MACOSX"
-
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
