--- source/aqua/CFStringUtilities.cxx	2008-04-11 02:12:33.000000000 -0700
+++ source/aqua/CFStringUtilities.cxx	2012-07-13 01:03:40.000000000 -0700
@@ -89,11 +89,14 @@
 
     switch(info) {
         case FULLPATH:
+        {
             OSL_TRACE("Extracting the full path of an item");
             sURLString = CFURLGetString(aUrlRef);
             CFRetain(sURLString);
             break;
+        }
         case FILENAME:
+        {
             OSL_TRACE("Extracting the file name of an item");
             CFStringRef fullString = CFURLGetString(aUrlRef);
             CFURLRef dirRef = CFURLCreateCopyDeletingLastPathComponent(NULL,aUrlRef);
@@ -103,13 +106,16 @@
             CFRange substringRange = CFRangeMake(dirLength, fullLength - dirLength);
             sURLString = CFStringCreateWithSubstring(NULL, fullString, substringRange);
             break;
+        }
         case PATHWITHOUTLASTCOMPONENT:
+        {
             OSL_TRACE("Extracting the last but one component of an item's path");
             CFURLRef directoryRef = CFURLCreateCopyDeletingLastPathComponent(NULL,aUrlRef);
             sURLString = CFURLGetString(directoryRef);
             CFRetain(sURLString);
             CFRelease(directoryRef);
             break;
+        }
         default:
             break;
     }
--- source/aqua/NSURL_OOoAdditions.mm	2008-04-11 02:16:52.000000000 -0700
+++ source/aqua/NSURL_OOoAdditions.mm	2012-07-13 01:05:36.000000000 -0700
@@ -43,11 +43,14 @@
 
     switch(info) {
         case FULLPATH:
+        {
             OSL_TRACE("Extracting the full path of an item");
             sURLString = [self absoluteString];
             [sURLString retain];
             break;
+        }
         case FILENAME:
+        {
             OSL_TRACE("Extracting the file name of an item");
             NSString *path = [self path];
             if (path == nil) {
@@ -58,9 +61,11 @@
             }
             [sURLString retain];
             break;
+        }
         case PATHWITHOUTLASTCOMPONENT:
+        {
             OSL_TRACE("Extracting the last but one component of an item's path");
-            path = [self absoluteString];
+            NSString *path = [self absoluteString];
             if (path == nil) {
                 sURLString = @"";
             }
@@ -71,6 +76,7 @@
             }
             [sURLString retain];
             break;
+        }
         default:
             break;
     }
