--- Library_fwk.mk	2014-02-25 00:30:06.000000000 -0800
+++ Library_fwk.mk	2015-09-30 18:28:02.000000000 -0700
@@ -109,7 +109,6 @@
 	framework/source/layoutmanager/panelmanager \
 	framework/source/layoutmanager/toolbarlayoutmanager \
 	framework/source/layoutmanager/uielement \
-	framework/source/loadenv/loadenv \
 	framework/source/loadenv/targethelper \
 	framework/source/register/registerservices \
 	framework/source/services/ContextChangeEventMultiplexer \
@@ -186,4 +185,17 @@
 	framework/source/xml/imagesdocumenthandler \
 ))
 
+ifeq ($(OS),MACOSX)
+# Fix infinite loop when pressing the "Create Table in Design View" link in
+# a Base database by disabling optimization
+$(eval $(call gb_Library_add_cxxobjects,fwk,\
+	framework/source/loadenv/loadenv \
+    , $(gb_COMPILERNOOPTFLAGS) $(gb_LinkTarget_EXCEPTIONFLAGS) \
+))
+else
+$(eval $(call gb_Library_add_exception_objects,fwk,\
+	framework/source/loadenv/loadenv \
+))
+endif
+
 # vim: set noet sw=4 ts=4:
