Index: source/localedata/makefile.mk
===================================================================
RCS file: /cvs/l10n/i18npool/source/localedata/makefile.mk,v
retrieving revision 1.12
diff -u -r1.12 makefile.mk
--- source/localedata/makefile.mk	10 Apr 2008 09:05:39 -0000	1.12
+++ source/localedata/makefile.mk	11 Aug 2008 08:29:35 -0000
@@ -58,6 +58,10 @@
 CDEFS+=-D__STD_LIMITS
 .ENDIF
 
+.IF "$(OS)" == "MACOSX"
+CDEFS+=-O0
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=   $(SLO)$/localedata.obj
