--- source/preproc/unix.c	2008-04-11 03:40:50.000000000 -0700
+++ source/preproc/unix.c	2012-07-11 20:49:02.000000000 -0700
@@ -218,7 +218,7 @@
 /* memmove is defined here because some vendors don't provide it at
    all and others do a terrible job (like calling malloc) */
 
-#if !defined(__IBMC__) && !defined(WNT) && !defined(__GLIBC__)
+#if !defined(__IBMC__) && !defined(WNT) && !defined(__GLIBC__) && !defined(__GNUC__)
 
 void *
     memmove(void *dp, const void *sp, size_t n)
