--- prj/build.lst.bak	2014-02-25 00:17:13.000000000 -0800
+++ prj/build.lst	2016-04-20 22:17:58.000000000 -0700
@@ -1,4 +1,4 @@
-oon	instsetoo_native	::	L10N:l10n postprocess packimages RAT:ratscan NULL
+oon	instsetoo_native	::	L10N:l10n postprocess packimages RAT:ratscan writerperfect NULL
 oon	instsetoo_native						usr1	-	all	oon_mkout NULL
 oon	instsetoo_native\inc_openoffice\unix	nmake	-	u	oon_unix NULL
 oon	instsetoo_native\inc_openoffice\windows\msi_languages	nmake	-	all	oon_msilang NULL
