Index: util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo_native/util/makefile.mk,v
retrieving revision 1.48
diff -u -r1.48 makefile.mk
--- util/makefile.mk	18 Nov 2005 16:14:01 -0000	1.48
+++ util/makefile.mk	27 Dec 2005 20:18:59 -0000
@@ -103,7 +103,7 @@
 	@echo "No EPM: do no packaging at this stage"
 .ELSE			# "$(GUI)"!="WNT" && "$(EPM)"=="NO" && "$(USE_PACKAGER)"==""
 .IF "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
-ALLTAR : openoffice sdkoo_en-US ure_en-US
+ALLTAR : openoffice sdkoo_en-US ure_en-US ooolanguagepack
 .ELSE			# "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 ALLTAR : updatepack
 .ENDIF			# "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
@@ -122,7 +122,7 @@
 
 .IF "$(alllangiso)"!=""
 
-openoffice: $(foreach,i,$(alllangiso) openoffice_$i)
+openoffice: openoffice_en-US
 
 openofficedev: $(foreach,i,$(alllangiso) openofficedev_$i)
 
