Index: util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo_native/util/makefile.mk,v
retrieving revision 1.62.2.2
diff -u -r1.62.2.2 makefile.mk
--- util/makefile.mk	1 Mar 2007 11:46:53 -0000	1.62.2.2
+++ util/makefile.mk	9 May 2007 21:37:23 -0000
@@ -109,9 +109,9 @@
 .ELSE			# "$(GUI)"!="WNT" && "$(EPM)"=="NO" && "$(USE_PACKAGER)"==""
 .IF "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 .IF "$(BUILD_TYPE)"=="$(BUILD_TYPE:s/ODK//)"
-ALLTAR : openoffice
+ALLTAR : openoffice_en-US ooolanguagepack
 .ELSE
-ALLTAR : openoffice sdkoo_en-US ure_en-US
+ALLTAR : openoffice_en-US sdkoo_en-US ure_en-US ooolanguagepack
 .ENDIF
 .ELSE			# "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 ALLTAR : updatepack