Index: util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo_native/util/makefile.mk,v
retrieving revision 1.56
diff -u -r1.56 makefile.mk
--- util/makefile.mk	26 Apr 2006 20:45:16 -0000	1.56
+++ util/makefile.mk	30 Jun 2006 19:26:21 -0000
@@ -100,7 +100,7 @@
 .IF "$(BUILD_TYPE)"=="$(BUILD_TYPE):s/ODK//"
 ALLTAR : openoffice
 .ELSE
-ALLTAR : openoffice sdkoo_en-US ure_en-US
+ALLTAR : openoffice sdkoo_en-US ure_en-US ooolanguagepack
 .ENDIF
 .ELSE			# "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 ALLTAR : updatepack
@@ -120,7 +120,7 @@
 
 .IF "$(alllangiso)"!=""
 
-openoffice: $(foreach,i,$(alllangiso) openoffice_$i)
+openoffice: openoffice_en-US
 
 openofficedev: $(foreach,i,$(alllangiso) openofficedev_$i)
 
