Index: source/thesaurus/mythes/mythes.cxx
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/mythes/mythes.cxx,v
retrieving revision 1.7
diff -u -r1.7 mythes.cxx
--- source/thesaurus/mythes/mythes.cxx	16 Sep 2006 16:13:38 -0000	1.7
+++ source/thesaurus/mythes/mythes.cxx	11 Jan 2007 21:54:39 -0000
@@ -52,6 +52,10 @@
     encoding = NULL;
     list = NULL;
     offst = NULL;
+#ifdef MACOSX
+    // Fix bug 1524 by properly initializing all data members
+    pdfile = NULL;
+#endif  // MACOSX
 
     if (thInitialize(idxpath, datpath) != 1) {
         fprintf(stderr,"Error - can't open %s or %s\n",idxpath, datpath);
