Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.181.22.23
diff -u -r1.181.22.23 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	25 Jun 2007 11:19:46 -0000	1.181.22.23
+++ scp2/source/ooo/file_library_ooo.scp	7 Jul 2007 17:05:14 -0000
@@ -792,6 +792,25 @@
 
 #endif
 
+#ifdef MACOSX
+
+File gid_File_Lib_Macab_1
+    TXT_FILE_BODY;
+    Styles = (PACKED,UNO_COMPONENT,PATCH);
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Dir = gid_Dir_Program;
+    Name = STRING(CONCAT2(libmacab1,UNXSUFFIX));
+End
+
+File gid_File_Lib_Macab_Drv_1
+    TXT_FILE_BODY;
+    Styles = (PACKED,PATCH);
+    Dir = gid_Dir_Program;
+    Name = STRING(CONCAT2(libmacabdrv1,UNXSUFFIX));
+End
+
+#endif
+
 #endif
 
 #ifdef ENABLE_EVOAB2
Index: scp2/source/ooo/module_hidden_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/module_hidden_ooo.scp,v
retrieving revision 1.63
diff -u -r1.63 module_hidden_ooo.scp
--- scp2/source/ooo/module_hidden_ooo.scp	20 Jun 2007 14:01:33 -0000	1.63
+++ scp2/source/ooo/module_hidden_ooo.scp	7 Jul 2007 17:05:15 -0000
@@ -313,6 +313,8 @@
 	gid_File_Lib_Evoab_1,
 	gid_File_Lib_Kab_1,
     gid_File_Lib_Kab_Drv_1,
+	gid_File_Lib_Macab_1,
+    gid_File_Lib_Macab_Drv_1,
 	gid_File_Lib_Evtatt,
 	gid_File_Lib_Fileacc,
 	gid_File_Lib_Filterconfig1,
Index: dbaccess/source/ui/app/AppController.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/app/AppController.cxx,v
retrieving revision 1.41
diff -u -r1.41 AppController.cxx
--- dbaccess/source/ui/app/AppController.cxx	9 May 2007 13:24:34 -0000	1.41
+++ dbaccess/source/ui/app/AppController.cxx	7 Jul 2007 17:05:17 -0000
@@ -785,7 +785,7 @@
 				if ( aReturn.bEnabled )
 				{
 					DATASOURCE_TYPE eType = m_aTypeCollection.getType(::comphelper::getString(m_xDataSource->getPropertyValue(PROPERTY_URL)));
-					aReturn.bEnabled = DST_EMBEDDED != eType && DST_MOZILLA != eType && DST_EVOLUTION != eType && DST_KAB != eType && DST_OUTLOOK != eType && DST_OUTLOOKEXP != eType;
+					aReturn.bEnabled = DST_EMBEDDED != eType && DST_MOZILLA != eType && DST_EVOLUTION != eType && DST_KAB != eType && DST_MACAB != eType && DST_OUTLOOK != eType && DST_OUTLOOKEXP != eType;
 				}
 				break;
 			case SID_DB_APP_DSCONNECTION_TYPE:
@@ -801,7 +801,7 @@
 				if ( aReturn.bEnabled )
 				{
 					DATASOURCE_TYPE eType = m_aTypeCollection.getType(::comphelper::getString(m_xDataSource->getPropertyValue(PROPERTY_URL)));
-					aReturn.bEnabled = DST_EMBEDDED != eType && DST_LDAP != eType && DST_CALC != eType && DST_MOZILLA != eType && DST_THUNDERBIRD != eType && DST_EVOLUTION != eType && DST_KAB != eType && DST_OUTLOOK != eType && DST_OUTLOOKEXP != eType;
+					aReturn.bEnabled = DST_EMBEDDED != eType && DST_LDAP != eType && DST_CALC != eType && DST_MOZILLA != eType && DST_THUNDERBIRD != eType && DST_EVOLUTION != eType && DST_KAB != eType && DST_MACAB != eType && DST_OUTLOOK != eType && DST_OUTLOOKEXP != eType;
 				}
 				break;
 			case SID_DB_APP_CONVERTTOVIEW:
Index: dbaccess/source/ui/dlg/AdvancedPageDlg.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/AdvancedPageDlg.cxx,v
retrieving revision 1.12
diff -u -r1.12 AdvancedPageDlg.cxx
--- dbaccess/source/ui/dlg/AdvancedPageDlg.cxx	10 May 2007 10:20:50 -0000	1.12
+++ dbaccess/source/ui/dlg/AdvancedPageDlg.cxx	7 Jul 2007 17:05:17 -0000
@@ -145,6 +145,7 @@
 			case DST_THUNDERBIRD:
 			case DST_EVOLUTION: 
 			case DST_KAB: 
+			case DST_MACAB:
 			case DST_OUTLOOK: 
 			case DST_OUTLOOKEXP:
 				break;
Index: dbaccess/source/ui/dlg/ConnectionPage.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/ConnectionPage.cxx,v
retrieving revision 1.18
diff -u -r1.18 ConnectionPage.cxx
--- dbaccess/source/ui/dlg/ConnectionPage.cxx	10 May 2007 10:21:36 -0000	1.18
+++ dbaccess/source/ui/dlg/ConnectionPage.cxx	7 Jul 2007 17:05:17 -0000
@@ -303,6 +303,7 @@
 			case DST_OUTLOOKEXP:
 			case DST_EVOLUTION:
 			case DST_KAB:
+			case DST_MACAB:
                 m_aFT_Connection.SetText(String(ModuleRes(STR_NO_ADDITIONAL_SETTINGS)));
 				{
 					String sText = m_aFT_Connection.GetText();
Index: dbaccess/source/ui/dlg/DriverSettings.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/DriverSettings.cxx,v
retrieving revision 1.7
diff -u -r1.7 DriverSettings.cxx
--- dbaccess/source/ui/dlg/DriverSettings.cxx	13 Dec 2006 16:48:37 -0000	1.7
+++ dbaccess/source/ui/dlg/DriverSettings.cxx	7 Jul 2007 17:05:18 -0000
@@ -160,6 +160,9 @@
 		case DST_KAB: 
 			_rDetailsIds.push_back(DSID_DOSLINEENDS);
 			break;
+		case DST_MACAB: 
+			_rDetailsIds.push_back(DSID_DOSLINEENDS);
+			break;
 		case DST_OUTLOOK: 
 			_rDetailsIds.push_back(DSID_DOSLINEENDS);
 			break;
Index: dbaccess/source/ui/dlg/dbwiz.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/dbwiz.cxx,v
retrieving revision 1.14
diff -u -r1.14 dbwiz.cxx
--- dbaccess/source/ui/dlg/dbwiz.cxx	10 May 2007 10:24:58 -0000	1.14
+++ dbaccess/source/ui/dlg/dbwiz.cxx	7 Jul 2007 17:05:18 -0000
@@ -183,6 +183,7 @@
 		case DST_OUTLOOKEXP:
 		case DST_EVOLUTION:
 		case DST_KAB:
+		case DST_MACAB:
             enableButtons(WZB_NEXT,sal_False);
 			enableButtons(WZB_FINISH,sal_True);
 			break;
@@ -207,6 +208,7 @@
 				case DST_OUTLOOKEXP:
 				case DST_EVOLUTION:
 				case DST_KAB:
+				case DST_MACAB:
 					nNextState = WZS_INVALID_STATE;
 					break;
                 default:
@@ -223,6 +225,7 @@
 				case DST_OUTLOOKEXP:
 				case DST_EVOLUTION:
 				case DST_KAB:
+				case DST_MACAB:
 				case DST_MSACCESS:
 				case DST_JDBC:
 				case DST_CALC:
Index: dbaccess/source/ui/dlg/dbwizsetup.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/dbwizsetup.cxx,v
retrieving revision 1.25
diff -u -r1.25 dbwizsetup.cxx
--- dbaccess/source/ui/dlg/dbwizsetup.cxx	10 May 2007 10:25:10 -0000	1.25
+++ dbaccess/source/ui/dlg/dbwizsetup.cxx	7 Jul 2007 17:05:19 -0000
@@ -264,10 +264,11 @@
 #define EVOLUTION_PATH_GROUPWISE 17
 #define EVOLUTION_PATH_LDAP      18
 #define KAB_PATH                 19
-#define THUNDERBIRD_PATH         20
-#define CREATENEW_PATH           21
-#define USERDEFINED_PATH         22
-#define OPEN_DOC_PATH            23
+#define MACAB_PATH               20
+#define THUNDERBIRD_PATH         21
+#define CREATENEW_PATH           22
+#define USERDEFINED_PATH         23
+#define OPEN_DOC_PATH            24
 
 OFinalDBPageSetup*          pFinalPage;
 
@@ -435,6 +436,11 @@
     else
         declarePath( KAB_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
 
+    if ( m_pCollection->hasAuthentication(DST_MACAB))
+        declarePath( MACAB_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+    else
+        declarePath( MACAB_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+
     if ( m_pCollection->hasAuthentication(m_pCollection->getEmbeddedDatabaseType(getORB())))
         declarePath( CREATENEW_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     else
@@ -579,6 +585,7 @@
             { DST_EVOLUTION_GROUPWISE, EVOLUTION_PATH_GROUPWISE },
             { DST_EVOLUTION_LDAP, EVOLUTION_PATH_LDAP },
             { DST_KAB,          KAB_PATH            },
+            { DST_MACAB,        MACAB_PATH          },
             { DST_USERDEFINE1,  USERDEFINED_PATH    },
 		    { DST_USERDEFINE2,  USERDEFINED_PATH    },
 		    { DST_USERDEFINE3,  USERDEFINED_PATH    },
@@ -649,6 +656,7 @@
 	switch ( m_eType )
     {
 		case DST_KAB:
+		case DST_MACAB:
         case DST_EVOLUTION:
         case DST_EVOLUTION_GROUPWISE:
         case DST_EVOLUTION_LDAP:
Index: dbaccess/source/ui/inc/dsntypes.hxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/inc/dsntypes.hxx,v
retrieving revision 1.22
diff -u -r1.22 dsntypes.hxx
--- dbaccess/source/ui/inc/dsntypes.hxx	10 May 2007 10:32:28 -0000	1.22
+++ dbaccess/source/ui/inc/dsntypes.hxx	7 Jul 2007 17:05:19 -0000
@@ -79,7 +79,8 @@
 	DST_EVOLUTION_GROUPWISE = 18,
 	DST_EVOLUTION_LDAP = 19,
  	DST_KAB         =	20,
- 	DST_EMBEDDED	=	21,
+ 	DST_MACAB       =	21,
+ 	DST_EMBEDDED	=	22,
 
 	DST_USERDEFINE1,	/// first user defined driver
 	DST_USERDEFINE2,
Index: dbaccess/source/ui/misc/dbumiscres.hrc
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/misc/dbumiscres.hrc,v
retrieving revision 1.9
diff -u -r1.9 dbumiscres.hrc
--- dbaccess/source/ui/misc/dbumiscres.hrc	28 Feb 2006 10:36:49 -0000	1.9
+++ dbaccess/source/ui/misc/dbumiscres.hrc	7 Jul 2007 17:05:19 -0000
@@ -58,8 +58,9 @@
 #define STR_EVOLUTION_GROUPWISE	17 
 #define STR_EVOLUTION_LDAP		18
 #define STR_KAB					19
-#define STR_EMBEDDED			20
-#define STR_THUNDERBIRD			21
+#define STR_MACAB				20
+#define STR_EMBEDDED			21
+#define STR_THUNDERBIRD			22
 
 #define STR_END					STR_THUNDERBIRD + 1
 
Index: dbaccess/source/ui/misc/dbumiscres.src
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/misc/dbumiscres.src,v
retrieving revision 1.74
diff -u -r1.74 dbumiscres.src
--- dbaccess/source/ui/misc/dbumiscres.src	19 Apr 2007 15:05:31 -0000	1.74
+++ dbaccess/source/ui/misc/dbumiscres.src	7 Jul 2007 17:05:19 -0000
@@ -123,6 +123,10 @@
 	{
 		Text  ="sdbc:address:kab";
 	};
+	String STR_MACAB
+	{
+		Text  ="sdbc:address:macab";
+	};
 	String STR_EMBEDDED
 	{
 		Text = "sdbc:embedded:";
@@ -208,6 +212,10 @@
 	{
 		Text[ en-US ] = "KDE Address Book";
 	};
+	String STR_MACAB
+	{
+		Text[ en-US ] = "Mac OS X Address Book";
+	};
 	String STR_EMBEDDED
 	{
 		Text[ en-US ] = "Embedded database";
Index: extensions/source/abpilot/abpresid.hrc
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/abpresid.hrc,v
retrieving revision 1.12
diff -u -r1.12 abpresid.hrc
--- extensions/source/abpilot/abpresid.hrc	1 Dec 2006 17:34:26 -0000	1.12
+++ extensions/source/abpilot/abpresid.hrc	7 Jul 2007 17:05:20 -0000
@@ -170,12 +170,13 @@
 #define RB_THUNDERBIRD          2
 #define RB_EVOLUTION			3
 #define RB_KAB					4
-#define RB_LDAP					5
-#define RB_OUTLOOK				6
-#define RB_OUTLOOKEXPRESS		7
-#define RB_OTHER				8
-#define RB_EVOLUTION_GROUPWISE  9
-#define RB_EVOLUTION_LDAP		10
+#define RB_MACAB				5
+#define RB_LDAP					6
+#define RB_OUTLOOK				7
+#define RB_OUTLOOKEXPRESS		8
+#define RB_OTHER				9
+#define RB_EVOLUTION_GROUPWISE  10
+#define RB_EVOLUTION_LDAP		11
 
 //........................................................................
 #define ET_DATASOURCENAME		1
Index: extensions/source/abpilot/abspilot.cxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/abspilot.cxx,v
retrieving revision 1.15
diff -u -r1.15 abspilot.cxx
--- extensions/source/abpilot/abspilot.cxx	16 Sep 2006 12:51:57 -0000	1.15
+++ extensions/source/abpilot/abspilot.cxx	7 Jul 2007 17:05:20 -0000
@@ -425,6 +425,10 @@
 				m_aNewDataSource = aContext.createNewKab( m_aSettings.sDataSourceName );
 				break;
 
+			case AST_MACAB:
+				m_aNewDataSource = aContext.createNewMacab( m_aSettings.sDataSourceName );
+				break;
+
 			case AST_LDAP:
 				m_aNewDataSource = aContext.createNewLDAP( m_aSettings.sDataSourceName );
 				break;
Index: extensions/source/abpilot/abspilot.src
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/abspilot.src,v
retrieving revision 1.57
diff -u -r1.57 abspilot.src
--- extensions/source/abpilot/abspilot.src	19 Apr 2007 11:41:38 -0000	1.57
+++ extensions/source/abpilot/abspilot.src	7 Jul 2007 17:05:20 -0000
@@ -154,6 +154,13 @@
 
 		Text [ en-US ] = "KDE address book";
 	};
+	RadioButton RB_MACAB
+	{
+		Pos		= MAP_APPFONT ( 15, 101 ) ;
+		Size	= MAP_APPFONT ( WINDOW_SIZE_X - 30, 8 ) ;
+
+		Text [ en-US ] = "Mac OS X address book";
+	};
 	RadioButton RB_LDAP
 	{
 		Pos		= MAP_APPFONT ( 15, 112 ) ;
Index: extensions/source/abpilot/addresssettings.hxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/addresssettings.hxx,v
retrieving revision 1.8
diff -u -r1.8 addresssettings.hxx
--- extensions/source/abpilot/addresssettings.hxx	28 Feb 2006 10:38:29 -0000	1.8
+++ extensions/source/abpilot/addresssettings.hxx	7 Jul 2007 17:05:20 -0000
@@ -59,6 +59,7 @@
 		AST_EVOLUTION_GROUPWISE,
 		AST_EVOLUTION_LDAP,
 		AST_KAB,
+		AST_MACAB,
 		AST_LDAP,
 		AST_OUTLOOK,
 		AST_OE,
Index: extensions/source/abpilot/datasourcehandling.cxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/datasourcehandling.cxx,v
retrieving revision 1.11
diff -u -r1.11 datasourcehandling.cxx
--- extensions/source/abpilot/datasourcehandling.cxx	16 Sep 2006 12:52:39 -0000	1.11
+++ extensions/source/abpilot/datasourcehandling.cxx	7 Jul 2007 17:05:21 -0000
@@ -355,6 +355,12 @@
 	}
 
 	//---------------------------------------------------------------------
+	ODataSource	ODataSourceContext::createNewMacab( const ::rtl::OUString& _rName) SAL_THROW (( ))
+	{
+		return lcl_implCreateAndSetURL( m_pImpl->xORB, _rName, "sdbc:address:macab" );
+	}
+
+	//---------------------------------------------------------------------
 	ODataSource	ODataSourceContext::createNewOutlook( const ::rtl::OUString& _rName) SAL_THROW (( ))
 	{
 		return lcl_implCreateAndSetURL( m_pImpl->xORB, _rName, "sdbc:address:outlook" );
Index: extensions/source/abpilot/datasourcehandling.hxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/datasourcehandling.hxx,v
retrieving revision 1.8
diff -u -r1.8 datasourcehandling.hxx
--- extensions/source/abpilot/datasourcehandling.hxx	28 Feb 2006 10:38:57 -0000	1.8
+++ extensions/source/abpilot/datasourcehandling.hxx	7 Jul 2007 17:05:21 -0000
@@ -101,6 +101,9 @@
 		/// creates a new KDE address book data source
 		ODataSource createNewKab( const ::rtl::OUString& _rName ) SAL_THROW (( ));
 
+		/// creates a new Mac OS X address book data source
+		ODataSource createNewMacab( const ::rtl::OUString& _rName ) SAL_THROW (( ));
+
 		/// creates a new LDAP data source
 		ODataSource	createNewLDAP( const ::rtl::OUString& _rName ) SAL_THROW (( ));
 
Index: extensions/source/abpilot/typeselectionpage.cxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/typeselectionpage.cxx,v
retrieving revision 1.12
diff -u -r1.12 typeselectionpage.cxx
--- extensions/source/abpilot/typeselectionpage.cxx	26 Apr 2007 08:04:57 -0000	1.12
+++ extensions/source/abpilot/typeselectionpage.cxx	7 Jul 2007 17:05:21 -0000
@@ -74,6 +74,7 @@
 		,m_aEvolutionGroupwise (this,	ResId(RB_EVOLUTION_GROUPWISE))
 		,m_aEvolutionLdap	   (this,	ResId(RB_EVOLUTION_LDAP))
 		,m_aKab				(this,	ResId(RB_KAB))
+		,m_aMacab			(this,	ResId(RB_MACAB))
 		,m_aLDAP			(this,	ResId(RB_LDAP))
 		,m_aOutlook			(this,	ResId(RB_OUTLOOK))
 		,m_aOE				(this,	ResId(RB_OUTLOOKEXPRESS))
@@ -89,6 +90,7 @@
 
 		bool bWithMozilla = true, bUnx = true;
 		bool bHaveEvolution = true, bHaveKab = true;
+		bool bHaveMacab = true;
 
 #ifndef WITH_MOZILLA
 		bWithMozilla = false;
@@ -97,6 +99,7 @@
 		bUnx = false;
 		bHaveEvolution = false;
 		bHaveKab = false;
+		bHaveMacab = false;
 #else
 		Reference< XDriverAccess> xManager(_pParent->getORB()->createInstance(::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("com.sun.star.sdbc.DriverManager"))), UNO_QUERY);
 
@@ -109,6 +112,11 @@
 		xDriver = xManager->getDriverByURL(::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("sdbc:address:kab")));
 		if ( !xDriver.is() )
 			bHaveKab = false;
+
+		// check whether Mac OS X address book is available
+		xDriver = xManager->getDriverByURL(::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("sdbc:address:macab")));
+		if ( !xDriver.is() )
+			bHaveMacab = false;
 #endif
 
 		// Items are displayed in list order
@@ -118,6 +126,7 @@
 		m_aAllTypes.push_back( ButtonItem( &m_aMORK, AST_MORK, bWithMozilla ) );
 		m_aAllTypes.push_back( ButtonItem( &m_aThunderbird, AST_THUNDERBIRD, bWithMozilla ) );
 		m_aAllTypes.push_back( ButtonItem( &m_aKab, AST_KAB, bHaveKab ) );
+		m_aAllTypes.push_back( ButtonItem( &m_aMacab, AST_MACAB, bHaveMacab ) );
 		m_aAllTypes.push_back( ButtonItem( &m_aLDAP, AST_LDAP, bWithMozilla ) );
 		m_aAllTypes.push_back( ButtonItem( &m_aOutlook, AST_OUTLOOK, bWithMozilla && !bUnx ) );
 		m_aAllTypes.push_back( ButtonItem( &m_aOE, AST_OE, bWithMozilla && !bUnx ) );
Index: extensions/source/abpilot/typeselectionpage.hxx
===================================================================
RCS file: /cvs/util/extensions/source/abpilot/typeselectionpage.hxx,v
retrieving revision 1.9
diff -u -r1.9 typeselectionpage.hxx
--- extensions/source/abpilot/typeselectionpage.hxx	28 Feb 2006 10:39:34 -0000	1.9
+++ extensions/source/abpilot/typeselectionpage.hxx	7 Jul 2007 17:05:21 -0000
@@ -65,6 +65,7 @@
 		RadioButton		m_aEvolutionLdap;
 		RadioButton		m_aEvolution;
 		RadioButton		m_aKab;
+		RadioButton		m_aMacab;
 		RadioButton		m_aLDAP;
 		RadioButton		m_aOutlook;
 		RadioButton		m_aOE;
Index: connectivity/version.mk
===================================================================
RCS file: /cvs/dba/connectivity/version.mk,v
retrieving revision 1.14
diff -u -r1.14 version.mk
--- connectivity/version.mk	15 Sep 2006 13:53:10 -0000	1.14
+++ connectivity/version.mk	7 Jul 2007 17:05:21 -0000
@@ -189,6 +189,18 @@
 KAB_MICRO=0
 
 
+# ----------------------------Macab settings-------------------------------------#
+# target
+MACAB_TARGET=macab
+
+# the major 
+MACAB_MAJOR=1
+# the minor 
+MACAB_MINOR=0
+# the micro 
+MACAB_MICRO=0
+
+
 # ----------------------------HSQLDB settings-----------------------------------#
 # target
 HSQLDB_TARGET=hsqldb
Index: connectivity/prj/build.lst
===================================================================
RCS file: /cvs/dba/connectivity/prj/build.lst,v
retrieving revision 1.33
diff -u -r1.33 build.lst
--- connectivity/prj/build.lst	13 Dec 2006 16:12:54 -0000	1.33
+++ connectivity/prj/build.lst	7 Jul 2007 17:05:22 -0000
@@ -12,6 +12,7 @@
 cn  connectivity\source\drivers\mozab\bootstrap     nmake   -   all cn_mozab_bootstrap cn_mozab_mozillasrc cn_inc NULL
 cn  connectivity\source\drivers\mozab               nmake   -   all cn_mozab cn_mozab_bootstrap cn_dbtools cn_inc NULL
 cn  connectivity\source\drivers\kab                 nmake   -   u   cn_kab cn_dbtools cn_inc NULL
+cn  connectivity\source\drivers\macab               nmake   -   u   cn_macab cn_dbtools cn_inc NULL
 cn  connectivity\source\drivers\evoab               nmake   -   u   cn_evoab cn_dbtools cn_file cn_inc NULL
 cn  connectivity\source\drivers\evoab2              nmake   -   u   cn_evoab2 cn_dbtools cn_file cn_inc NULL
 cn  connectivity\source\drivers\calc                nmake   -   all cn_calc cn_file cn_inc NULL
Index: connectivity/prj/d.lst
===================================================================
RCS file: /cvs/dba/connectivity/prj/d.lst,v
retrieving revision 1.28
diff -u -r1.28 d.lst
--- connectivity/prj/d.lst	11 Apr 2007 12:35:16 -0000	1.28
+++ connectivity/prj/d.lst	7 Jul 2007 17:05:22 -0000
@@ -8,6 +8,7 @@
 ..\source\drivers\adabas\*.xml %_DEST%\xml%_EXT%\*.xml
 ..\source\drivers\ado\*.xml %_DEST%\xml%_EXT%\*.xml
 ..\source\drivers\kab\*.xml %_DEST%\xml%_EXT%\*.xml
+..\source\drivers\macab\*.xml %_DEST%\xml%_EXT%\*.xml
 ..\source\drivers\file\*.xml %_DEST%\xml%_EXT%\*.xml
 ..\source\drivers\flat\*.xml %_DEST%\xml%_EXT%\*.xml
 ..\source\drivers\dbase\*.xml %_DEST%\xml%_EXT%\*.xml
