Index: makefile.mk
===================================================================
RCS file: /cvs/external/moz/makefile.mk,v
retrieving revision 1.14
diff -u -r1.14 makefile.mk
--- makefile.mk	2008-10-14 09:20:57.000000000 -0700
+++ makefile.mk	2013-07-06 12:33:34.000000000 -0700
@@ -121,6 +121,18 @@
 MOZILLA_CONFIGURE_FLAGS+= --enable-default-toolkit=$(DEFAULT_MOZILLA_TOOLKIT)
 .ENDIF
 
+.IF "$(OS)"=="MACOSX"
+.IF "$(BUILD_OS_MINOR)"!="4"
+.IF "$(BUILD_OS_MINOR)"=="5" || "$(BUILD_OS_MINOR)"=="6"
+#10.5+ X11 includes and libraries cannot be located by autoconf
+MOZILLA_CONFIGURE_FLAGS+= --x-libraries=/usr/X11/lib --x-includes=/usr/include/X11
+.ELSE
+MOZILLA_CONFIGURE_FLAGS+= --x-libraries=/opt/X11/lib --x-includes=/opt/X11/include
+MOZILLA_CONFIGURE_FLAGS+= --with-macos-sdk=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.7.sdk
+.ENDIF
+.ENDIF
+.ENDIF
+
 CONFIGURE_ACTION=sh -c "./configure $(MOZILLA_CONFIGURE_FLAGS)"
 
 BUILD_DIR=
--- mozilla-source-1.7.5.patch	2013-07-06 12:43:13.000000000 -0700
+++ mozilla-source-1.7.5.patch	2013-09-10 01:08:40.000000000 -0700
@@ -9645,3 +9645,98 @@
   // "This code is for IA64 only"
   
 --- 25,30 ----
+*** misc/build/mozilla/xpcom/ds/nsInterfaceHashtable.h	2003-12-29 05:22:49.000000000 -0800
+--- misc/build/mozilla/xpcom/ds/nsInterfaceHashtable.h	2013-09-10 01:02:45.000000000 -0700
+@@ -111,7 +111,7 @@
+   (KeyType aKey, UserDataType* pInterface) const
+ {
+   typename nsBaseHashtable<KeyClass, nsCOMPtr<Interface>, Interface*>::EntryType* ent =
+-    GetEntry(aKey);
++    nsInterfaceHashtable<KeyClass,Interface>::GetEntry(aKey);
+ 
+   if (ent)
+   {
+@@ -139,7 +139,7 @@
+   (KeyType aKey, PRBool* aFound) const
+ {
+   typename nsBaseHashtable<KeyClass, nsCOMPtr<Interface>, Interface*>::EntryType* ent =
+-    GetEntry(aKey);
++    nsInterfaceHashtable<KeyClass,Interface>::GetEntry(aKey);
+ 
+   if (ent)
+   {
+*** misc/build/mozilla/xpcom/ds/nsBaseHashtable.h	2003-08-09 12:24:24.000000000 -0700
+--- misc/build/mozilla/xpcom/ds/nsBaseHashtable.h	2013-09-10 01:02:53.000000000 -0700
+@@ -124,7 +124,7 @@
+    */
+   PRBool Get(KeyType aKey, UserDataType* pData) const
+   {
+-    EntryType* ent = GetEntry(aKey);
++    EntryType* ent = nsTHashtable< nsBaseHashtableET<KeyClass,DataType> >::GetEntry(aKey);
+ 
+     if (!ent)
+       return PR_FALSE;
+@@ -143,7 +143,7 @@
+    */
+   PRBool Put(KeyType aKey, UserDataType aData)
+   {
+-    EntryType* ent = PutEntry(aKey);
++    EntryType* ent = nsTHashtable< nsBaseHashtableET<KeyClass,DataType> >::PutEntry(aKey);
+ 
+     if (!ent)
+       return PR_FALSE;
+@@ -157,7 +157,7 @@
+    * remove the data for the associated key
+    * @param aKey the key to remove from the hashtable
+    */
+-  void Remove(KeyType aKey) { RemoveEntry(aKey); }
++  void Remove(KeyType aKey) { nsTHashtable< nsBaseHashtableET<KeyClass,DataType> >::RemoveEntry(aKey); }
+ 
+   /**
+    * function type provided by the application for enumeration.
+*** misc/build/mozilla/xpcom/ds/nsClassHashtable.h	2003-06-19 11:42:23.000000000 -0700
+--- misc/build/mozilla/xpcom/ds/nsClassHashtable.h	2013-09-10 01:02:49.000000000 -0700
+@@ -98,7 +98,7 @@
+ nsClassHashtable<KeyClass,T>::Get(KeyType aKey, T** retVal) const
+ {
+   typename nsBaseHashtable<KeyClass,nsAutoPtr<T>,T*>::EntryType* ent =
+-    GetEntry(aKey);
++    nsClassHashtable<KeyClass,T>::GetEntry(aKey);
+ 
+   if (ent)
+   {
+*** misc/build/mozilla/configure	2013-07-08 20:42:54.000000000 -0700
+--- misc/build/mozilla/configure	2013-07-08 20:44:18.000000000 -0700
+@@ -3080,8 +3080,8 @@
+ 
+ if test "$MACOS_SDK_DIR"; then
+   export NEXT_ROOT=$MACOS_SDK_DIR
+-  CFLAGS="-I${MACOS_SDK_DIR}/usr/include $CFLAGS"
+-  CXXFLAGS="-I${MACOS_SDK_DIR}/usr/include $CXXFLAGS"
++  CFLAGS="-isysroot ${MACOS_SDK_DIR} $CFLAGS"
++  CXXFLAGS="-isysroot ${MACOS_SDK_DIR} $CXXFLAGS"
+   
+   GCC_VERSION=`echo $CXX_VERSION | perl -pe "s/gcc version ([^ ]*).*/\1/"`
+   
+@@ -11041,7 +11041,7 @@
+     ;;
+ mac|cocoa)
+     TK_LIBS='-framework Carbon'
+-    TK_CFLAGS="-I${MACOS_SDK_DIR}/Developer/Headers/FlatCarbon"
++    TK_CFLAGS="-I/Developer/Headers/FlatCarbon"
+     CFLAGS="$CFLAGS $TK_CFLAGS"
+     CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
+     MOZ_USER_DIR="Mozilla"
+*** misc/build/mozilla/nsprpub/configure	2013-07-08 20:54:50.000000000 -0700
+--- misc/build/mozilla/nsprpub/configure	2013-07-08 20:57:01.000000000 -0700
+@@ -3346,8 +3346,8 @@
+ 
+     # do the right thing for panther SDK support
+     if test "$NEXT_ROOT"; then
+-        CFLAGS="-I${NEXT_ROOT}/usr/include $CFLAGS"
+-        CXXFLAGS="-I${NEXT_ROOT}/usr/include $CXXFLAGS"
++        CFLAGS="-isysroot ${NEXT_ROOT} $CFLAGS"
++        CXXFLAGS="-isysroot ${NEXT_ROOT} $CXXFLAGS"
+     fi
+     ;;
+ 
