--- source/core/makefile.mk	2014-02-25 00:27:12.000000000 -0800
+++ source/core/makefile.mk	2016-06-22 14:32:19.000000000 -0700
@@ -36,6 +36,11 @@
 
 .IF "$(SYSTEM_OPENSSL)" == "YES"
 CFLAGS+= $(OPENSSL_CFLAGS)
+.IF "$(OS)"=="MACOSX"
+# On OS X 10.10, the openssl include files are not in the XCode SDKs. Instead,
+# they are installed in /usr/include by the Xcode Command Line Tools.
+CDEFS+=-I/usr/include
+.ENDIF
 .ENDIF
 
 # --- Files --------------------------------------------------------
