Index: source/makefile.mk
===================================================================
RCS file: /cvs/gsl/padmin/source/makefile.mk,v
retrieving revision 1.18
diff -u -r1.18 makefile.mk
--- source/makefile.mk	8 Sep 2005 16:25:28 -0000	1.18
+++ source/makefile.mk	30 Jun 2006 19:13:01 -0000
@@ -103,7 +103,11 @@
 	$(CPPUHELPERLIB)	\
 	$(CPPULIB)			\
 	$(SALLIB)			\
+
+.IF "$(OS)"!="MACOSX"
+APP1STDLIBS+=	\
 	-lXext -lX11
+.ENDIF
 
 .ENDIF
 
