Index: examples/cpp/counter/countermain.cxx
===================================================================
RCS file: /cvs/udk/product/examples/cpp/counter/countermain.cxx,v
retrieving revision 1.6.14.1
diff -u -r1.6.14.1 countermain.cxx
--- examples/cpp/counter/countermain.cxx	30 Apr 2003 06:43:25 -0000	1.6.14.1
+++ examples/cpp/counter/countermain.cxx	16 Mar 2004 19:19:25 -0000
@@ -55,7 +55,11 @@
 		xImplReg->registerImplementation(
 			OUString::createFromAscii("com.sun.star.loader.SharedLibrary"), // loader for component
 #ifdef UNX
+#ifdef MACOSX
+			OUString::createFromAscii("libcounter.dylib"),		// component location
+#else
 			OUString::createFromAscii("libcounter.so"),		// component location
+#endif
 #else
 			OUString::createFromAscii("counter.dll"),		// component location
 #endif
Index: examples/cpp/remoteclient/remoteclient.cxx
===================================================================
RCS file: /cvs/udk/product/examples/cpp/remoteclient/remoteclient.cxx,v
retrieving revision 1.4.20.1
diff -u -r1.4.20.1 remoteclient.cxx
--- examples/cpp/remoteclient/remoteclient.cxx	30 Apr 2003 06:43:27 -0000	1.4.20.1
+++ examples/cpp/remoteclient/remoteclient.cxx	16 Mar 2004 19:19:26 -0000
@@ -309,3 +309,10 @@
 	return pRet;
 }
 }
+
+#if defined(MACOSX) && __GNUC__ == 2
+
+// Initialize static data members
+ClassData1 WeakImplHelper1< XMain >::s_aCD = NULL;
+
+#endif	// MACOSX && __GNUC__ == 2
Index: settings/settings.mk
===================================================================
RCS file: /cvs/udk/product/settings/settings.mk,v
retrieving revision 1.4.6.1
diff -u -r1.4.6.1 settings.mk
--- settings/settings.mk	30 Apr 2003 06:43:29 -0000	1.4.6.1
+++ settings/settings.mk	16 Mar 2004 19:19:26 -0000
@@ -275,3 +275,96 @@
 
 endif
 
+
+
+###########################################################################
+#
+# Macosx specific settings
+#
+###########################################################################
+ifeq "$(PLATFORM)" "Darwin"
+# Settings for Mac OS X using gcc compiler
+
+# Default is Mac OS X on a powerpc machine    
+PLATFORM=macosx
+PACKAGE_LIB_DIR=macosx_ppc.plt
+JAVA_PROC_TYPE=ppc
+    
+OS=MACOSX
+PS=/
+CC=gcc
+LINK=gcc
+LIB=gcc
+ECHO=@echo
+MKDIR=mkdir -p
+CAT=cat
+OBJ_EXT=o
+SHAREDLIB_EXT=dylib
+SHAREDLIB_PRE=lib
+SHAREDLIB_OUT=$(OUT_LIB)
+    
+GCC_VERSION=$(shell $(CC) -dumpversion)
+
+ifeq "$(shell echo $(GCC_VERSION) | cut -c 1)" "3"
+COMID=gcc3
+CPPU_ENV=gcc3
+else
+COMID=$(UDK_MAJOR)GCC
+CPPU_ENV=gcc2
+endif
+
+OSEP=\<
+CSEP=\>
+QUOTE=$(subst S,\,S)
+QM=\"
+    
+DEL=rm -f
+DELRECURSIVE=rm -rf
+COPY=cp
+URLPREFIX=file://
+
+# Include UDK version numbers
+include $(PRJ)/include/udkversion.mk
+
+SALLIB=-lsal
+CPPULIB=-lcppu
+CPPUHELPERLIB=-lcppuhelper$(COMID)
+SALHELPERLIB=-lsalhelper$(COMID)
+STLPORTLIB=-lstlport_gcc
+
+EMPTYSTRING=
+PATH_SEPARATOR=:
+
+# -O is necessary for inlining (see gcc documentation)
+ifeq "$(DEBUG)" "yes"
+CC_FLAGS=-c -g
+else
+CC_FLAGS=-c -O
+endif
+
+CC_FLAGS+=-fPIC
+
+SDK_JAVA_INCLUDES = -I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers -I/System/Library/Frameworks/JavaVM.framework/Headers
+CC_INCLUDES=-I. -I/usr/include -I$(OUT)/inc/examples -I$(PRJ)/include
+STL_INCLUDES=-I$(OO_STLPORT_HOME)/stlport
+CC_DEFINES=-DUNX -DGCC -DMACOSX -DCPPU_ENV=$(CPPU_ENV)
+
+# define for used compiler necessary for UNO
+#-DCPPU_ENV=gcc2 -- gcc 2.91/2.95
+#-DCPPU_ENV=gcc3 -- gcc3 3.0
+
+CC_OUTPUT_SWITCH=-o
+
+LIBRARY_LINK_FLAGS=-dynamiclib -Wl,-single_module -fPIC -fno-common
+
+EXE_LINK_FLAGS=
+LINK_LIBS=-L$(OUT)/lib -L$(PRJ)/$(PLATFORM)/lib -L$(OFFICE_PROGRAM_PATH)
+
+ifeq "$(CPPU_ENV)" "gcc2"
+LINK_LIBS+=-lstaticmxp
+endif
+
+LINK_JAVA_LIBS=-framework JavaVM
+
+endif
+
