--- java/com/sun/star/report/SDBCReportDataFactory.java	2009-01-23 01:37:27.000000000 -0800
+++ java/com/sun/star/report/SDBCReportDataFactory.java.new	2010-02-10 22:39:20.000000000 -0800
@@ -397,7 +397,7 @@
         try
         {
             final Object[] param = new Object[2];
-            param[0] = commandType;
+            param[0] = new Integer(commandType);
             param[1] = command;
             Method call = tools.getClass().getMethod("getComposer", parameter);
             return (XSingleSelectQueryComposer) call.invoke(tools, param);
--- java/com/sun/star/report/pentaho/SOFormulaParser.java	2009-04-15 00:09:22.000000000 -0700
+++ java/com/sun/star/report/pentaho/SOFormulaParser.java.new	2010-02-11 08:27:56.000000000 -0800
@@ -115,7 +115,7 @@
             opCodes = mapper.getAvailableMappings(FormulaLanguage.ODFF, FormulaMapGroup.BINARY_OPERATORS);
             addOpCodes(names, opCodes, BINARY_OPERATORS);
 
-            names = GeneratedFormulaParserConstants.tokenImage.clone();
+            names = (String[])GeneratedFormulaParserConstants.tokenImage.clone();
             for (int i = 0; i < names.length; i++)
             {
                 final String token = names[i];
@@ -136,7 +136,7 @@
             for (int i = 0; i < opCodes.length; i++)
             {
                 final FormulaOpCodeMapEntry opCode = opCodes[i];
-                parserAllOpCodes.put(opCode.Token.OpCode, opCode);
+                parserAllOpCodes.put(new Integer(opCode.Token.OpCode), opCode);
                 specialOpCodes.add(opCode);
             }
         // addOpCodes(names, opCodes,SPECIAL,false);
@@ -251,7 +251,7 @@
             {
                 ret.append(formulaToken.Data);
             }
-            else if ( parserAllOpCodes.containsKey(formulaToken.OpCode) )
+            else if ( parserAllOpCodes.containsKey(new Integer(formulaToken.OpCode)) )
             {
                 final FormulaOpCodeMapEntry opCode = (FormulaOpCodeMapEntry) parserAllOpCodes.get(formulaToken.OpCode);
                 if ( opCode.Name.length() > 0 )
@@ -379,8 +379,8 @@
                 opCode = new FormulaOpCodeMapEntry(names[j], token);
             }
             parserNames.put(names[j], opCode);
-            parserAllOpCodes.put(opCode.Token.OpCode, opCode);
-            groupOpCodes[group].put(opCode.Token.OpCode, opCode);
+            parserAllOpCodes.put(new Integer(opCode.Token.OpCode), opCode);
+            groupOpCodes[group].put(new Integer(opCode.Token.OpCode), opCode);
         }
     }
 
