--- java/com/sun/star/report/pentaho/makefile.mk	2014-02-25 00:20:44.000000000 -0800
+++ java/com/sun/star/report/pentaho/makefile.mk	2015-09-29 17:21:11.000000000 -0700
@@ -58,6 +58,14 @@
 
 .IF "$(SYSTEM_JFREEREPORT)" == "YES" || "$(SYSTEM_APACHE_COMMONS)" == "YES"
 COMP=fix_system_libs
+.ELIF "$(SYSTEM_JFREEREPORT)" != "YES"
+COMP=fix_jar_names
+.ENDIF
+
+.IF "$(OS)" == "MACOSX"
+SEDINPLACEOPTION=-i ''
+.ELSE
+SEDINPLACEOPTION=-i
 .ENDIF
 
 JAVAFILES       := $(shell @$(FIND) . -name "*.java")
@@ -126,6 +134,10 @@
 	-i $<
 .ENDIF
 
+fix_jar_names: $(JARMANIFEST)
+	@$(SED) $(SEDINPLACEOPTION) '/flute/,/sac/d' $<
+	@$(SED) $(SEDINPLACEOPTION) "s#^Class-Path.*#& flute-$(FLUTE_VERSION).jar libserializer-$(LIBSERIALIZER_VERSION).jar libbase-$(LIBBASE_VERSION).jar libfonts-$(LIBFONTS_VERSION).jar libformula-$(LIBFORMULA_VERSION).jar liblayout-$(LIBLAYOUT_VERSION).jar libloader-$(LIBLOADER_VERSION).jar librepository-$(LIBREPOSITORY_VERSION).jar libxml-$(LIBXML_VERSION).jar flow-engine-$(FLOW_ENGINE_VERSION).jar sac.jar commons-logging-1.1.1.jar#" $<
+
 $(CLASSDIR)$/$(PACKAGE)$/%.properties : %.properties
 	@@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
