--- java/com/sun/star/report/SDBCReportDataFactory.java	2009-01-23 01:37:27.000000000 -0800
+++ java/com/sun/star/report/SDBCReportDataFactory.java.new	2010-02-10 22:39:20.000000000 -0800
@@ -397,7 +397,7 @@
         try
         {
             final Object[] param = new Object[2];
-            param[0] = commandType;
+            param[0] = new Integer(commandType);
             param[1] = command;
             Method call = tools.getClass().getMethod("getComposer", parameter);
             return (XSingleSelectQueryComposer) call.invoke(tools, param);
