--- reportdesign/source/ui/dlg/Formula.cxx	2009-01-13 12:09:25.000000000 -0800
+++ reportdesign/source/ui/dlg/Formula.cxx	2012-09-27 00:05:02.000000000 -0700
@@ -264,7 +264,7 @@
 {
     ::std::auto_ptr<formula::FormulaTokenArray> pArray(new FormulaTokenArray());
     pArray->Fill(_aTokenList, NULL);
-    return pArray;
+    return ::std::auto_ptr<formula::FormulaTokenArray>(pArray);
 }
 // =============================================================================
 } // rptui
