--- inc/sal/types.h	2014-02-25 00:31:58.000000000 -0800
+++ inc/sal/types.h	2015-09-27 14:31:44.000000000 -0700
@@ -17,6 +17,14 @@
  * specific language governing permissions and limitations
  * under the License.
  * 
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  *************************************************************/
 
 
@@ -383,6 +391,24 @@
 	*/
 	SAL_NO_ACQUIRE
 };
+
+#if SUPD == 413
+
+/** C++11 "override" feature.
+
+    With HAVE_CXX11_OVERRIDE, force the method to override a existing method in
+    parent, error out if the method with the correct signature does not exist.
+
+    @since LibreOffice 4.1
+*/
+#if HAVE_CXX11_OVERRIDE
+#define SAL_OVERRIDE override
+#else
+#define SAL_OVERRIDE
+#endif
+
+#endif	// SUPD == 413
+
 #endif /* __cplusplus */
 
 
