Index: inc/sal/types.h
===================================================================
RCS file: /cvs/porting/sal/inc/sal/types.h,v
retrieving revision 1.29
diff -u -r1.29 types.h
--- inc/sal/types.h	20 Jun 2006 04:15:50 -0000	1.29
+++ inc/sal/types.h	8 Jan 2007 17:10:07 -0000
@@ -350,6 +350,21 @@
     return static_cast< T1 >(n);
 }
 
+#if defined MACOSX && __GNUC__ > 3
+/**
+   When compiling with gcc 4.0.1 on Mac OS X 10.4 or higher, exceptions thrown
+   across shared object boundaries will not be caught when run on Mac OS X
+   10.3.9 unless the exception's class has a non-inlined virtual method. So
+   we use this class as base class for all OOo classes that exhibit this
+   Mac OS X 10.3.9 bug.
+*/
+class SAL_DLLPUBLIC_EXPORT MacOSXGCC4BaseClass
+{
+public:
+    virtual ~MacOSXGCC4BaseClass();
+};
+#endif	/* MACOSX && __GNUC__ > 3 */
+
 }
 
 #else /* __cplusplus */
Index: osl/all/utility.cxx
===================================================================
RCS file: /cvs/porting/sal/osl/all/utility.cxx,v
retrieving revision 1.4
diff -u -r1.4 utility.cxx
--- osl/all/utility.cxx	17 Sep 2006 08:45:16 -0000	1.4
+++ osl/all/utility.cxx	8 Jan 2007 17:10:08 -0000
@@ -42,6 +42,17 @@
 #include <stdio.h>
 
 
+#if defined MACOSX && __GNUC__ > 3
+namespace sal
+{
+
+MacOSXGCC4BaseClass::~MacOSXGCC4BaseClass()
+{
+}
+
+}
+#endif	/* MACOSX && __GNUC__ > 3 */
+
 /*
  *   mfe : maybe it would be wishful to include initialization
  *         of the global timer in dllmain or _init directly.
