--- source/core/tool/detfunc.cxx	2012-07-12 17:33:20.000000000 -0700
+++ source/core/tool/detfunc.cxx	2012-09-27 20:37:09.000000000 -0700
@@ -1432,6 +1432,13 @@
     }
 }
 
+namespace boost
+{
+void throw_exception(std::exception const & e)
+{
+}
+}
+
 void ScDetectiveFunc::GetAllSuccs(SCCOL nCol1, SCROW nRow1, SCCOL nCol2, SCROW nRow2, 
                                   vector<ScSharedTokenRef>& rRefTokens)
 {
--- source/ui/formdlg/formula.cxx	2009-03-06 04:55:10.000000000 -0800
+++ source/ui/formdlg/formula.cxx	2012-09-27 20:30:44.000000000 -0700
@@ -657,10 +657,13 @@
 {
     return m_xOpCodeMapper;
 }
+
+template class ::std::auto_ptr<formula::FormulaTokenArray>;
+
 ::std::auto_ptr<formula::FormulaTokenArray> ScFormulaDlg::convertToTokenArray(const uno::Sequence< sheet::FormulaToken >& _aTokenList)
 {
     ::std::auto_ptr<formula::FormulaTokenArray> pArray(new ScTokenArray());
     pArray->Fill( _aTokenList, pDoc->GetExternalRefManager());
-    return pArray;
+    return ::std::auto_ptr<formula::FormulaTokenArray>(pArray);
 }
 
