Index: util/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/util/makefile.mk,v
retrieving revision 1.18.42.1
diff -u -r1.18.42.1 makefile.mk
--- util/makefile.mk	15 Aug 2003 11:52:48 -0000	1.18.42.1
+++ util/makefile.mk	27 Dec 2004 21:33:55 -0000
@@ -70,6 +70,10 @@
 
 # --- Settings -----------------------------------------------------------
 
+# Indic resources are too big for the resource compiler
+RES_HINDI :=
+RES_KANNADA :=
+
 .INCLUDE :  svpre.mk
 .INCLUDE :  connectivity/version.mk
 .INCLUDE :  settings.mk
