Index: source/global/setupzip.scp
===================================================================
RCS file: /cvs/installation/scp/source/global/Attic/setupzip.scp,v
retrieving revision 1.114.10.5
diff -u -r1.114.10.5 setupzip.scp
--- source/global/setupzip.scp	28 Jan 2004 17:18:13 -0000	1.114.10.5
+++ source/global/setupzip.scp	5 Mar 2004 19:27:40 -0000
@@ -656,3 +656,16 @@
 End
 
 #endif
+
+#ifdef MACOSX
+#if BUILD_OS_MAJOR==10 && BUILD_OS_MINOR==2
+
+File gid_File_Lib_Dl
+    TXT_FILE_BODY;
+    Styles  = (PACKED,SETUPZIP);
+    Dir         = GCFG_BINARY_DIR;
+    Name    = STRING(CONCAT2(libdl.1,UNXSUFFIX));   
+End
+
+#endif
+#endif
Index: source/office/config.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/Attic/config.scp,v
retrieving revision 1.19.40.1
diff -u -r1.19.40.1 config.scp
--- source/office/config.scp	9 Jan 2004 18:36:50 -0000	1.19.40.1
+++ source/office/config.scp	5 Mar 2004 19:27:41 -0000
@@ -50,3 +50,14 @@
 
 #endif
 #endif
+
+#ifdef MACOSX
+#if BUILD_OS_MAJOR==10 && BUILD_OS_MINOR==2
+
+ScpAction SCP_COPY_LIBDL
+        Copy                    = STRING(CONCAT2(libdl.1,UNXSUFFIX));
+        Name                    = STRING(CONCAT2(libdl.1,UNXSUFFIX));
+End
+
+#endif
+#endif
