Index: source/office/files.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/Attic/files.scp,v
retrieving revision 1.369.16.21
diff -u -r1.369.16.21 files.scp
--- source/office/files.scp	12 Jul 2005 13:48:28 -0000	1.369.16.21
+++ source/office/files.scp	26 Sep 2005 01:07:19 -0000
@@ -1038,7 +1038,11 @@
         Styles          = (PACKED);
         Dir             = GID_DIR_PROGRAM;
         #ifdef UNX
+        #ifdef MACOSX
+        Name            = "libdb_java-4.2.jnilib";
+        #else
         Name            = STRING(CONCAT2(libdb_java-4.2,UNXSUFFIX));
+        #endif
         #else
         Name            = "libdb_java42.dll";
         #endif
