--- source/ooo/file_library_ooo.scp	2011-02-01 19:30:04.000000000 -0800
+++ source/ooo/file_library_ooo.scp	2011-12-20 10:44:48.000000000 -0800
@@ -123,7 +123,7 @@
   #endif
 End
 
-#ifdef WITH_LDAP
+#if defined WITH_LDAP && SUPD != 310
 File gid_File_Lib_Ldapbe2
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT);
@@ -1072,7 +1072,7 @@
 STD_UNO_LIB_FILE( gid_File_Lib_Offacc, offacc)
 
 #ifdef SOLAR_JAVA
-#ifndef MACOSX
+#if !defined(MACOSX) && !defined(WNT)
 
 File gid_File_Lib_Officebean
     TXT_FILE_BODY;
@@ -1085,7 +1085,7 @@
   #endif
 End
 
-#endif
+#endif	// !MACOSX && !WNT
 #endif
 
 #if defined(WNT) && !defined(DISABLE_ATL)
@@ -1863,3 +1863,141 @@
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_For, for)
 STD_LIB_FILE_PATCH( gid_File_Lib_Forui, forui)
 
+#if !defined(SYSTEM_MOZILLA) && !defined(WITHOUT_MOZILLA) && SUPD == 310
+
+File gid_File_Lib_Freebl3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libfreebl3,UNXSUFFIX));
+    #else
+        Name = "freebl3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Nspr4
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libnspr4,UNXSUFFIX));
+    #else
+        Name = "nspr4.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Nss3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libnss3,UNXSUFFIX));
+    #else
+        Name = "nss3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Nsckbi
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libnssckbi,UNXSUFFIX));
+    #else
+        Name = "nssckbi.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Nssdbm3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libnssdbm3,UNXSUFFIX));
+    #else
+        Name = "nssdbm3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Nssutil3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libnssutil3,UNXSUFFIX));
+    #else
+        Name = "nssutil3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Plc4
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libplc4,UNXSUFFIX));
+    #else
+        Name = "plc4.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Plds4
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libplds4,UNXSUFFIX));
+    #else
+        Name = "plds4.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Smime3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libsmime3,UNXSUFFIX));
+    #else
+        Name = "smime3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Softokn3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libsoftokn3,UNXSUFFIX));
+    #else
+        Name = "softokn3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Ssl3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libssl3,UNXSUFFIX));
+    #else
+        Name = "ssl3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+#ifndef MACOSX
+File gid_File_Lib_Sqlite3
+    TXT_FILE_BODY;
+	#ifdef UNX
+        Name = STRING(CONCAT2(libsqlite3,UNXSUFFIX));
+    #else
+        Name = "sqlite3.dll";
+    #endif
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+#endif
+
+#endif
--- source/ooo/file_ooo.scp	2011-02-01 19:30:06.000000000 -0800
+++ source/ooo/file_ooo.scp	2011-12-20 10:45:11.000000000 -0800
@@ -480,7 +480,7 @@
 
 #ifdef SOLAR_JAVA
 
-#ifndef MACOSX
+#if !defined(MACOSX) && !defined(WNT)
 
 File gid_File_Jar_Officebean
     TXT_FILE_BODY;
@@ -490,6 +490,6 @@
 End
 
-#endif
+#endif	// !MACOSX && !WNT
 
 #endif
 
@@ -1152,7 +1152,7 @@
     Name = "text.mod";
 End
 
-#if !defined(SYSTEM_MOZILLA) && !defined(WITHOUT_MOZILLA) 
+#if !defined(SYSTEM_MOZILLA) && !defined(WITHOUT_MOZILLA) && SUPD != 310
 
 File gid_File_Mozilla_Runtime
     TXT_FILE_BODY;
--- prj/build.lst	2008-07-30 06:38:31.000000000 -0700
+++ prj/build.lst	2012-09-29 16:18:12.000000000 -0700
@@ -1,4 +1,4 @@
-cp    scp2    :    setup_native transex3 PYTHON:python ICU:icu REDLAND:redland LIBXSLT:libxslt NULL
+cp    scp2    :    setup_native transex3 ICU:icu REDLAND:redland LIBXSLT:libxslt NULL
 cp    scp2                        usr1     -    all    cp_mkout NULL
 cp    scp2\macros                 nmake    -    all    cp_langmacros NULL
 cp    scp2\source\templates       nmake    -    all    cp_langtemplates NULL
@@ -16,7 +16,6 @@
 cp    scp2\source\layout          nmake    -    all    cp_layout cp_langmacros NULL
 cp    scp2\source\math            nmake    -    all    cp_math cp_langmacros cp_langtemplates NULL
 cp    scp2\source\ooo             nmake    -    all    cp_ooo cp_langmacros cp_langtemplates NULL
-cp    scp2\source\python          nmake    -    all    cp_python cp_langmacros cp_langtemplates NULL
 cp    scp2\source\quickstart      nmake    -    all    cp_quickstart cp_langmacros cp_langtemplates NULL
 cp    scp2\source\sdkoo           nmake    -    all    cp_sdkoo cp_langmacros cp_langtemplates NULL
 cp    scp2\source\testtool        nmake    -    all    cp_testtool cp_langmacros cp_langtemplates NULL
@@ -25,4 +24,4 @@
 cp    scp2\source\xsltfilter      nmake    -    all    cp_xsltfilter cp_langmacros cp_langtemplates NULL
 cp    scp2\source\winexplorerext  nmake    -    all    cp_winexplorerext cp_langmacros cp_langtemplates NULL
 cp    scp2\source\onlineupdate    nmake    -    all    cp_update cp_langmacros cp_langtemplates NULL
-cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_binfilter cp_calc cp_canvas cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_layout cp_math cp_ooo cp_python cp_quickstart cp_testtool cp_writer cp_base cp_xsltfilter cp_winexplorerext cp_sdkoo cp_update NULL
+cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_binfilter cp_calc cp_canvas cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_layout cp_math cp_ooo cp_quickstart cp_testtool cp_writer cp_base cp_xsltfilter cp_winexplorerext cp_sdkoo cp_update NULL
--- util/makefile.mk	2009-02-16 04:45:52.000000000 -0800
+++ util/makefile.mk	2012-09-29 16:19:36.000000000 -0700
@@ -81,10 +81,6 @@
              file_testtool.par             \
              module_xsltfilter.par         \
              file_xsltfilter.par           \
-             module_python.par             \
-             module_python_mailmerge.par   \
-             file_python.par               \
-             profileitem_python.par        \
              ure_into_ooo.par              \
              ure.par                       \
              module_improvement.par        \
@@ -234,10 +230,6 @@
              file_testtool.par             \
              module_xsltfilter.par         \
              file_xsltfilter.par           \
-             module_python.par             \
-             module_python_mailmerge.par   \
-             file_python.par               \
-             profileitem_python.par        \
              ure_into_ooo.par              \
              ure.par                       \
              module_improvement.par        \
