--- source/ooo/file_library_ooo.scp	2011-02-01 19:30:04.000000000 -0800
+++ source/ooo/file_library_ooo.scp	2011-12-20 10:44:48.000000000 -0800
@@ -1072,7 +1072,7 @@
 STD_UNO_LIB_FILE( gid_File_Lib_Offacc, offacc)
 
 #ifdef SOLAR_JAVA
-#ifndef MACOSX
+#if !defined(MACOSX) && !defined(WNT)
 
 File gid_File_Lib_Officebean
     TXT_FILE_BODY;
@@ -1085,7 +1085,7 @@
   #endif
 End
 
-#endif
+#endif	// !MACOSX && !WNT
 #endif
 
 #if defined(WNT) && !defined(DISABLE_ATL)
--- source/ooo/file_ooo.scp	2011-02-01 19:30:06.000000000 -0800
+++ source/ooo/file_ooo.scp	2011-12-20 10:45:11.000000000 -0800
@@ -480,7 +480,7 @@
 
 #ifdef SOLAR_JAVA
 
-#ifndef MACOSX
+#if !defined(MACOSX) && !defined(WNT)
 
 File gid_File_Jar_Officebean
     TXT_FILE_BODY;
@@ -490,6 +490,7 @@
 End
 
 #endif
+#endif	// !MACOSX && !WNT
 
 #endif
 
