Index: source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.209
diff -u -r1.209 file_library_ooo.scp
--- source/ooo/file_library_ooo.scp	4 Aug 2006 12:39:52 -0000	1.209
+++ source/ooo/file_library_ooo.scp	29 Nov 2006 16:51:13 -0000
@@ -567,9 +567,9 @@
     Styles = (PACKED);
     Dir = gid_Dir_Program;
   #ifdef UNX
-    Name = STRING(CONCAT2(libdb-4.2,UNXSUFFIX));
+    Name = STRING(CONCAT2(libdb-4.3,UNXSUFFIX));
   #else
-    Name = "libdb42.dll";
+    Name = "libdb43.dll";
   #endif
 End
 
@@ -612,9 +612,9 @@
     Dir = gid_Dir_Program;
   #ifdef UNX
     #ifdef MACOSX
-        Name = "libdb_java-4.2.jnilib";
+        Name = "libdb_java-4.3.jnilib";
     #else
-    	Name = STRING(CONCAT2(libdb_java-4.2,UNXSUFFIX));
+    	Name = STRING(CONCAT2(libdb_java-4.3,UNXSUFFIX));
     #endif
  #else
     Name = "libdb_java42.dll";
@@ -2454,6 +2454,15 @@
 End
 #endif
 
+#ifdef ENABLE_ACCESS
+#ifdef UNX
+#ifndef MACOSX
+STD_UNO_LIB_FILE( gid_File_Lib_Mdb2, mdb )
+STD_UNO_LIB_FILE( gid_File_Lib_Mdb2_Impl, mdbimpl )
+#endif
+#endif
+#endif
+
 // AVMedia
 STD_LIB_FILE_PATCH( gid_File_Lib_AVMedia, avmedia )
 
