Index: source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.184.2.2
diff -u -r1.184.2.2 file_library_ooo.scp
--- source/ooo/file_library_ooo.scp	17 Feb 2006 16:52:02 -0000	1.184.2.2
+++ source/ooo/file_library_ooo.scp	12 Mar 2006 21:51:54 -0000
@@ -1474,7 +1474,9 @@
     PACKED_UNO_LIB_FILE_BODY_PATCH;
 End
 
+#ifndef WITHOUT_MOZILLA
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Pl , pl)
+#endif
 
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Preload, preload)
 
Index: source/ooo/file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.141.2.1
diff -u -r1.141.2.1 file_ooo.scp
--- source/ooo/file_ooo.scp	10 Feb 2006 11:49:41 -0000	1.141.2.1
+++ source/ooo/file_ooo.scp	12 Mar 2006 21:51:55 -0000
@@ -163,6 +163,8 @@
 End
 #endif
 
+#ifndef WITHOUT_MOZILLA
+
 #if !defined(UNX) || defined(ENABLE_GTK)
 File gid_File_Lib_Npsoplugin
         BIN_FILE_BODY;
@@ -184,6 +186,7 @@
   #endif
 End
 #endif
+#endif
 
 File gid_File_Bin_Uno
     BIN_FILE_BODY;
@@ -291,7 +294,7 @@
     Name = "gengal.rdb";
 End
 
-#ifdef UNX
+#if !defined(WITHOUT_MOZILLA) && defined(UNX)
 
 File gid_File_Bin_Pluginapp
     BIN_FILE_BODY;
