Index: source/ui/dlg/docprev.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/dlg/docprev.cxx,v
retrieving revision 1.20
diff -u -r1.20 docprev.cxx
--- source/ui/dlg/docprev.cxx	12 Dec 2006 17:04:46 -0000	1.20
+++ source/ui/dlg/docprev.cxx	9 May 2007 01:13:35 -0000
@@ -336,7 +336,8 @@
 
 		// Use new StandardCheckVisisbilityRedirector
 		StandardCheckVisisbilityRedirector aRedirector;
-		const Rectangle aRedrawRectangle( Point(), aNewSize );
+		Point aPoint;
+		const Rectangle aRedrawRectangle( aPoint, aNewSize );
 		Region aRedrawRegion(aRedrawRectangle);
 		pView->SdrPaintView::CompleteRedraw(&aVDev,aRedrawRegion,0,&aRedirector);
 
