--- makefile.mk	2014-02-25 00:39:58.000000000 -0800
+++ makefile.mk	2016-06-22 14:38:32.000000000 -0700
@@ -94,6 +94,12 @@
 .ENDIF
 .ENDIF
 
+.IF "$(OS)"=="MACOSX" && "$(SYSTEM_OPENSSL)"=="YES"
+# On OS X 10.10, the openssl include files are not in the XCode SDKs. Instead,
+# they are installed in /usr/include by the Xcode Command Line Tools.
+CDEFS+=-I/usr/include
+.ENDIF
+
 # On Linux/Mac we need the content of CDEFS in CFLAGS so that the ssl headers are searched for
 .IF "$(OS)"=="MACOSX" || "$(OS)"=="LINUX"
 # in a directory that corresponds to the directory that is searched for the ssl library.
