--- set_soenv.in.orig	2010-02-10 11:55:52.000000000 -0800
+++ set_soenv.in	2010-02-10 14:36:30.000000000 -0800
@@ -2147,6 +2147,39 @@
   ToFile( ".pl",   	"perl",	"e");
   ToFile( "SWITCHAR",   "-",	"e");
 }
+elsif ( $platform =~ m/darwin/ ) 
+{
+   # Append old BUILD_OS_* macros needed by the NeoOffice patches
+   my $macosxCdefs="";
+
+   open OSXVER, "/usr/bin/sw_vers -productVersion |";
+   while (<OSXVER>) {
+       if (/^\s*(\d+?)\.(\d+?)(\.(\d+?))*\s*$/) {
+           ToFile("BUILD_OS_MAJOR", $1, "e");
+           $macosxCdefs.=" -DBUILD_OS_MAJOR=" . $1;
+
+           ToFile("BUILD_OS_MINOR", $2, "e");
+           $macosxCdefs.=" -DBUILD_OS_MINOR=" . $2;
+
+           if($3 ne "") {
+               ToFile("BUILD_OS_REV", $4, "e");
+               $macosxCdefs.=" -DBUILD_OS_REV=" . $4;
+           } else {
+               ToFile("BUILD_OS_REV", 0, "e");
+               $macosxCdefs.=" -DBUILD_OS_REV=0";
+           }
+           last;
+       }
+   }
+   close(OSXVER);
+
+   # Append NeoOffice build environment variables
+   ToFile( "LIBIDL_CONFIG", $ENV{"LIBIDL_CONFIG"}, "e");
+   ToFile( "PKG_CONFIG", $ENV{"PKG_CONFIG"}, "e");
+   ToFile( "PKG_CONFIG_PATH", $ENV{"PKG_CONFIG_PATH"}, "e");
+   ToFile( "TMP", $ENV{"TMP"}, "e");
+   ToFile( "TEMP", $ENV{"TEMP"}, "e");
+}
 ToFile( "SAL_ENABLE_FILE_LOCKING",	"1",	"e");
 #
 # Writing unset variables you might not need to file.
