--- ./inc/sfx2/objsh.hxx	2015-07-02 19:42:38.000000000 -0700
+++ ./inc/sfx2/objsh.hxx	2015-08-05 09:17:41.000000000 -0700
@@ -220,6 +220,7 @@
 	sal_uInt16					nViewNo;			// Numerierung der MDIWins
 	sal_Bool					bHasName :1,		// sal_True := bestehendes Objekt, sal_False := es ist ein neues Objekt
                                 bIsTmp :1;          // temp. Storage
+	sal_Bool					bIsDeleted :1;
 
 private:
 //#if 0 // _SOLAR__PRIVATE
@@ -298,6 +299,8 @@
     sal_Bool                    HasName() const { return bHasName; }
     virtual String              GetAPIName() const;
 	void                        SetHasName( sal_Bool bSet = sal_True ) { bHasName = bSet; }
+    sal_Bool                    IsDeleted() const { return bIsDeleted; }
+	void                        SetIsDeleted( sal_Bool bSet = sal_True ) { bIsDeleted = bSet; }
     sal_Bool                    IsReadOnly() const;
     sal_Bool                    IsReadOnlyMedium() const;
 	void						SetReadOnlyUI( sal_Bool bReadOnly = sal_True );
--- ./inc/sfx2/prnmon.hxx	2008-04-11 04:38:31.000000000 -0700
+++ ./inc/sfx2/prnmon.hxx	2015-08-05 09:19:21.000000000 -0700
@@ -92,6 +92,9 @@
 	SfxViewShell*           pViewSh;
 	SfxItemSet*             pOptions;
 	SfxTabPage*             pPage;
+#ifdef MACOSX
+	bool					bShowPrintSetupDialog;
+#endif	// MACOSX
 
 public:
 							SfxPrintOptionsDialog( Window *pParent,
--- ./inc/sfx2/topfrm.hxx	2008-04-11 04:45:57.000000000 -0700
+++ ./inc/sfx2/topfrm.hxx	2015-08-05 09:33:27.000000000 -0700
@@ -40,6 +40,10 @@
 #include <sfx2/frame.hxx>
 #include <sfx2/viewfrm.hxx>
 
+#ifdef MACOSX
+#include <vcl/timer.hxx>
+#endif	// MACOSX
+
 class SfxViewShell;
 class SfxTopFrame;
 class SfxTopFrame_Impl;
@@ -92,6 +96,10 @@
 
 	SfxTopViewFrame_Impl*	pImp;
     svtools::AsynchronLink* pCloser;
+#ifdef MACOSX
+    BOOL					bNeedsUpdateTitle;
+    Timer					aTimer;
+#endif	// MACOSX
 
 protected:
 
--- ./source/doc/objxtor.cxx	2015-07-02 19:42:20.000000000 -0700
+++ ./source/doc/objxtor.cxx	2015-08-05 09:23:54.000000000 -0700
@@ -386,6 +386,7 @@
 
 	bHasName = sal_False;
 	nViewNo = 0;
+	bIsDeleted = sal_False;
 
 	pImp->bWaitingForPicklist = sal_True;
 
--- ./source/view/printer.cxx	2008-04-11 06:38:58.000000000 -0700
+++ ./source/view/printer.cxx	2015-08-05 09:30:14.000000000 -0700
@@ -539,6 +539,9 @@
 	pViewSh		( pViewShell ),
 	pOptions	( pSet->Clone() ),
 	pPage		( NULL )
+#ifdef MACOSX
+	, bShowPrintSetupDialog	( true )
+#endif	// MACOSX
 
 {
 	SetText( SfxResId( STR_PRINT_OPTIONS_TITLE ) );
--- ./source/view/topfrm.cxx	2015-07-02 19:42:31.000000000 -0700
+++ ./source/view/topfrm.cxx	2015-08-05 09:38:53.000000000 -0700
@@ -1186,6 +1186,9 @@
 */
 
     : SfxViewFrame( *(new SfxBindings), pFrame, pObjShell, SFXFRAME_HASTITLE )
+#ifdef MACOSX
+    , bNeedsUpdateTitle( FALSE )
+#endif	// MACOSX
 {
 	DBG_CTOR(SfxTopViewFrame, 0);
 
