Index: source/jscpp/sjapplet_impl.cxx
===================================================================
RCS file: /cvs/oi/sj2/source/jscpp/sjapplet_impl.cxx,v
retrieving revision 1.26
diff -u -r1.26 sjapplet_impl.cxx
--- source/jscpp/sjapplet_impl.cxx	13 Oct 2005 17:05:18 -0000	1.26
+++ source/jscpp/sjapplet_impl.cxx	27 Dec 2005 22:17:49 -0000
@@ -114,7 +114,7 @@
 #endif
 
 #ifdef SOLAR_JAVA
-#ifdef UNX
+#if defined UNX && !defined MACOSX
 struct EmbeddedWindow {
 	jobject _joWindow;
 
