Index: source/jscpp/sjapplet_impl.cxx
===================================================================
RCS file: /cvs/oi/sj2/source/jscpp/sjapplet_impl.cxx,v
retrieving revision 1.26
diff -u -r1.26 sjapplet_impl.cxx
--- source/jscpp/sjapplet_impl.cxx	13 Oct 2005 17:05:18 -0000	1.26
+++ source/jscpp/sjapplet_impl.cxx	28 Dec 2005 18:11:38 -0000
@@ -114,7 +114,7 @@
 #endif
 
 #ifdef SOLAR_JAVA
-#ifdef UNX
+#if defined UNX && !defined MACOSX
 struct EmbeddedWindow {
 	jobject _joWindow;
 
@@ -215,7 +215,7 @@
 {
 jclass jcFrame = pEnv->FindClass("java/awt/Frame");                     testJavaException(pEnv);
 jmethodID jmFrame_rinit = pEnv->GetMethodID(jcFrame, "<init>", "()V");  testJavaException(pEnv);
-joFrame = pEnv->AllocObject(jcFrame);                                   testJavaException(pEnv);
+jobject joFrame = pEnv->AllocObject(jcFrame);                                   testJavaException(pEnv);
 pEnv->CallVoidMethod(joFrame, jmFrame_rinit);                           testJavaException(pEnv);
 
 _joWindow = pEnv->NewGlobalRef(joFrame);
