Index: inc/rules.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/rules.mk,v
retrieving revision 1.80.18.2
diff -u -r1.80.18.2 rules.mk
--- inc/rules.mk	8 May 2007 12:11:11 -0000	1.80.18.2
+++ inc/rules.mk	23 Dec 2007 20:11:23 -0000
@@ -328,6 +328,63 @@
 	@echo "No recipe for compiling Objective-C files is available for this platform"
 .ENDIF		"$(OS)"=="MACOSX"
 
+#
+# [ed] 12/28/07 Add support for ObjC++ files
+#
+
+# Objective-C++ files
+$(OBJ)$/%.obj : %.mm
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(OS)"=="MACOSX"
+.IF "$(TEST)"!=""
+	$(objc) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(OBJCFLAGS) -ObjC++ $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $*.mm
+.ELSE
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(objc) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(OBJCFLAGS) -ObjC++ $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $*.mm
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ENDIF
+.ELSE		"$(OS)"=="MACOSX"
+	@echo "No recipe for compiling Objective-C++ files is available for this platform"
+.ENDIF		"$(OS)"=="MACOSX"
+
+# Objective-C++ files
+$(OBJ)$/%.obj : $(MISC)$/%.mm
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(OS)"=="MACOSX"
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(objc) $(CFLAGS) $(INCLUDE) $(CFLAGSCCC) $(OBJCFLAGS) -ObjC++ $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $(MISC)$/$*.mm
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ELSE		"$(OS)"=="MACOSX"
+	@echo "No recipe for compiling Objective-C files is available for this platform"
+.ENDIF		"$(OS)"=="MACOSX"
+
+# Objective-C++ files
+$(SLO)$/%.obj : $(MISC)$/%.mm
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(OS)"=="MACOSX"
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(objc) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(OBJCFLAGS) -ObjC++ $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(SLO)$/$*.o $(MISC)$/$*.mm
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ELSE		"$(OS)"=="MACOSX"
+	@echo "No recipe for compiling Objective-C files is available for this platform"
+.ENDIF		"$(OS)"=="MACOSX"
+
+# Objective-C files
+$(SLO)$/%.obj : %.mm
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(OS)"=="MACOSX"
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(objc) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(OBJCFLAGS) -ObjC++ $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(SLO)$/$*.o $*.mm
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ELSE		"$(OS)"=="MACOSX"
+	@echo "No recipe for compiling Objective-C files is available for this platform"
+.ENDIF		"$(OS)"=="MACOSX"
+
+
 # dependencies c / c++
 
 not_existing$/s_%.dpcc : %.c;@noop $(assign all_local_slo+:=$<)
@@ -414,6 +471,56 @@
 	@echo LAZY_DEPS=were_used_to_generate >> $@
 .ENDIF			# "$(LAZY_DEPS)"==""	
 
+#
+# [ed] 12/28/07 Add ObjC++ support
+#
+
+# dependencies objective-c++
+
+$(MISC)$/s_%.dpcc : %.mm
+	@+-$(RM) $@ >& $(NULLDEV)
+	@$(MAKEDEPEND) -f - -p$(SLO) $(MKDEPFLAGS) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $< > $@
+.IF "$(LAZY_DEPS)"==""	
+	@echo $@ : $(SLO)$/$(<:b).obj >> $@
+.ELSE			# "$(LAZY_DEPS)"==""	
+	@echo LAZY_DEPS=were_used_to_generate >> $@
+.ENDIF			# "$(LAZY_DEPS)"==""	
+
+$(MISC)$/o_%.dpcc : %.mm
+	@+-$(RM) $@ >& $(NULLDEV)
+	@$(MAKEDEPEND) -f - -p$(OBJ) $(MKDEPFLAGS) $(CDEFS) $(CDEFSOBJ) $(CDEFSMT) $< > $@
+.IF "$(LAZY_DEPS)"==""	
+	@echo $@ : $(OBJ)$/$(<:b).obj >> $@
+.ELSE			# "$(LAZY_DEPS)"==""	
+	@echo LAZY_DEPS=were_used_to_generate >> $@
+.ENDIF			# "$(LAZY_DEPS)"==""	
+
+$(MISC)$/s_%.dpcc : $(MISC)$/%.mm
+	@+-$(RM) $@ >& $(NULLDEV)
+.IF "$(GUI)"=="UNX"	
+	@$(MAKEDEPEND) -f - -p$(SLO) $(MKDEPFLAGS) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $< | sed s\#$(MISC)$/\#\# > $@
+.ELSE			# "$(GUI)"=="UNX"	
+	@$(MAKEDEPEND) -f - -p$(SLO) $(MKDEPFLAGS) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $< | $(SED) s/$(MISC:s/\/\\/)\\// > $@
+.ENDIF			# "$(GUI)"=="UNX"	
+.IF "$(LAZY_DEPS)"==""	
+	@echo $@ : $(SLO)$/$(<:b).obj >> $@
+.ELSE			# "$(LAZY_DEPS)"==""	
+	@echo LAZY_DEPS=were_used_to_generate >> $@
+.ENDIF			# "$(LAZY_DEPS)"==""	
+
+$(MISC)$/o_%.dpcc : $(MISC)$/%.mm
+	@+-$(RM) $@ >& $(NULLDEV)
+.IF "$(GUI)"=="UNX"	
+	@$(MAKEDEPEND) -f - -p$(OBJ) $(MKDEPFLAGS) $(CDEFS) $(CDEFSOBJ) $(CDEFSMT) $< | sed s\#$(MISC)$/\#\# > $@
+.ELSE			# "$(GUI)"=="UNX"	
+	@$(MAKEDEPEND) -f - -p$(OBJ) $(MKDEPFLAGS) $(CDEFS) $(CDEFSOBJ) $(CDEFSMT) $< | $(SED) s/$(MISC:s/\/\\/)\\// > $@
+.ENDIF			# "$(GUI)"=="UNX"	
+.IF "$(LAZY_DEPS)"==""	
+	@echo $@ : $(OBJ)$/$(<:b).obj >> $@
+.ELSE			# "$(LAZY_DEPS)"==""	
+	@echo LAZY_DEPS=were_used_to_generate >> $@
+.ENDIF			# "$(LAZY_DEPS)"==""	
+
 # dependency dummy for *.s files
 
 $(MISC)$/s_%.dpcc : %.s
Index: inc/unxmacx.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxmacx.mk,v
retrieving revision 1.7.30.3
diff -u -r1.7.30.3 unxmacx.mk
--- inc/unxmacx.mk	1 Feb 2007 19:07:01 -0000	1.7.30.3
+++ inc/unxmacx.mk	23 Dec 2007 20:11:23 -0000
@@ -230,10 +230,11 @@
 STDSLOCUI=
 
 STDLIBGUIST=-lm
-STDLIBCUIST=-lX11 -lpthread CPPRUNTIME -lm
+STDLIBCUIST=-lm
 STDLIBGUIMT=-lpthread CPPRUNTIME -lm
-STDLIBCUIMT=-lX11 CPPRUNTIME -lm
-STDSHLGUIMT=-lX11 -lXext -lpthread CPPRUNTIME -lm -framework CoreFoundation
+STDLIBCUIMT=-lpthread CPPRUNTIME -lm
+STDSHLGUIMT=-lpthread CPPRUNTIME -lm
+STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 
 LIBMGR=ar
