Index: inc/unxmacxp.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxmacxp.mk,v
retrieving revision 1.57
diff -u -r1.57 unxmacxp.mk
--- inc/unxmacxp.mk	17 Oct 2005 14:20:53 -0000	1.57
+++ inc/unxmacxp.mk	27 Dec 2005 20:10:42 -0000
@@ -198,10 +198,10 @@
 STDSLOCUI=
 
 STDLIBGUIST=-lm
-STDLIBCUIST=-lX11 -lpthread CPPRUNTIME -lm
+STDLIBCUIST=-lm
 STDLIBGUIMT=-lpthread CPPRUNTIME -lm
-STDLIBCUIMT=-lX11 CPPRUNTIME -lm
-STDSHLGUIMT=-lX11 -lXext -lpthread CPPRUNTIME -lm
+STDLIBCUIMT=-lpthread CPPRUNTIME -lm
+STDSHLGUIMT=-lpthread CPPRUNTIME -lm
 STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 
 LIBMGR=ar
