--- inc/unxmacx.mk	2011-02-01 08:54:38.000000000 -0800
+++ inc/unxmacx.mk	2011-02-01 18:40:49.000000000 -0800
@@ -51,9 +51,12 @@
 .IF "$(GUIBASE)"=="unx" && "$(USE_SYSTEM_STL)"!="YES"
 CDEFS+=-DX_LOCALE
 .ENDIF
-.IF "$(GUIBASE)"=="aqua"
+.IF "$(GUIBASE)"=="aqua" || "$(GUIBASE)"=="java"
 # TODO: use MACOSX_DEPLOYMENT_TARGET instead of MAC_OS_X_VERSION_MIN_REQUIRED?
-CDEFS+=-DQUARTZ -DMAC_OS_X_VERSION_MIN_REQUIRED=1040
+CDEFS+=-DQUARTZ -DMAC_OS_X_VERSION_MIN_REQUIRED=1050
+.ENDIF
+.IF "$(GUIBASE)"=="java"
+CDEFS+=-DUSE_JAVA
 .ENDIF
 
 # Name of library where static data members are initialized
@@ -116,7 +116,11 @@
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # Normal C++ compilation flags
+.IF "$(BUILD_OS_MAJOR)"=="10" && "$(BUILD_OS_MINOR)"=="5"
 CFLAGSCXX=-pipe -malign-natural -fsigned-char -Wno-long-double $(ARCH_FLAGS)
+.ELSE		# "$(BUILD_OS_MAJOR)"=="10" && "$(BUILD_OS_MINOR)"=="5"
+CFLAGSCXX=-pipe -malign-natural -fsigned-char $(ARCH_FLAGS)
+.ENDIF		# "$(BUILD_OS_MAJOR)"=="10" && "$(BUILD_OS_MINOR)"=="5"
 CFLAGSCXX+= -Wno-ctor-dtor-privacy
 
 PICSWITCH:=-fPIC
@@ -259,11 +262,11 @@
 STDOBJCUI=
 STDSLOCUI=
 
-.IF "$(GUIBASE)" == "aqua"
+.IF "$(GUIBASE)" == "aqua" || "$(GUIBASE)" == "java"
 	STDLIBCUIMT=CPPRUNTIME -lm
-	STDLIBGUIMT=-framework Carbon -framework Cocoa -lpthread CPPRUNTIME -lm
+	STDLIBGUIMT=-framework AppKit -lpthread CPPRUNTIME -lm
 	STDSHLCUIMT=-lpthread CPPRUNTIME -lm
-	STDSHLGUIMT=-framework Carbon -framework CoreFoundation -framework Cocoa -lpthread CPPRUNTIME -lm
+	STDSHLGUIMT=-framework AppKit -lpthread CPPRUNTIME -lm
 	PSPLIB=-lpsp
 .ELSE
 	STDLIBCUIMT= CPPRUNTIME -lm
--- bin/deliver.pl	2011-08-08 12:06:01.445046400 -0700
+++ bin/deliver.pl	2011-08-09 13:14:56.864985600 -0700
@@ -114,7 +114,15 @@
 if ($^O ne 'cygwin') {              # iz59477 - cygwin needes a dot "." at the end of filenames to disable
   $maybedot     = '';               # some .exe transformation magic.
 } else {
-  $maybedot     = '.';
+    my $cygvernum = `uname -r`;
+    my @cygvernum = split( /\./, $cygvernum);
+    $cygvernum = shift @cygvernum;
+    $cygvernum .= shift @cygvernum;
+    if ( $cygvernum < 17 ) {
+        $maybedot     = '.';
+    } else {
+        $maybedot     = '';               # no longer works with cygwin 1.7. other magic below.
+    }
 }
 
 ($gui		= lc($ENV{GUI})) 		|| die "Can't determine 'GUI'. Please set environment.\n";
