Index: inc/unxmacx.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxmacx.mk,v
retrieving revision 1.7.30.3
diff -u -r1.7.30.3 unxmacx.mk
--- inc/unxmacx.mk	1 Feb 2007 19:07:01 -0000	1.7.30.3
+++ inc/unxmacx.mk	7 May 2007 04:11:23 -0000
@@ -230,10 +230,11 @@
 STDSLOCUI=
 
 STDLIBGUIST=-lm
-STDLIBCUIST=-lX11 -lpthread CPPRUNTIME -lm
+STDLIBCUIST=-lm
 STDLIBGUIMT=-lpthread CPPRUNTIME -lm
-STDLIBCUIMT=-lX11 CPPRUNTIME -lm
-STDSHLGUIMT=-lX11 -lXext -lpthread CPPRUNTIME -lm -framework CoreFoundation
+STDLIBCUIMT=-lpthread CPPRUNTIME -lm
+STDSHLGUIMT=-lpthread CPPRUNTIME -lm
+STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 
 LIBMGR=ar
