Index: inc/settings.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/settings.mk,v
retrieving revision 1.189
diff -u -r1.189 settings.mk
--- inc/settings.mk	4 May 2006 09:24:53 -0000	1.189
+++ inc/settings.mk	30 Jun 2006 19:18:50 -0000
@@ -180,7 +180,7 @@
 .ENDIF
 
 #required arguments
-JAVAC=$(JAVACOMPILER)
+JAVAC=$(JAVACOMPILER) -source 1.4
 JAVAI=$(JAVAINTERPRETER)
 .IF "$(JAVACACHE)" != ""
 JAVAC+=--encoding=UTF-8 -O2 -fno-assert -Wno-deprecated -C
Index: inc/unxmacx.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxmacx.mk,v
retrieving revision 1.2
diff -u -r1.2 unxmacx.mk
--- inc/unxmacx.mk	8 Mar 2006 14:13:02 -0000	1.2
+++ inc/unxmacx.mk	30 Jun 2006 19:18:50 -0000
@@ -200,10 +200,10 @@
 STDSLOCUI=
 
 STDLIBGUIST=-lm
-STDLIBCUIST=-lX11 -lpthread CPPRUNTIME -lm
+STDLIBCUIST=-lm
 STDLIBGUIMT=-lpthread CPPRUNTIME -lm
-STDLIBCUIMT=-lX11 CPPRUNTIME -lm
-STDSHLGUIMT=-lX11 -lXext -lpthread CPPRUNTIME -lm
+STDLIBCUIMT=-lpthread CPPRUNTIME -lm
+STDSHLGUIMT=-lpthread CPPRUNTIME -lm
 STDSHLCUIMT=-lpthread CPPRUNTIME -lm
 
 LIBMGR=ar
