--- cpp/_unix.c	2007-11-12 07:30:29.000000000 -0800
+++ cpp/_unix.c	2012-07-11 15:31:06.000000000 -0700
@@ -190,7 +190,7 @@
 /* memmove is defined here because some vendors don't provide it at
    all and others do a terrible job (like calling malloc) */
 
-#if !defined(__IBMC__) && !defined(_WIN32) && !defined(__GLIBC__)
+#if !defined(__IBMC__) && !defined(_WIN32) && !defined(__GLIBC__) && !defined(__GNUC__)
 
 void *
     memmove(void *dp, const void *sp, size_t n)
