--- inc/svx/svdoashp.hxx	2015-07-02 19:42:39.000000000 -0700
+++ inc/svx/svdoashp.hxx	2015-08-04 20:07:06.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 #ifndef _SVDOASHP_HXX
@@ -139,6 +147,9 @@
 protected:
 
 	String		aName;
+#if SUPD == 310
+    Size          m_aSuggestedTextFrameSize;
+#endif	// SUPD == 310
 
 public:
 
@@ -212,6 +223,9 @@
 	virtual FASTBOOL MovCreate(SdrDragStat& rStat);	// #i37448#
 	virtual FASTBOOL EndCreate(SdrDragStat& rStat, SdrCreateCmd eCmd);
 
+#if SUPD == 310
+    void SuggestTextFrameSize(Size aSuggestedTextFrameSize);
+#endif	// SUPD == 310
 	virtual FASTBOOL AdjustTextFrameWidthAndHeight(Rectangle& rR, FASTBOOL bHgt=TRUE, FASTBOOL bWdt=TRUE) const;
 	virtual FASTBOOL NbcAdjustTextFrameWidthAndHeight(FASTBOOL bHgt=TRUE, FASTBOOL bWdt=TRUE);
 	virtual FASTBOOL AdjustTextFrameWidthAndHeight(FASTBOOL bHgt=TRUE, FASTBOOL bWdt=TRUE);
--- source/svdraw/svdoashp.cxx	2015-07-02 19:42:39.000000000 -0700
+++ source/svdraw/svdoashp.cxx	2015-08-04 20:10:15.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
@@ -2520,9 +2528,25 @@
 		}
 	}
 }
+
+#if SUPD == 310
+
+void SdrObjCustomShape::SuggestTextFrameSize(Size aSuggestedTextFrameSize)
+{
+    m_aSuggestedTextFrameSize = aSuggestedTextFrameSize;
+}
+
+#endif	// SUPD == 310
+
 FASTBOOL SdrObjCustomShape::AdjustTextFrameWidthAndHeight(Rectangle& rR, FASTBOOL bHgt, FASTBOOL bWdt) const
 {
+#if SUPD == 310
+    // Either we have text or the application has native text and suggested its size to us.
+    bool bHasText = HasText() || (m_aSuggestedTextFrameSize.Width() != 0 && m_aSuggestedTextFrameSize.Height() != 0);
+    if ( pModel && bHasText && !rR.IsEmpty() )
+#else	// SUPD == 310
  	if ( pModel && HasText() && !rR.IsEmpty() )
+#endif	// SUPD == 310
 	{
 		FASTBOOL bWdtGrow=bWdt && IsAutoGrowWidth();
 		FASTBOOL bHgtGrow=bHgt && IsAutoGrowHeight();
@@ -2561,6 +2585,10 @@
 			if ( aSiz.Height() < 2 )
 				aSiz.Height() = 2; // Mindestgroesse 2
 
+#if SUPD == 310
+            if (HasText())
+            {
+#endif	// SUPD == 310
 			if(pEdtOutl)
 			{
 				pEdtOutl->SetMaxAutoPaperSize( aSiz );
@@ -2598,6 +2626,14 @@
 					nHgt = rOutliner.GetTextHeight()+1; // lieber etwas Tolleranz
 				rOutliner.Clear();
 			}
+#if SUPD == 310
+            }
+            else
+            {
+                nHgt = m_aSuggestedTextFrameSize.Height();
+                nWdt = m_aSuggestedTextFrameSize.Width();
+            }
+#endif	// SUPD == 310
 			if ( nWdt < nMinWdt )
 				nWdt = nMinWdt;
 			if ( nWdt > nMaxWdt )
--- ./inc/svx/escherex.hxx	2015-07-02 19:42:33.000000000 -0700
+++ ./inc/svx/escherex.hxx	2015-08-04 20:48:24.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 #ifndef _SVX_ESCHEREX_HXX
@@ -1405,6 +1413,9 @@
 		BOOL					mbEscherDgg;
 		BOOL					mbEscherDg;
 		BOOL					mbOleEmf;					// OLE is EMF instead of WMF
+#if SUPD == 310
+        rtl::OUString               mEditAs;
+#endif	// SUPD == 310
 
 
 		virtual BOOL DoSeek( UINT32 nKey );
@@ -1432,6 +1443,10 @@
 		virtual BOOL InsertAtPersistOffset( UINT32 nKey, UINT32 nValue );// nValue wird im Stream an entrsprechender Stelle eingefuegt(overwrite modus), ohne dass sich die
 																	// aktuelle StreamPosition aendert
 
+#if SUPD == 310
+    void            SetEditAs( const rtl::OUString& rEditAs );
+    rtl::OUString   GetEditAs() { return mEditAs; }
+#endif	// SUPD == 310
 		SvStream&	GetStream() const	{ return *mpOutStrm; }
 		ULONG	GetStreamPos() const	{ return mpOutStrm->Tell(); }
 
--- ./inc/svx/msocximex.hxx	2015-07-02 19:42:28.000000000 -0700
+++ ./inc/svx/msocximex.hxx	2015-08-04 20:49:41.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 #ifndef _MSOCXIMEX_HXX
 #define _MSOCXIMEX_HXX
@@ -71,6 +79,11 @@
         namespace uno{
                 class XComponentContext;
         }
+#if SUPD == 310
+        namespace frame{
+                class XModel;
+        }
+#endif	// SUPD == 310
 
 }}}
 
@@ -94,6 +107,9 @@
 class SVX_DLLPUBLIC SvxMSConvertOCXControls
 {
 public:
+#if SUPD == 310
+    SvxMSConvertOCXControls( const  ::com::sun::star::uno::Reference< ::com::sun::star::frame::XModel >& xModel );
+#endif	// SUPD == 310
     SvxMSConvertOCXControls( SfxObjectShell *pDSh,SwPaM *pP );
     virtual ~SvxMSConvertOCXControls();
 
@@ -155,6 +171,9 @@
 	const com::sun::star::uno::Reference<
 		com::sun::star::container::XIndexContainer > & 	GetFormComps();
 
+#if SUPD == 310
+    ::com::sun::star::uno::Reference< ::com::sun::star::frame::XModel > mxModel;
+#endif	// SUPD == 310
 	SfxObjectShell *pDocSh;
 	SwPaM *pPaM;
 
--- ./inc/svx/svdobj.hxx	2009-01-20 02:49:16.000000000 -0800
+++ ./inc/svx/svdobj.hxx	2015-08-04 21:55:51.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 #ifndef _SVDOBJ_HXX
@@ -77,6 +85,9 @@
 class SfxPoolItem;
 class SdrVirtObj;
 class SdrDragView;
+#if SUPD == 310
+class SfxGrabBagItem;
+#endif	// SUPD == 310
 
 namespace sdr
 {
@@ -1122,6 +1133,10 @@
 	/** do not use directly, always use getSvxShape() if you have to! */
 	SvxShape* mpSvxShape;
 
+#if SUPD == 310
+    SfxGrabBagItem*             pGrabBagItem; // holds the GrabBagItem property
+#endif	// SUPD == 310
+
 };
 
 //************************************************************
--- ./inc/svx/ulspitem.hxx	2008-04-10 13:44:47.000000000 -0700
+++ ./inc/svx/ulspitem.hxx	2015-08-04 21:08:20.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 #ifndef _SVX_ULSPITEM_HXX
 #define _SVX_ULSPITEM_HXX
@@ -56,6 +64,9 @@
 {
 	USHORT nUpper;  //Oberer Rand
 	USHORT nLower;  //Unterer Rand
+#if SUPD == 310
+    bool       bContext; // Contextual spacing?
+#endif	// SUPD == 310
 	USHORT nPropUpper, nPropLower;		// relativ oder absolut (=100%)
 public:
 	TYPEINFO();
@@ -88,11 +99,17 @@
 
 	void SetUpperValue( const USHORT nU ) { nUpper = nU; }
 	void SetLowerValue( const USHORT nL ) { nLower = nL; }
+#if SUPD == 310
+    void SetContextValue( const bool bC )     { bContext = bC; }
+#endif	// SUPD == 310
 	void SetPropUpper( const USHORT nU ) { nPropUpper = nU; }
 	void SetPropLower( const USHORT nL ) { nPropLower = nL; }
 
 	USHORT GetUpper() const { return nUpper; }
 	USHORT GetLower() const { return nLower; }
+#if SUPD == 310
+    bool GetContext() const { return bContext; }
+#endif	// SUPD == 310
 	USHORT GetPropUpper() const { return nPropUpper; }
 	USHORT GetPropLower() const { return nPropLower; }
 };
--- ./source/items/frmitems.cxx	2008-11-10 08:06:12.000000000 -0800
+++ ./source/items/frmitems.cxx	2015-08-04 21:35:45.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
@@ -822,6 +830,9 @@
 
     nUpper( 0 ),
     nLower( 0 ),
+#if SUPD == 310
+    bContext( false ),
+#endif	// SUPD == 310
     nPropUpper( 100 ),
     nPropLower( 100 )
 {
@@ -836,6 +847,9 @@
 
     nUpper( nUp  ),
     nLower( nLow ),
+#if SUPD == 310
+    bContext( false ),
+#endif	// SUPD == 310
     nPropUpper( 100 ),
     nPropLower( 100 )
 {
@@ -932,6 +946,9 @@
 
 	return ( nUpper == ( (SvxULSpaceItem&)rAttr ).nUpper &&
 			 nLower == ( (SvxULSpaceItem&)rAttr ).nLower &&
+#if SUPD == 310
+			 bContext == ( (SvxULSpaceItem&)rAttr ).bContext &&
+#endif	// SUPD == 310
 			 nPropUpper == ( (SvxULSpaceItem&)rAttr ).nPropUpper &&
 			 nPropLower == ( (SvxULSpaceItem&)rAttr ).nPropLower );
 }
--- ./source/msfilter/escherex.cxx	2015-07-02 19:42:34.000000000 -0700
+++ ./source/msfilter/escherex.cxx	2015-08-04 21:16:01.000000000 -0700
@@ -4455,6 +4455,15 @@
 	PtInsert( ESCHER_Persist_PrivateEntry | nKey, nOffset );
 }
 
+#if SUPD == 310
+
+void EscherEx::SetEditAs( const OUString& rEditAs )
+{
+    mEditAs = rEditAs;
+}
+
+#endif	// SUPD == 310
+
 // ---------------------------------------------------------------------------------------------
 
 BOOL EscherEx::DoSeek( UINT32 nKey )
--- ./source/msfilter/msocximex.cxx	2015-07-02 19:42:28.000000000 -0700
+++ ./source/msfilter/msocximex.cxx	2015-08-04 21:18:10.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
@@ -1194,6 +1202,17 @@
 const uno::Reference< drawing::XDrawPage >&
 	SvxMSConvertOCXControls::GetDrawPage()
 {
+#if SUPD == 310
+    if( !xDrawPage.is() && mxModel.is() )
+    {
+        uno::Reference< drawing::XDrawPageSupplier > xTxtDoc(mxModel,
+            uno::UNO_QUERY);
+        OSL_ENSURE(xTxtDoc.is(),"no XDrawPageSupplier from XModel");
+        xDrawPage = xTxtDoc->getDrawPage();
+        OSL_ENSURE( xDrawPage.is(), "no XDrawPage" );
+    }
+#endif	// SUPD == 310
+
 	if( !xDrawPage.is() && pDocSh )
 	{
 		uno::Reference< drawing::XDrawPageSupplier > xTxtDoc(pDocSh->GetModel(),
@@ -1210,6 +1229,16 @@
 const uno::Reference< lang::XMultiServiceFactory >&
 	SvxMSConvertOCXControls::GetServiceFactory()
 {
+#if SUPD == 310
+    if( !xServiceFactory.is() && mxModel.is() )
+    {
+        xServiceFactory = uno::Reference< lang::XMultiServiceFactory >
+            (mxModel, uno::UNO_QUERY);
+        OSL_ENSURE( xServiceFactory.is(),
+                "no XMultiServiceFactory from doc Model" );
+    }
+#endif	// SUPD == 310
+
 	if( !xServiceFactory.is() && pDocSh )
 	{
 		xServiceFactory = uno::Reference< lang::XMultiServiceFactory >
@@ -4648,6 +4677,17 @@
 
 const int NO_OCX = sizeof( aOCXTab ) / sizeof( *aOCXTab );
 
+#if SUPD == 310
+
+SvxMSConvertOCXControls::SvxMSConvertOCXControls( const uno::Reference< frame::XModel >& rxModel) : mxModel(rxModel)
+#if SUPD == 310
+    , pDocSh(0), pPaM(0), nEdit(0), nCheckbox(0)
+#endif	// SUPD == 310
+{
+}
+
+#endif	// SUPD == 310
+
 SvxMSConvertOCXControls::SvxMSConvertOCXControls(SfxObjectShell *pDSh, SwPaM *pP) :
     pDocSh(pDSh), pPaM(pP), nEdit(0), nCheckbox(0)
 {
--- ./source/svdraw/svdobj.cxx	2009-07-17 00:24:42.000000000 -0700
+++ ./source/svdraw/svdobj.cxx	2015-08-04 21:55:12.000000000 -0700
@@ -26,6 +26,14 @@
  * <http://www.openoffice.org/license.html>
  * for a copy of the LGPLv3 License.
  *
+ * This file incorporates work covered by the following license notice:
+ *
+ *   Portions of this file are part of the LibreOffice project.
+ *
+ *   This Source Code Form is subject to the terms of the Mozilla Public
+ *   License, v. 2.0. If a copy of the MPL was not distributed with this
+ *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
  ************************************************************************/
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
@@ -441,6 +449,9 @@
     mnNavigationPosition(SAL_MAX_UINT32),
 	mnLayerID(0),
 	mpSvxShape(0)
+#if SUPD == 310
+    ,pGrabBagItem(NULL)
+#endif	// SUPD == 310
 {
 	DBG_CTOR(SdrObject,NULL);
 	bVirtObj         =FALSE;
--- ./inc/svx/tbcontrl.hxx	2008-04-10 13:42:37.000000000 -0700
+++ ./inc/svx/tbcontrl.hxx	2015-08-04 22:28:59.000000000 -0700
@@ -162,6 +162,7 @@
 #include <sfx2/tbxctrl.hxx>
 #include <svx/strarray.hxx>
 #include "svx/svxdllapi.h"
+#include <svx/boxitem.hxx>
 
 #include <com/sun/star/awt/FontDescriptor.hpp>
 
@@ -289,6 +290,7 @@
 class SVX_DLLPUBLIC SvxFontColorToolBoxControl : public SfxToolBoxControl
 {
     ::svx::ToolboxButtonColorUpdater*   pBtnUpdater;
+    Color                               aCurColor;
 
 public:
 	SFX_DECL_TOOLBOX_CONTROL();
@@ -299,6 +301,8 @@
 											  const SfxPoolItem* pState );
 	virtual SfxPopupWindowType	GetPopupWindowType() const;
 	virtual SfxPopupWindow*		CreatePopupWindow();
+	virtual void				Select( BOOL bMod1 = FALSE );
+	DECL_LINK( ColorChangedHdl, Color * );
 };
 
 
@@ -316,6 +320,7 @@
 
 	//	SfxStatusForwarder 			aForward;
     ::svx::ToolboxButtonColorUpdater*   pBtnUpdater;
+    Color                               aCurColor;
 
 public:
 	SFX_DECL_TOOLBOX_CONTROL();
@@ -340,6 +345,7 @@
 class SVX_DLLPUBLIC SvxColorToolBoxControl : public SfxToolBoxControl
 {
     ::svx::ToolboxButtonColorUpdater*   pBtnUpdater;
+    Color                               aCurColor;
 public:
 	SFX_DECL_TOOLBOX_CONTROL();
 	SvxColorToolBoxControl( USHORT nSlotId, USHORT nId, ToolBox& rTbx );
@@ -349,6 +355,8 @@
 											  const SfxPoolItem* pState );
 	virtual SfxPopupWindowType	GetPopupWindowType() const;
 	virtual SfxPopupWindow*		CreatePopupWindow();
+	virtual void				Select( BOOL bMod1 = FALSE );
+	DECL_LINK( ColorChangedHdl, Color * );
 };
 
 
@@ -358,8 +366,12 @@
 //========================================================================
 
 
+class SvxFrameWindowState_Impl;
+
 class SVX_DLLPUBLIC SvxFrameToolBoxControl : public SfxToolBoxControl
 {
+    SvxBoxItem                          aCurBorderOuter;
+    SvxBoxInfoItem                      aCurBorderInner;
 public:
 	SFX_DECL_TOOLBOX_CONTROL();
 	SvxFrameToolBoxControl( USHORT nSlotId, USHORT nId, ToolBox& rTbx );
@@ -368,7 +380,8 @@
 	virtual SfxPopupWindow*		CreatePopupWindow();
 	virtual void				StateChanged( USHORT nSID, SfxItemState eState,
 											  const SfxPoolItem* pState );
-
+	virtual void				Select( BOOL bMod1 = FALSE );
+	DECL_LINK( BorderChangedHdl, SvxFrameWindowState_Impl * );
 };
 
 
--- ./source/tbxctrls/colorwindow.hxx	2008-04-10 20:34:47.000000000 -0700
+++ ./source/tbxctrls/colorwindow.hxx	2015-08-04 22:35:39.000000000 -0700
@@ -9,6 +9,8 @@
 #include <rtl/ustring.hxx>
 #include <com/sun/star/frame/XFrame.hpp>
 
+#include <map>
+
 //========================================================================
 // class SvxColorWindow_Impl --------------------------------------------------
 //========================================================================
@@ -21,6 +23,8 @@
 	const USHORT	                                                    theSlotId;
 	ValueSet		                                                    aColorSet;
     rtl::OUString                                                       maCommand;
+    std::map< ColorData, USHORT >                                       maColorSetMap;
+    Link                                                                maColorChangedHdl;
 
 #if _SOLAR__PRIVATE
 	DECL_LINK( SelectHdl, void * );
@@ -43,6 +47,8 @@
     virtual void        StateChanged( USHORT nSID, SfxItemState eState, const SfxPoolItem* pState );
 
 	virtual SfxPopupWindow* Clone() const;
+
+    void                SetColorChangedHdl( const Link& rLink ) { maColorChangedHdl = rLink; }
 };
 
 #endif
--- ./source/tbxctrls/tbcontrl.cxx	2008-11-10 08:06:12.000000000 -0800
+++ ./source/tbxctrls/tbcontrl.cxx	2015-08-04 22:31:59.000000000 -0700
@@ -248,6 +248,16 @@
 	ValueSet::MouseButtonUp(rMEvt);
 }
 
+class SAL_DLLPRIVATE SvxFrameWindowState_Impl
+{
+public:
+	const SvxBoxItem		maBorderOuter;
+	const SvxBoxInfoItem	maBorderInner;
+
+							SvxFrameWindowState_Impl( SvxBoxItem& rBorderOuter, SvxBoxInfoItem& rBorderInner ) : maBorderOuter( rBorderOuter ), maBorderInner( rBorderInner ) {}
+							~SvxFrameWindowState_Impl() {}
+};
+
 class SvxFrameWindow_Impl : public SfxPopupWindow
 {
 	using FloatingWindow::StateChanged;
@@ -256,6 +266,7 @@
     SvxFrmValueSet_Impl  aFrameSet;
 	ImageList 		aImgList;
     sal_Bool        bParagraphMode;
+    Link            maBorderChangedHdl;
 
 #if _SOLAR__PRIVATE
 	DECL_LINK( SelectHdl, void * );
@@ -278,6 +289,7 @@
 	virtual void	DataChanged( const DataChangedEvent& rDCEvt );
 
 	inline BOOL		IsHighContrast( void ) const;
+	void			SetBorderChangedHdl( const Link& rLink ) { maBorderChangedHdl = rLink; }
 };
 
 inline BOOL SvxFrameWindow_Impl::IsHighContrast( void ) const
@@ -891,8 +903,15 @@
 		{
 			pEntry = pColorTable->GetColor(i);
 			aColorSet.InsertItem( i+1, pEntry->GetColor(), pEntry->GetName() );
+			maColorSetMap[ pEntry->GetColor().GetColor() ] = i+1;
 		}
 
+		if ( i < PALETTE_SIZE )
+		{
+			std::map< ColorData, USHORT >::const_iterator it = maColorSetMap.find( aColWhite.GetColor() );
+			if ( it == maColorSetMap.end() )
+				maColorSetMap[ aColWhite.GetColor() ] = i+1;
+		}
         while ( i < PALETTE_SIZE )
 		{
             // fill empty elements if less then PALLETTE_SIZE colors are available
@@ -914,6 +933,26 @@
 	aColorSet.Show();
 
     AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:ColorTableState" )));
+    switch ( theSlotId )
+    {
+        case SID_ATTR_CHAR_COLOR_BACKGROUND:
+            AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:FillColor" )));
+            break;
+        case SID_ATTR_CHAR_COLOR:
+            AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:Color" )));
+            break;
+        case SID_ATTR_CHAR_COLOR2:
+            AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:Color" )));
+            break;
+        case SID_BACKGROUND_COLOR:
+            AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:BackgroundColor" )));
+            break;
+        case SID_EXTRUSION_3D_COLOR:
+            AddStatusListener( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:Extrusion3DColor" )));
+            break;
+        default:
+            break;
+    }
 	if ( bKillTable )
 		delete pColorTable;
 }
@@ -953,6 +992,11 @@
         SfxToolBoxControl::Dispatch( Reference< XDispatchProvider >( GetFrame()->getController(), UNO_QUERY ),
                                      maCommand,
                                      aArgs );
+        if ( maColorChangedHdl.IsSet() )
+        {
+            Color aColor( COL_TRANSPARENT );
+            maColorChangedHdl.Call( &aColor );
+        }
     }
     else if ( !nItemId && (SID_ATTR_CHAR_COLOR == theSlotId || SID_ATTR_CHAR_COLOR2  == theSlotId || SID_EXTRUSION_3D_COLOR == theSlotId) )
     {
@@ -967,6 +1011,11 @@
         SfxToolBoxControl::Dispatch( Reference< XDispatchProvider >( GetFrame()->getController(), UNO_QUERY ),
                                      maCommand,
                                      aArgs );
+        if ( maColorChangedHdl.IsSet() )
+        {
+            Color aColor( COL_TRANSPARENT );
+            maColorChangedHdl.Call( &aColor );
+        }
     }
     else
 	{
@@ -980,6 +1029,11 @@
         SfxToolBoxControl::Dispatch( Reference< XDispatchProvider >( GetFrame()->getController(), UNO_QUERY ),
                                      maCommand,
                                      aArgs );
+        if ( maColorChangedHdl.IsSet() )
+        {
+            Color aColor( aColorItem.GetValue() );
+            maColorChangedHdl.Call( &aColor );
+        }
 	}
 
 	return 0;
@@ -1032,14 +1086,22 @@
 				else
 					nBits |= WB_VSCROLL;
 				aColorSet.SetStyle( nBits );
+				maColorSetMap.clear();
 
 				for ( i = 0; i < nCount; ++i )
 				{
 					pEntry = pColorTable->GetColor(i);
 					aColorSet.SetItemColor( i + 1, pEntry->GetColor() );
 					aColorSet.SetItemText ( i + 1, pEntry->GetName() );
+					maColorSetMap[ pEntry->GetColor().GetColor() ] = i + 1;
 				}
 
+				if ( i < PALETTE_SIZE )
+				{
+					std::map< ColorData, USHORT >::const_iterator it = maColorSetMap.find( aColWhite.GetColor() );
+					if ( it == maColorSetMap.end() )
+						maColorSetMap[ aColWhite.GetColor() ] = i + 1;
+				}
         		while ( i < PALETTE_SIZE )
 				{
 					aColorSet.SetItemColor( i + 1, aColWhite );
@@ -1048,6 +1110,22 @@
 				}
 			}
 		}
+		else if ( pState->ISA( SvxColorItem ) )
+		{
+			Color aColor( ((SvxColorItem *)pState)->GetValue() );
+			if ( aColorSet.IsNoSelection() || aColor != aColorSet.GetItemColor( aColorSet.GetSelectItemId() ) )
+			{
+				aColorSet.SetNoSelection();
+				std::map< ColorData, USHORT >::const_iterator it = maColorSetMap.find( aColor.GetColor() );
+				if ( it != maColorSetMap.end() )
+					aColorSet.SelectItem( it->second );
+				else
+					aColorSet.SelectItem( 0 );
+
+				if ( maColorChangedHdl.IsSet() )
+					maColorChangedHdl.Call( &aColor );
+			}
+		}
 	}
 }
 
@@ -1250,6 +1328,12 @@
     aBorderInner.QueryValue( a );
     aArgs[1].Value = a;
 
+    if ( maBorderChangedHdl.IsSet() )
+    {
+        SvxFrameWindowState_Impl aBorderState( aBorderOuter, aBorderInner );
+        maBorderChangedHdl.Call( &aBorderState );
+    }
+
     /*  #i33380# DR 2004-09-03 Moved the following line above the Dispatch() call.
         This instance may be deleted in the meantime (i.e. when a dialog is opened
         while in Dispatch()), accessing members will crash in this case. */
@@ -2314,6 +2398,8 @@
                     nSlotId, nId, &GetToolBox(), TBX_UPDATER_MODE_CHAR_COLOR_NEW ))
 {
 	rTbx.SetItemBits( nId, TIB_DROPDOWN | rTbx.GetItemBits( nId ) );
+	aCurColor = Color( COL_TRANSPARENT );
+	pBtnUpdater->Update( aCurColor );
 }
 
 // -----------------------------------------------------------------------
@@ -2346,6 +2432,7 @@
         FLOATWIN_POPUPMODE_GRABFOCUS|FLOATWIN_POPUPMODE_ALLOWTEAROFF );
 	pColorWin->StartSelection();
     SetPopupWindow( pColorWin );
+    pColorWin->SetColorChangedHdl( LINK( this, SvxFontColorToolBoxControl, ColorChangedHdl ) );
 	return pColorWin;
 }
 
@@ -2358,18 +2445,40 @@
 {
 	USHORT nId = GetId();
 	ToolBox& rTbx = GetToolBox();
-	const SvxColorItem*	pItem = 0;
-
-	if ( SFX_ITEM_DONTCARE != eState )
-	   pItem = PTR_CAST( SvxColorItem, pState );
-
-	if ( pItem )
-		pBtnUpdater->Update( pItem->GetValue());
 
 	rTbx.EnableItem( nId, SFX_ITEM_DISABLED != eState );
 	rTbx.SetItemState( nId, ( SFX_ITEM_DONTCARE == eState ) ? STATE_DONTKNOW : STATE_NOCHECK );
 }
 
+// -----------------------------------------------------------------------
+
+void SvxFontColorToolBoxControl::Select( BOOL )
+{
+    OUString aCommand( RTL_CONSTASCII_USTRINGPARAM( ".uno:Color" ) );
+    SvxColorItem aColorItem( aCurColor.GetTransparency() ? COL_AUTO : aCurColor, GetSlotId() );
+    INetURLObject aObj( aCommand );
+
+    Any a;
+    Sequence< PropertyValue > aArgs( 1 );
+    aArgs[0].Name = aObj.GetURLPath();
+    aColorItem.QueryValue( a );
+    aArgs[0].Value = a;
+    Dispatch( aCommand, aArgs );
+}
+
+// -----------------------------------------------------------------------
+
+IMPL_LINK( SvxFontColorToolBoxControl, ColorChangedHdl, Color *, pColor )
+{
+	if ( pColor )
+	{
+		aCurColor = *pColor;
+		pBtnUpdater->Update( aCurColor );
+	}
+
+	return 0;
+}
+
 //========================================================================
 // class SvxColorToolBoxControl --------------------------------
 //========================================================================
@@ -2378,12 +2487,11 @@
 
 	SfxToolBoxControl( nSlotId, nId, rTbx )
 {
-    if ( nSlotId == SID_BACKGROUND_COLOR )
-        rTbx.SetItemBits( nId, TIB_DROPDOWNONLY | rTbx.GetItemBits( nId ) );
-    else
-        rTbx.SetItemBits( nId, TIB_DROPDOWN | rTbx.GetItemBits( nId ) );
+    rTbx.SetItemBits( nId, TIB_DROPDOWN | rTbx.GetItemBits( nId ) );
 	rTbx.Invalidate();
     pBtnUpdater = new ::svx::ToolboxButtonColorUpdater( nSlotId, nId, &GetToolBox() );
+	aCurColor = Color( COL_TRANSPARENT );
+	pBtnUpdater->Update( aCurColor );
 }
 
 // -----------------------------------------------------------------------
@@ -2417,6 +2525,7 @@
         FLOATWIN_POPUPMODE_GRABFOCUS|FLOATWIN_POPUPMODE_ALLOWTEAROFF );
 	pColorWin->StartSelection();
     SetPopupWindow( pColorWin );
+    pColorWin->SetColorChangedHdl( LINK( this, SvxColorToolBoxControl, ColorChangedHdl ) );
 	return pColorWin;
 }
 
@@ -2427,19 +2536,49 @@
 	USHORT , SfxItemState eState, const SfxPoolItem* pState )
 
 {
-	const SvxColorItem*	pItem	= 0;
-	if ( SFX_ITEM_DONTCARE != eState )
-		pItem = PTR_CAST( SvxColorItem, pState );
-
-	if ( pItem )
-		pBtnUpdater->Update( pItem->GetValue() );
-
 	USHORT nId = GetId();
 	ToolBox& rTbx = GetToolBox();
 	rTbx.EnableItem( nId, SFX_ITEM_DISABLED != eState );
 	rTbx.SetItemState( nId, ( SFX_ITEM_DONTCARE == eState ) ? STATE_DONTKNOW : STATE_NOCHECK );
 }
 
+// -----------------------------------------------------------------------
+
+void SvxColorToolBoxControl::Select( BOOL )
+{
+    OUString aCommand( RTL_CONSTASCII_USTRINGPARAM( ".uno:BackgroundColor" ) );
+    if ( aCurColor.GetTransparency() )
+    {
+        Sequence< PropertyValue > aArgs;
+        Dispatch( aCommand, aArgs );
+    }
+    else
+    {
+        SvxColorItem aColorItem( aCurColor, GetSlotId() );
+        INetURLObject aObj( aCommand );
+
+        Any a;
+        Sequence< PropertyValue > aArgs( 1 );
+        aArgs[0].Name = aObj.GetURLPath();
+        aColorItem.QueryValue( a );
+        aArgs[0].Value = a;
+        Dispatch( aCommand, aArgs );
+    }
+}
+
+// -----------------------------------------------------------------------
+
+IMPL_LINK( SvxColorToolBoxControl, ColorChangedHdl, Color *, pColor )
+{
+	if ( pColor )
+	{
+		aCurColor = *pColor;
+		pBtnUpdater->Update( aCurColor );
+	}
+
+	return 0;
+}
+
 //========================================================================
 // class SvxFontColorExtToolBoxControl --------------------------------------
 //========================================================================
@@ -2462,6 +2601,7 @@
     USHORT nMode =	SID_ATTR_CHAR_COLOR2 == nSlotId
 		? TBX_UPDATER_MODE_CHAR_COLOR_NEW :	TBX_UPDATER_MODE_CHAR_COLOR_NEW;
     pBtnUpdater = new ::svx::ToolboxButtonColorUpdater( nSlotId, nId, &GetToolBox(), nMode );
+    aCurColor = Color( COL_TRANSPARENT );
 }
 
 // -----------------------------------------------------------------------
@@ -2527,7 +2667,10 @@
 		   pItem = PTR_CAST( SvxColorItem, pState );
 
 		if ( pItem )
-			pBtnUpdater->Update( pItem->GetValue() );
+		{
+			aCurColor = pItem->GetValue();
+			pBtnUpdater->Update( aCurColor );
+		}
 	}
 }
 
@@ -2548,10 +2691,27 @@
         aParamName  = OUString( RTL_CONSTASCII_USTRINGPARAM( "CharBackgroundExt" ));
     }
 
+    // Fix bug reported in the following NeoOffice forum post by forcefully
+    // unsetting the highlight color when it was set to transparent via the
+    // color palette:
+    // http://trinity.neooffice.org/modules.php?name=Forums&file=viewtopic&p=63965#63965
+    BOOL bIsItemChecked = GetToolBox().IsItemChecked( GetId() );
+    if ( bIsItemChecked && GetSlotId() == SID_ATTR_CHAR_COLOR_BACKGROUND && aCurColor.GetTransparency() )
+    {
+        Sequence< PropertyValue > aColorArgs;
+        Dispatch( OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:BackColor" ) ), aColorArgs );
+    }
+
     Sequence< PropertyValue > aArgs( 1 );
     aArgs[0].Name  = aParamName;
     aArgs[0].Value = makeAny( GetToolBox().IsItemChecked( GetId() ));
     Dispatch( aCommand, aArgs );
+
+    if ( !bIsItemChecked && GetSlotId() == SID_ATTR_CHAR_COLOR_BACKGROUND && aCurColor.GetTransparency() )
+    {
+        Sequence< PropertyValue > aColorArgs;
+        Dispatch( OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:BackColor" ) ), aColorArgs );
+    }
 }
 
 //========================================================================
@@ -2564,8 +2724,10 @@
 	ToolBox&    rTbx )
 
     :   SfxToolBoxControl( nSlotId, nId, rTbx )
+    , aCurBorderOuter( SID_ATTR_BORDER_OUTER )
+    , aCurBorderInner( SID_ATTR_BORDER_INNER )
 {
-	rTbx.SetItemBits( nId, TIB_DROPDOWNONLY | rTbx.GetItemBits( nId ) );
+    rTbx.SetItemBits( nId, TIB_DROPDOWN | rTbx.GetItemBits( nId ) );
 }
 
 // -----------------------------------------------------------------------
@@ -2585,6 +2747,7 @@
 	pFrameWin->StartPopupMode( &GetToolBox(), FLOATWIN_POPUPMODE_GRABFOCUS | FLOATWIN_POPUPMODE_ALLOWTEAROFF );
 	pFrameWin->StartSelection();
     SetPopupWindow( pFrameWin );
+    pFrameWin->SetBorderChangedHdl( LINK( this, SvxFrameToolBoxControl, BorderChangedHdl ) );
 
 	return pFrameWin;
 }
@@ -2605,6 +2768,34 @@
                             : STATE_NOCHECK );
 }
 
+// -----------------------------------------------------------------------
+
+void SvxFrameToolBoxControl::Select( BOOL )
+{
+	Any a;
+	Sequence< PropertyValue > aArgs( 2 );
+	aArgs[0].Name = OUString( RTL_CONSTASCII_USTRINGPARAM( "OuterBorder" ));
+	aCurBorderOuter.QueryValue( a );
+	aArgs[0].Value = a;
+	aArgs[1].Name = OUString( RTL_CONSTASCII_USTRINGPARAM( "InnerBorder" ));
+	aCurBorderInner.QueryValue( a );
+	aArgs[1].Value = a;
+	Dispatch( OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:SetBorderStyle" ) ), aArgs );
+}
+
+// -----------------------------------------------------------------------
+
+IMPL_LINK( SvxFrameToolBoxControl, BorderChangedHdl, SvxFrameWindowState_Impl *, pBorderState )
+{
+	if ( pBorderState )
+	{
+		aCurBorderOuter = pBorderState->maBorderOuter;
+		aCurBorderInner = pBorderState->maBorderInner;
+	}
+
+	return 0;
+}
+
 //========================================================================
 // class SvxFrameLineStyleToolBoxControl ---------------------------------
 //========================================================================
