Index: prj/build.lst
===================================================================
RCS file: /cvs/tools/testshl2/prj/build.lst,v
retrieving revision 1.6
diff -u -r1.6 build.lst
--- prj/build.lst	11 Jul 2008 13:43:11 -0000	1.6
+++ prj/build.lst	22 Jul 2008 17:36:58 -0000
@@ -1,6 +1 @@
-t2  testshl2   :  cppunit NULL
-t2  testshl2                           usr1    -   all t2_mkout NULL
-t2  testshl2\inc                       nmake   -    all t2_inc NULL
-t2  testshl2\source                    nmake   -   all t2_src t2_inc NULL
-t2  testshl2\util                      nmake   -   all t2_util t2_src NULL
-
+t2  testshl2   :  NULL
