Index: util/makefile.mk
===================================================================
RCS file: /cvs/gsl/toolkit/util/makefile.mk,v
retrieving revision 1.18
diff -u -r1.18 makefile.mk
--- util/makefile.mk	9 Sep 2005 13:25:15 -0000	1.18
+++ util/makefile.mk	22 Dec 2005 18:48:54 -0000
@@ -78,16 +78,7 @@
 DEF1DES		=TK
 DEFLIB1NAME	=tk
 
-.IF "$(OS)"=="MACOSX" 
-
-# [ed] 6/16/02 Add in X libraries if we're building X
-
-.IF "$(GUIBASE)"=="unx"
-SHL1STDLIBS +=\
-	-lX11 -lXt -lXmu
-.ENDIF
-
-.ELSE
+.IF "$(OS)"!="MACOSX" 
 .IF "$(GUI)"=="UNX"
 SHL1STDLIBS +=\
 	-lX11
