Index: source/ucp/webdav/makefile.mk
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/makefile.mk,v
retrieving revision 1.29
diff -u -r1.29 makefile.mk
--- source/ucp/webdav/makefile.mk	19 Jul 2006 09:36:39 -0000	1.29
+++ source/ucp/webdav/makefile.mk	23 Nov 2006 23:04:53 -0000
@@ -121,9 +121,16 @@
         $(SALHELPERLIB)         \
         $(VOSLIB)               \
         $(UCBHELPERLIB)         \
-        $(NEON3RDLIB)           \
         $(XML2LIB)
 
+# With gcc 4.0.1 on Mac OS X, linking will fail unless we explicitly include
+# the archive in the link list
+.IF "$(OS)"=="MACOSX"
+SHL1STDLIBS+= $(SOLARLIBDIR)/libneon.a
+.ELSE
+SHL1STDLIBS+= $(NEON3RDLIB)
+.ENDIF
+
 .IF "$(GUI)"=="WNT"
 SHL1STDLIBS+= wsock32.lib
 .ENDIF # WNT
