Index: inc/ucbhelper/remotecontentproviderscontrol.hxx
===================================================================
RCS file: /cvs/ucb/ucbhelper/inc/ucbhelper/Attic/remotecontentproviderscontrol.hxx,v
retrieving revision 1.3
diff -u -r1.3 remotecontentproviderscontrol.hxx
--- inc/ucbhelper/remotecontentproviderscontrol.hxx	20 Jun 2001 07:48:19 -0000	1.3
+++ inc/ucbhelper/remotecontentproviderscontrol.hxx	27 Sep 2005 05:25:07 -0000
@@ -163,7 +163,7 @@
 
     void closed(Listeners::iterator aIt, bool bDistribution);
 
-    friend Listener; // to access Listeners, setClosed()
+    friend class Listener; // to access Listeners, setClosed()
 };
 
 }
Index: source/client/proxydecider.cxx
===================================================================
RCS file: /cvs/ucb/ucbhelper/source/client/proxydecider.cxx,v
retrieving revision 1.3.16.1
diff -u -r1.3.16.1 proxydecider.cxx
--- source/client/proxydecider.cxx	20 Oct 2004 12:04:36 -0000	1.3.16.1
+++ source/client/proxydecider.cxx	27 Sep 2005 05:25:08 -0000
@@ -125,6 +125,9 @@
 #else
 #include <netdb.h>
 #include <unistd.h>
+#ifdef MACOSX
+#include <sys/socket.h>
+#endif	// MACOSX
 #endif
 
 #ifdef SOLARIS
