Index: com/sun/star/uno/Exception.idl
===================================================================
RCS file: /cvs/api/udkapi/com/sun/star/uno/Exception.idl,v
retrieving revision 1.10
diff -u -r1.10 Exception.idl
--- com/sun/star/uno/Exception.idl	9 Sep 2005 16:31:32 -0000	1.10
+++ com/sun/star/uno/Exception.idl	6 Jan 2007 22:42:29 -0000
@@ -51,7 +51,11 @@
 	
 	<p>All exceptions defined in UNO idl should derive from this exception.</p>
  */
+#if defined MACOSX && __GNUC__ > 3
+published exception Exception: sal::MacOSXGCC4BaseClass
+#else	/* MACOSX && __GNUC__ > 3 */
 published exception Exception
+#endif	/* MACOSX && __GNUC__ > 3 */
 {
 	/** gives a detailed description of the reason, why the exception was thrown.
         <p>
