Index: inc/salnativewidgets.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/salnativewidgets.hxx,v
retrieving revision 1.7
diff -u -r1.7 salnativewidgets.hxx
--- inc/salnativewidgets.hxx	9 Sep 2005 11:21:13 -0000	1.7
+++ inc/salnativewidgets.hxx	10 May 2006 04:30:06 -0000
@@ -134,6 +134,9 @@
 // tool tips
 #define CTRL_TOOLTIP            140
 
+// progress bars
+#define CTRL_PROGRESSBAR		150
+
 /* Control Parts:
  *
  *   Uniquely identify a part of a control,
@@ -334,6 +337,25 @@
     int             maTopDockingAreaHeight;
 };
 
+/*  ProgressbarValue:
+ *
+ *  Value container for progressbars indicating task completion
+ */
+class ProgressbarValue
+{
+	public:
+		BOOL	mbIndeterminate;	// indcates if the progress bar is indeterminate (unknown action length) or determinate (able to determine completion)
+		double	mdPercentComplete;	// percentage in range [0.0, 100.0], only used for determinate progress bars
+		
+		ProgressbarValue()
+			{
+				mbIndeterminate = FALSE;
+				mdPercentComplete = 0;
+			};
+		
+		~ProgressbarValue() {};
+};
+
 /* ImplControlValue:
  *
  *   Generic value container for all control parts.
