--- inc/salgdi.hxx	2007-01-20 15:17:49.000000000 -0800
+++ inc/salgdi.hxx	2007-01-20 14:50:33.000000000 -0800
@@ -171,6 +171,8 @@
     virtual BOOL        getNativeControlRegion( ControlType nType, ControlPart nPart, const Region& rControlRegion, ControlState nState,
                                                 const ImplControlValue& aValue, SalControlHandle& rControlHandle, const rtl::OUString& aCaption,
                                                 Region &rNativeBoundingRegion, Region &rNativeContentRegion );
+    virtual BOOL        getNativeControlTextColor( ControlType nType, ControlPart nPart, ControlState nState, const ImplControlValue& aValue, SalColor& textColor );
+
 public:
     // public SalGraphics methods, the interface to the independent vcl part
 
@@ -439,6 +441,13 @@
 										 Region &rNativeBoundingRegion,
 										 Region &rNativeContentRegion,
                                          const OutputDevice *pOutDev );
+
+    // Query the VCL implementation for the text color that should be used to draw a native control
+    BOOL GetNativeControlTextColor( ControlType nType, 
+    									 ControlPart nPart, 
+    									 ControlState nState, 
+    									 const ImplControlValue& aValue, 
+    									 Color& textColor );
                                          
     static void AddDevFontSubstitute( OutputDevice* pOutDev,
                                       const String& rFontName,
Index: inc/salnativewidgets.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/salnativewidgets.hxx,v
retrieving revision 1.9
diff -u -r1.9 salnativewidgets.hxx
--- inc/salnativewidgets.hxx	19 Jun 2006 19:07:59 -0000	1.9
+++ inc/salnativewidgets.hxx	7 Jan 2007 02:17:52 -0000
@@ -138,6 +138,18 @@
 // tool tips
 #define CTRL_TOOLTIP            140
 
+// progress bars
+#define CTRL_PROGRESSBAR		150
+
+// hierarchical tree control expand/collapse handle
+#define CTRL_DISCLOSUREBTN		160
+
+// list view header cell appearing above columns in lists
+#define CTRL_LISTVIEWHEADER		170
+
+// native frame to draw around list views.  Default is sunken Win95 well.
+#define CTRL_LISTVIEWBOX		180
+
 /* Control Parts:
  *
  *   Uniquely identify a part of a control,
@@ -163,6 +175,7 @@
 #define PART_MENU_ITEM              250
 #define PART_MENU_ITEM_CHECK_MARK   251
 #define PART_MENU_ITEM_RADIO_MARK   252
+#define PART_LISTVIEWHEADER_SORT_MARK	150	// used for list view headers to indicate whether they draw an ascending/descending sort indicator or whether VCL should handle the indicator
 
 /** The edit field part of a control, e.g. of the combo box.
 
@@ -340,6 +353,79 @@
     int             maTopDockingAreaHeight;
 };
 
+/*  ProgressbarValue:
+ *
+ *  Value container for progressbars indicating task completion
+ */
+class ProgressbarValue
+{
+	public:
+		BOOL	mbIndeterminate;	// indcates if the progress bar is indeterminate (unknown action length) or determinate (able to determine completion)
+		double	mdPercentComplete;	// percentage in range [0.0, 100.0], only used for determinate progress bars
+		
+		ProgressbarValue()
+			{
+				mbIndeterminate = FALSE;
+				mdPercentComplete = 0;
+			};
+		
+		~ProgressbarValue() {};
+};
+
+#define DISCLOSUREBTN_CLOSED	0		// group for this control is currently closed, displaying the container only
+#define DISCLOSUREBTN_OPEN		1		// group for this control is currently expanded, displaying all elements within the container
+
+#define DISCLOSUREBTN_ALIGN_LEFT	0	// disclosure controls appear to the left of any items in the left margin of the column
+#define DISCLOSUREBTN_ALIGN_RIGHT	1	// disclosure contorls appear to the right of any items in the right margin of the column
+
+/*	DisclosureBtnValue:
+ *
+ *	Value container for disclosure buttons used for control of expanding/collapsing
+ *	tree views
+ */
+class DisclosureBtnValue
+{
+	public:
+		USHORT mnOpenCloseState;		// indicates whether the controls associated container is open or closed, DISCLOSURE_OPEN/CLOSED
+		USHORT mnAlignment;				// indicates whether the disclosure control appears to the left or right of containers
+		BOOL mbHasChildren;				// true if the node has any children, false if not
+		
+		DisclosureBtnValue()
+			{
+				mnOpenCloseState = DISCLOSUREBTN_CLOSED;
+				mnAlignment = DISCLOSUREBTN_ALIGN_LEFT;
+				mbHasChildren = 1;
+			};
+		
+		~DisclosureBtnValue() {};
+};
+
+#define LISTVIEWHEADER_SORT_DONTKNOW	0	// indicates the current sort of the column is unknown
+#define LISTVIEWHEADER_SORT_DESCENDING	1	// indicates the column is sorted in descending order
+#define LISTVIEWHEADER_SORT_ASCENDING	2	// indicates the column is sorted in ascending order
+#define LISTVIEWHEADER_SORT_UNSORTED	3	// indicates the column is not sorted
+
+/*	ListViewHeaderValue
+ *
+ *	Value container for list view header cells drawn above columns of scrollable
+ *	lists.  Indicates whether the clumn is the primary sort column and
+ *	any direction of the sort
+ */
+class ListViewHeaderValue
+{
+	public:
+		BOOL mbPrimarySortColumn;		// true if the column is the primary active column on which data is sorted, false if it is a secondary informational column
+		USHORT mnSortDirection;			// sort direction for the column.
+		
+		ListViewHeaderValue()
+			{
+				mbPrimarySortColumn = FALSE;
+				mnSortDirection = LISTVIEWHEADER_SORT_DONTKNOW;
+			};
+		
+		~ListViewHeaderValue() {};
+};
+
 /* ImplControlValue:
  *
  *   Generic value container for all control parts.
--- inc/window.hxx	2006-10-12 07:57:43.000000000 -0700
+++ inc/window.hxx	2007-01-20 14:50:43.000000000 -0800
@@ -1079,6 +1079,13 @@
 										  Region &rNativeBoundingRegion,
 										  Region &rNativeContentRegion );
 
+    // Query the VCL implementation for the text color that should be used to draw a native control
+    BOOL GetNativeControlTextColor( ControlType nType, 
+    									 ControlPart nPart, 
+    									 ControlState nState, 
+    									 const ImplControlValue& aValue, 
+    									 Color& textColor );
+
     // a helper method for a Control's Draw method
     void PaintToDevice( OutputDevice* pDevice, const Point& rPos, const Size& rSize );
 };
--- source/gdi/salgdilayout.cxx	2006-09-17 05:10:36.000000000 -0700
+++ source/gdi/salgdilayout.cxx	2007-01-20 15:26:56.000000000 -0800
@@ -620,6 +620,19 @@
                                                 rNativeBoundingRegion, rNativeContentRegion );
 }
 
+BOOL SalGraphics::GetNativeControlTextColor( ControlType nType, 
+									 ControlPart nPart, 
+									 ControlState nState, 
+									 const ImplControlValue& aValue, 
+									 Color& textColor )
+{
+	SalColor aSalColor;
+	BOOL bRet = getNativeControlTextColor( nType, nPart, nState, aValue, aSalColor );
+	if ( bRet )
+		textColor = Color( aSalColor );
+	return bRet;
+}
+
 bool SalGraphics::filterText( const String&, String&, xub_StrLen, xub_StrLen&, xub_StrLen&, xub_StrLen& )
 {
     return false;
Index: source/gdi/salnativewidgets-none.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/salnativewidgets-none.cxx,v
retrieving revision 1.5
diff -u -r1.5 salnativewidgets-none.cxx
--- source/gdi/salnativewidgets-none.cxx	17 Sep 2006 12:11:27 -0000	1.5
+++ source/gdi/salnativewidgets-none.cxx	7 Jan 2007 02:17:53 -0000
@@ -147,3 +147,17 @@
 	return( FALSE );
 }
 
+/*
+ * getNativeControlTextColor()
+ *
+ *  If the return value is TRUE, textColor contains the color with which the text
+ *	should be drawn given the control state and other values.  If the return value
+ *	is FALSE, the control text should be drawn with the platform independent Sal
+ *	control colors returned by each SalApplication derivative.
+ *
+ *  aValue:		An optional value (tristate/numerical/string)
+ */
+BOOL SalGraphics::getNativeControlTextColor(  ControlType nType, ControlPart nPart, ControlState nState, const ImplControlValue& aValue, SalColor& textColor )
+{
+	return( FALSE );
+}
--- source/window/window3.cxx	2006-09-17 05:24:10.000000000 -0700
+++ source/window/window3.cxx	2007-01-20 14:50:41.000000000 -0800
@@ -312,6 +312,29 @@
 
 // -----------------------------------------------------------------------
 
+BOOL Window::GetNativeControlTextColor( ControlType nType, 
+									 ControlPart nPart, 
+									 ControlState nState, 
+									 const ImplControlValue& aValue, 
+									 Color& textColor )
+{
+    if( !IsNativeWidgetEnabled() )
+        return FALSE;
+
+    if ( !mpGraphics )
+        if ( !ImplGetGraphics() )
+            return FALSE;
+
+    ImplInitSalControlHandle();
+	
+	BOOL bRet = mpGraphics->GetNativeControlTextColor(nType, nPart, nState, aValue, textColor);
+	
+	return bRet;
+}
+
+
+// -----------------------------------------------------------------------
+
 WaitObject::~WaitObject()
 {
     if ( mpWindow )
Index: util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.82
diff -u -r1.82 makefile.mk
--- util/makefile.mk	6 Oct 2006 10:07:49 -0000	1.82
+++ util/makefile.mk	7 Jan 2007 02:17:54 -0000
@@ -275,7 +275,7 @@
 .IF "$(USE_XINERAMA)" != "NO"
 
 .IF "$(OS)"=="MACOSX"
-XINERAMALIBS=-lXinerama
+XINERAMALIBS= /usr/X11R6/lib/libXinerama.a
 .ELSE
 .IF "$(OS)" != "SOLARIS" || "$(USE_XINERAMA_VERSION)" == "Xorg"
 .IF "$(XINERAMA_LINK)" == "dynamic"
