Index: inc/glyphcache.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/glyphcache.hxx,v
retrieving revision 1.19.42.1
diff -u -r1.19.42.1 glyphcache.hxx
--- inc/glyphcache.hxx	25 Aug 2004 09:35:11 -0000	1.19.42.1
+++ inc/glyphcache.hxx	27 Sep 2005 05:27:50 -0000
@@ -124,7 +124,7 @@
     void                        RemoveFont( const ImplFontData* );
 
 private:
-friend ServerFont;
+friend class ServerFont;
     // used by ServerFont class only
     void                        AddedGlyph( ServerFont& rServerFont, GlyphData& rGlyphData );
     void                        RemovingGlyph( ServerFont& rSF, GlyphData& rGD, int nGlyphIndex );
@@ -244,8 +244,8 @@
     void*                       GetExtPointer() { return mpExtData; }
 
 protected:
-friend GlyphCache;
-friend ServerFontLayout;
+friend class GlyphCache;
+friend class ServerFontLayout;
                                 ServerFont( const ImplFontSelectData&, const glyphKernMap* pGlyphKern = NULL, const unicodeKernMap* pUniKern = NULL );
     virtual                     ~ServerFont();
 
Index: source/glyphs/gcach_ftyp.hxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_ftyp.hxx,v
retrieving revision 1.24.40.1
diff -u -r1.24.40.1 gcach_ftyp.hxx
--- source/glyphs/gcach_ftyp.hxx	25 Aug 2004 09:35:49 -0000	1.24.40.1
+++ source/glyphs/gcach_ftyp.hxx	27 Sep 2005 05:27:51 -0000
@@ -213,7 +213,7 @@
     int                         GetEmUnits() const;
 
 protected:
-friend GlyphCache;
+friend class GlyphCache;
 
     int                         ApplyGlyphTransform( int nGlyphFlags, FT_GlyphRec_* ) const;
     virtual void                InitGlyphData( int nGlyphIndex, GlyphData& ) const;
Index: source/glyphs/gcach_vdev.hxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_vdev.hxx,v
retrieving revision 1.5
diff -u -r1.5 gcach_vdev.hxx
--- source/glyphs/gcach_vdev.hxx	15 Feb 2002 15:58:57 -0000	1.5
+++ source/glyphs/gcach_vdev.hxx	27 Sep 2005 05:27:51 -0000
@@ -72,7 +72,7 @@
 	virtual bool				GetGlyphOutline( int nGlyphIndex, PolyPolygon& ) const;
 
 protected:
-friend GlyphCache;
+friend class GlyphCache;
 	static long					FetchFontList( ImplDevFontList* );
 	static void					ClearFontList();
 
Index: unx/source/gdi/xlfd_extd.hxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/xlfd_extd.hxx,v
retrieving revision 1.9.120.2
diff -u -r1.9.120.2 xlfd_extd.hxx
--- unx/source/gdi/xlfd_extd.hxx	26 Jan 2004 17:27:37 -0000	1.9.120.2
+++ unx/source/gdi/xlfd_extd.hxx	27 Sep 2005 05:27:52 -0000
@@ -93,8 +93,8 @@
 
 class ExtendedXlfd {
 
-	friend VirtualXlfd;
-	friend XlfdStorage;
+	friend class VirtualXlfd;
+	friend class XlfdStorage;
 
 	public:
 			 				ExtendedXlfd();
Index: util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.47.10.4
diff -u -r1.47.10.4 makefile.mk
--- util/makefile.mk	14 Sep 2004 13:17:44 -0000	1.47.10.4
+++ util/makefile.mk	27 Sep 2005 05:27:52 -0000
@@ -299,7 +299,7 @@
 SHL1STDLIBS += -lXp -lXext -lSM -lICE -lX11
 .ELSE
 .IF "$(USE_XINERAMA)" != "NO"
-.IF "$(XINERAMA_LINK)" == "dynamic"
+.IF "$(XINERAMA_LINK)" == "dynamic" || "$(OS)" == "MACOSX"
 SHL1STDLIBS += -lXinerama
 .ELSE
 SHL1STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
