Index: util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.47.10.2.28.2
diff -u -r1.47.10.2.28.2 makefile.mk
--- util/makefile.mk	13 Aug 2004 01:57:38 -0000	1.47.10.2.28.2
+++ util/makefile.mk	17 Dec 2004 04:53:43 -0000
@@ -299,7 +299,7 @@
 SHL1STDLIBS += -lXp -lXext -lSM -lICE -lX11
 .ELSE
 .IF "$(USE_XINERAMA)" != "NO"
-.IF "$(XINERAMA_LINK)" == "dynamic"
+.IF "$(XINERAMA_LINK)" == "dynamic" || "$(OS)" == "MACOSX"
 SHL1STDLIBS += -lXinerama
 .ELSE
 SHL1STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
