--- com/sun/star/wizards/ui/FilterComponent.java	2008-12-01 05:31:27.000000000 -0800
+++ com/sun/star/wizards/ui/FilterComponent.java.bak	2010-02-10 20:21:26.000000000 -0800
@@ -697,7 +697,7 @@
                     {
                         nValue = 1.0;
                     }
-                    Helper.setUnoPropertyValue(UnoDialog.getModel(ControlElements[SOTXTVALUE]), "EffectiveValue", nValue);
+                    Helper.setUnoPropertyValue(UnoDialog.getModel(ControlElements[SOTXTVALUE]), "EffectiveValue", new Double(nValue));
                 }
                 else
                 {
