--- source/xmlsec/nss/seinitializer_nssimpl.cxx	2008-08-18 06:16:11.000000000 -0700
+++ source/xmlsec/nss/seinitializer_nssimpl.cxx	2014-04-28 15:10:44.000000000 -0700
@@ -66,7 +66,7 @@
 #include "nspr.h"
 #include "prtypes.h"
 #include "pk11func.h"
-#ifdef SYSTEM_MOZILLA
+#if defined SYSTEM_MOZILLA || SUPD == 310
 #include "nssrenam.h"
 #include "secmod.h"        
 #endif
--- source/xmlsec/nss/x509certificate_nssimpl.hxx	2008-04-10 17:39:15.000000000 -0700
+++ source/xmlsec/nss/x509certificate_nssimpl.hxx	2014-04-28 15:10:24.000000000 -0700
@@ -40,7 +40,7 @@
 #include "com/sun/star/uno/SecurityException.hpp"
 #include <com/sun/star/security/XCertificate.hpp>
 
-#ifdef SYSTEM_MOZILLA
+#if defined SYSTEM_MOZILLA || SUPD == 310
 #include "nssrenam.h"
 #endif
 #include "cert.h"
