Index: source/Common/OdfConverterLib/Makefile
===================================================================
--- source/Common/OdfConverterLib/Makefile	(revision 4259)
+++ source/Common/OdfConverterLib/Makefile	(working copy)
@@ -28,10 +28,12 @@
 all-local: OdfConverterLib.dll OdfConverterLibStatic.dll 
 OdfConverterLib.dll: $(SOURCE) Makefile $(OOX2ODF_RES) $(ODF2OOX_RES)
 	gmcs $(CSC_DEBUG) $(RESOURCE_FLAGS) -target:library -reference:../OdfZipUtils/OdfZipUtils.dll \
+        -r:System.Data \
 		-out:OdfConverterLib.dll $(SOURCE)
 
 OdfConverterLibStatic.dll: $(SOURCE) Makefile $(OOX2ODF_RES) $(ODF2OOX_RES)
 	gmcs $(CSC_DEBUG) $(RESOURCE_FLAGS) -target:library -reference:../OdfZipUtils/OdfZipUtilsStatic.dll \
+        -r:System.Data \
 		-out:OdfConverterLibStatic.dll $(SOURCE)
 
 clean-local:
Index: source/Common/OdfZipUtils/Makefile
===================================================================
--- source/Common/OdfZipUtils/Makefile	(revision 4259)
+++ source/Common/OdfZipUtils/Makefile	(working copy)
@@ -10,10 +10,12 @@
 
 OdfZipUtils.dll : $(SOURCE)
 	$(CSC) $(CSC_DEBUG) -target:library \
+         -r:System.Drawing \
 	     -out:OdfZipUtils.dll $(SOURCE)
 
 OdfZipUtilsStatic.dll : $(SOURCE)
 	$(CSC) $(CSC_DEBUG) -d:STATIC -target:library \
+         -r:System.Drawing \
 	     -out:OdfZipUtilsStatic.dll $(SOURCE)
 
 clean-local:
Index: source/Spreadsheet/Converter/OOXStyleCellPostProcessor.cs
===================================================================
--- source/Spreadsheet/Converter/OOXStyleCellPostProcessor.cs	(revision 4259)
+++ source/Spreadsheet/Converter/OOXStyleCellPostProcessor.cs	(working copy)
@@ -128,6 +128,8 @@
             else if (this.changeCellName)
             {
                 this.changeCellName = false;
+				// Fix bug described at http://bugzilla.neooffice.org/bug.php?op=show&bugid=2794&pos=5 using the fix described at https://bugzilla.novell.com/show_bug.cgi?id=332246
+				this.nextWriter.WriteEndAttribute();
             }
             else
             {
Index: source/Spreadsheet/Converter/resources/odf2oox/sharedStrings.xsl
===================================================================
--- source/Spreadsheet/Converter/resources/odf2oox/sharedStrings.xsl	(revision 4259)
+++ source/Spreadsheet/Converter/resources/odf2oox/sharedStrings.xsl	(working copy)
@@ -68,7 +68,8 @@
       select="document('content.xml')/office:document-content/office:body/office:spreadsheet/descendant::table:table-cell[text:p and not(@office:value-type='float') and (@office:value-type='string' or @office:value-type='boolean' or not((number(text:p) or text:p = 0 or contains(text:p,',') or contains(text:p,'%') or @office:value-type='currency' or @office:value-type='date' or @office:value-type='time')))]
       ">
       <si>
-        <pxsi:maxlength xmlns:pxsi="urn:cleverage:xmlns:post-processings:cellText">        
+        <!--Disable maxlength as it makes round trip ODF to XLSX to ODF break-->
+        <!--<pxsi:maxlength xmlns:pxsi="urn:cleverage:xmlns:post-processings:cellText">-->
           <xsl:choose>
             <xsl:when test="text:span|text:p/text:span">
               <xsl:apply-templates mode="run" select="text:p"/>
@@ -77,7 +78,7 @@
               <t xml:space="preserve"><xsl:apply-templates mode="text" select="text:p"/></t>
             </xsl:otherwise>
           </xsl:choose>
-        </pxsi:maxlength>
+        <!--</pxsi:maxlength>-->
       </si>
     </xsl:for-each>
   </xsl:template>
@@ -133,16 +134,16 @@
         <xsl:value-of select="."/>
       </xsl:variable>
       <!-- caution with 'Enters' because they can result with additional space in output text -->
-      <t xml:space="preserve"><xsl:choose>
+      <xsl:choose>
-        <xsl:when test="not(contains($value, '_x'))"><xsl:value-of select="$value"/></xsl:when>
+        <xsl:when test="not(contains($value, '_x'))"><t xml:space="preserve"><xsl:value-of select="$value"/></t></xsl:when>
         <xsl:otherwise>
           <xsl:call-template name="HexaDecimalValue">
             <xsl:with-param name="value">
-              <xsl:value-of select="$value"/>
+              <t xml:space="preserve"><xsl:value-of select="$value"/></t>
             </xsl:with-param>
           </xsl:call-template>
         </xsl:otherwise>
-      </xsl:choose></t>
+      </xsl:choose>
     </r>
   </xsl:template>
 
Index: source/Spreadsheet/Converter/resources/odf2oox/worksheets.xsl
===================================================================
--- source/Spreadsheet/Converter/resources/odf2oox/worksheets.xsl	(revision 4259)
+++ source/Spreadsheet/Converter/resources/odf2oox/worksheets.xsl	(working copy)
@@ -528,7 +528,7 @@
           <xsl:for-each select="style:table-properties">
             <xsl:if test="attribute::style:writing-mode='rl-tb'">
               <xsl:attribute name="rightToLeft">
-                <xsl:value-of select="1"> </xsl:value-of>
+                <xsl:value-of select="1"></xsl:value-of>
               </xsl:attribute>
             </xsl:if>
           </xsl:for-each>
Index: source/Spreadsheet/Converter/resources/oox2odf/insert_text.xsl
===================================================================
--- source/Spreadsheet/Converter/resources/oox2odf/insert_text.xsl	(revision 4259)
+++ source/Spreadsheet/Converter/resources/oox2odf/insert_text.xsl	(working copy)
@@ -587,11 +587,7 @@
 
       <xsl:when test="not(contains($string,'../')) and not(contains($string,'..\..\'))">
 
-        <xsl:value-of select="concat('../',$string)">
-          <!--xsl:call-template name="Change20PercentToSpace">
-              <xsl:with-param name="slash" select="concat('..\',substring-after($string,''))"/>
-              </xsl:call-template-->
-        </xsl:value-of>
+        <xsl:value-of select="concat('../',$string)"></xsl:value-of>
 
       </xsl:when>
 
