--- ./source/Shell/OdfConverter/OdfValidator.cs.bak	2009-01-04 15:43:51.000000000 -0800
+++ ./source/Shell/OdfConverter/OdfValidator.cs	2009-01-05 09:07:50.000000000 -0800
@@ -49,9 +49,9 @@
 	
 #if MONO
 	// Dummy validation class
-	public class OdfValidator
+	public class OdfValidator : IValidator
 	{
-		public OdfValidator(Report report)
+		public OdfValidator(ConversionReport report)
 		{ 
 		}
 		public void validate(String fileName)
--- ./source/Spreadsheet/Converter/OdfConditionalPostProcessor.cs.bak	2009-01-04 15:43:59.000000000 -0800
+++ ./source/Spreadsheet/Converter/OdfConditionalPostProcessor.cs	2009-01-05 09:00:43.000000000 -0800
@@ -39,8 +39,11 @@
 using CleverAge.OdfConverter.OdfConverterLib;
 using CleverAge.OdfConverter.OdfZipUtils;
 using System.Text.RegularExpressions;
+#if MONO
+#else
 using System.Drawing;
 using System.Windows.Forms;
+#endif
 
 
 
--- ./source/Spreadsheet/Converter/OdfSharedStringsPostProcessor.cs.bak	2009-01-04 15:43:58.000000000 -0800
+++ ./source/Spreadsheet/Converter/OdfSharedStringsPostProcessor.cs	2009-01-05 09:00:54.000000000 -0800
@@ -40,8 +40,11 @@
 using CleverAge.OdfConverter.OdfConverterLib;
 using CleverAge.OdfConverter.OdfZipUtils;
 using System.Text.RegularExpressions;
+#if MONO
+#else
 using System.Windows.Forms;
 using System.Drawing;
+#endif
 
 
 namespace CleverAge.OdfConverter.Spreadsheet
