--- source/Shell/OdfConverter/Makefile	2009-04-24 20:53:31.000000000 +0200
+++ source/Shell/OdfConverter/Makefile	2009-04-24 20:32:14.000000000 +0200
@@ -84,6 +84,7 @@ ifeq ($(OS),Darwin)
 		`pkg-config --libs-only-l mono` bundle.o
 else
 	$(CC) -o OdfConverter$(EXEEXT) -Wall `pkg-config --cflags mono` main.c \
+	    -Wl,--stack=20000000 \
 	    $(KILLAT) \
 	    bundle.o \
 	    $(EXPORTDYNAMIC) \
--- source/Spreadsheet/Converter/Makefile	2009-08-17 18:49:04.000000000 -0700
+++ source/Spreadsheet/Converter/Makefile	2009-08-17 18:51:05.000000000 -0700
@@ -30,12 +30,16 @@
 	$(CSC) $(CSC_DEBUG) -d:MONO $(RESOURCE_FLAGS) -target:library \
 		-reference:../../Common/OdfZipUtils/OdfZipUtils.dll \
 		-reference:../../Common/OdfConverterLib/OdfConverterLib.dll \
+		-reference:System.Drawing \
+		-reference:System.Windows.Forms \
 		-out:$@ $(SOURCE)
 
 SpreadsheetStatic.dll: $(SOURCE) Makefile $(OOX2ODF_RES) $(ODF2OOX_RES)
 	$(CSC) $(CSC_DEBUG) -d:MONO $(RESOURCE_FLAGS) -target:library \
 		-reference:../../Common/OdfZipUtils/OdfZipUtilsStatic.dll \
 		-reference:../../Common/OdfConverterLib/OdfConverterLibStatic.dll \
+		-reference:System.Drawing \
+		-reference:System.Windows.Forms \
 		-out:$@ $(SOURCE)
 
 clean-local:
