--- patches/dev300/apply	2010-02-09 19:27:56.000000000 -0800
+++ patches/dev300/apply	2010-02-09 19:34:48.000000000 -0800
@@ -651,6 +651,7 @@
 sc-standard-filter-options-ods-hack.diff, i#35579, jonp
 
 # Hides the filtered rows when height of rows are changed bxc #62161
+# Disable because it causes NeoOffice bug 3437
 # sc-filter-hide-filteredrows.diff, i#35581, jody
 
 # Autofill doesnt fill filtered rows bxc 62499
@@ -678,7 +679,8 @@
 sc-change-record-dnd.diff, i#44982, jody
 
 # keybindings: <ctrl> +/‑ to insert/delete a complete line in calc (X#62530)
-calc-new-acceleration.diff, i#67029, michael
+# Disable because it causes NeoOffice bug 2802
+# calc-new-acceleration.diff, i#67029, michael
 
 sc-dp-gridlayout.diff, i#68544 n#190970, jody
 
@@ -884,7 +886,8 @@
 
 [ Misc ]
 # Insert symbol defaults to the Symbol font - not a random one
-gui-sw-insert-symbol.diff, i#15512, michael
+# Disable because it causes NeoOffice bug 3187
+# gui-sw-insert-symbol.diff, i#15512, michael
 # Address data source ordering x#62860
 addrbk-datasrc-ordering.diff, michael
 # Changed the default option in the Templates and Documents dialog.
@@ -902,8 +905,9 @@
 desktop-cmdhelp-stdout.diff, kohei
 
 # always enable file save even when the document is not modified.
-always-allow-save-document-sfx2.diff, n#347423, n#450789, i#5226, kohei
-always-allow-save-document-sc.diff,   n#347423, n#450789, i#5226, kohei
+# Disable because we want to preserve the existing NeoOffice behavior
+# always-allow-save-document-sfx2.diff, n#347423, n#450789, i#5226, kohei
+# always-allow-save-document-sc.diff,   n#347423, n#450789, i#5226, kohei
 
 # Adjust scroll speed while extending selection beyoud visible sheet area.
 scroll-accel-sc.diff, n#375909, i#71362, kohei
@@ -927,7 +931,8 @@
 
 [ EasterEgg ]
 # go-oo.org people active in this release
-go-oo-team.diff, michael
+# Disable EasterEgg because it is unnecessary for NeoOffice
+# go-oo-team.diff, michael
 
 
 [ NovellOnly ]
@@ -1959,7 +1964,8 @@
 writerfilter-source-dmapper-domainmapper-debug.diff
 docx-import-fixes.diff
 
-buildfix-oox-depends-on-unotools.diff
+# Disable because we want NeoOffice to decide which filter to use
+# buildfix-oox-depends-on-unotools.diff
 
 oox-fix-placeholder-layout.diff, n#485316, n#480223, rodo
 oox-fix-list-style-apply.diff, n#485417, rodo
@@ -1996,10 +2002,12 @@
 # Make the formula separators changeable per locale setting, and add new 
 # configuration page for formula syntax and separators.
 
-calc-formula-variable-separators-sc.diff,        n#447164, i#92056, kohei
+# Disable because it causes NeoOffice bug 3394
+# calc-formula-variable-separators-sc.diff,        n#447164, i#92056, kohei
 
-calc-formula-variable-separators-svx.diff,       n#447164, i#92056, kohei
-calc-formula-variable-separators-officecfg.diff, n#447164, i#92056, kohei
+# Disable because it causes NeoOffice bug 3394
+# calc-formula-variable-separators-svx.diff,       n#447164, i#92056, kohei
+# calc-formula-variable-separators-officecfg.diff, n#447164, i#92056, kohei
 
 [ CalcFixes < dev300-m52 ]
 # Display references when entering input mode, even when the separators are not ';'.
@@ -2427,8 +2435,9 @@
 layout-simple-dialogs-sfx2.diff
 layout-simple-dialogs-solenv.diff
 layout-simple-dialogs-starmath.diff
-layout-simple-dialogs-svx.diff
-#layout-simple-dialogs-svx-no-gtk.diff
+# Fix build breakage by using layout-simple-dialogs-svx-no-gtk.diff instead of
+#layout-simple-dialogs-svx.diff
+layout-simple-dialogs-svx-no-gtk.diff
 layout-simple-dialogs-sw.diff
 layout-simple-dialogs-toolkit.diff
 layout-simple-dialogs-vcl.diff
@@ -2459,20 +2468,22 @@
 SectionIssue => i#62234
 
 # Correct accidental use of cAPS LOCK KEY!
-autocorrect-accidental-caps-lock-offapi.diff
-autocorrect-accidental-caps-lock-officecfg.diff
-autocorrect-accidental-caps-lock-sc.diff
-autocorrect-accidental-caps-lock-sw.diff
-autocorrect-accidental-caps-lock-svx.diff
-autocorrect-accidental-caps-lock-vcl.diff
+# Disable section because we want to preserve the existing NeoOffice behavior
+# autocorrect-accidental-caps-lock-offapi.diff
+# autocorrect-accidental-caps-lock-officecfg.diff
+# autocorrect-accidental-caps-lock-sc.diff
+# autocorrect-accidental-caps-lock-sw.diff
+# autocorrect-accidental-caps-lock-svx.diff
+# autocorrect-accidental-caps-lock-vcl.diff
 
 # Fancy document modified status window
-statusbar-fancy-modified-status-sc.diff,       i#103862, kohei
-statusbar-fancy-modified-status-sd.diff,       i#103862, kohei
-statusbar-fancy-modified-status-sfx2.diff,     i#103862, kohei
-statusbar-fancy-modified-status-starmath.diff, i#103862, kohei
-statusbar-fancy-modified-status-svx.diff,      i#103862, kohei
-statusbar-fancy-modified-status-sw.diff,       i#103862, kohei
+# Disable section because we want to preserve the existing NeoOffice behavior
+# statusbar-fancy-modified-status-sc.diff,       i#103862, kohei
+# statusbar-fancy-modified-status-sd.diff,       i#103862, kohei
+# statusbar-fancy-modified-status-sfx2.diff,     i#103862, kohei
+# statusbar-fancy-modified-status-starmath.diff, i#103862, kohei
+# statusbar-fancy-modified-status-svx.diff,      i#103862, kohei
+# statusbar-fancy-modified-status-sw.diff,       i#103862, kohei
 
 
 [ UnitTesting ]
@@ -2765,7 +2776,8 @@
 [ Fixes ]
 fix-ppt-linespacing-import-export.diff, n#355302, rodo
 sw-section-import-fix.diff, n#364533, freuter
-sw-allow-negative-spacing.diff, n#364534, freuter
+# Disable because we want to preserve the existing NeoOffice behavior
+# sw-allow-negative-spacing.diff, n#364534, freuter
 sw-tab-export-fix-i100264.diff, n#479068, freuter
 sw-section-header-export-problem.diff, i#100918, freuter
 oox-fix-abs-relations.diff, n#493528, freuter
@@ -3165,7 +3177,8 @@
 xlsx-export-set-xml-2007-flavour.diff, n#502090, janneke
 xlsx-export-simple-autofilter.diff, n#497559, janneke
 # hack to ignore writerfilter when odf-converter is present
-odf-converter-ignore-writerfilter.diff, n#348471, n#502173, jholesov
+# Disable because we want NeoOffice to decide which filter to use
+# odf-converter-ignore-writerfilter.diff, n#348471, n#502173, jholesov
 xlsx-export-skip-auto-format.diff, n#497560, janneke
 
 # fixes visibility to make it work with some compilers
