Index: patches/dev300/apply
===================================================================
--- patches/dev300/apply	(revision 9613)
+++ patches/dev300/apply	(working copy)
@@ -161,7 +161,8 @@
 linkoo-type.diff
 
 #Improve the behavior of input field dialog.
-sw-input-field-dialog.diff, Zhang Yun
+# Disable because it causes NeoOffice bug 2234
+# sw-input-field-dialog.diff, Zhang Yun
 
 # Improve the behavior of input fields inline editing.
 sw-input-field-inline-edit.diff, Zhang Yun