Index: patches/src680/apply
===================================================================
--- patches/src680/apply	(revision 9363)
+++ patches/src680/apply	(working copy)
@@ -596,7 +596,8 @@
 [ NotMacOSX ]
 
 # Make the delete key happy again in starcalc  - wontfix [!]
-gui-sc-delete.diff, i#17965, michael
+# Disable because it causes NeoOffice backspace and delete events to be backwards
+# gui-sc-delete.diff, i#17965, michael
 
 [ Misc ]
 # Insert symbol defaults to the Symbol font - not a random one
