Index: patches/dev300/apply
===================================================================
--- patches/dev300/apply	(revision 13546)
+++ patches/dev300/apply	(working copy)
@@ -23,7 +23,8 @@
 	      FedoraLinuxOnlyFixes, LinuxOnly, SystemBits, \
 	      msaccess, OpenGLTransitions, \
 	      Fpickers, Mono, AddressBooks, QuickStarter
-MacOSXCommon : Common, Defaults, TangoIcons, SystemBits
+# Add Binfilter to avoid build breakage in binfilter module
+MacOSXCommon : Common, Defaults, TangoIcons, SystemBits, Binfilter
 Win32Common : Common, Defaults, TangoIcons, \
 	      Win32Only, NotDebian
 # Experimental sections
@@ -418,7 +419,8 @@
 sd-builddep.diff, i#92662, michael
 
 # use basic (advanced) shapes for rectangle/ellipse on draw/impress's toolbars. make shapes wrap words by default, n#171052, rodo
-sd-toolbar-advanced-shapes.diff, n#171052, rodo
+# Disable because it causes NeoOffice bug 2668
+# sd-toolbar-advanced-shapes.diff, n#171052, rodo
 svx-shapes-default-word-wrap-enable.diff, n#171052, rodo
 svx-honour-customshape-capabilities.diff, n#395372, thorsten
 
@@ -458,7 +460,8 @@
 #FIXME src680: xmlsecurity-verify-usages.diff, i#76871, pmladek
 
 # fix that allows OOo to work with a standard (unpatched) libjpeg,
-jpegc.c.diff, i#80674, n#272574, flr
+# Disable because it causes NeoOffice bug 2613
+# jpegc.c.diff, i#80674, n#272574, flr
 
 # handle hyperlinks to files better
 sw-hyperlinks-relative.diff, n#405712, i#91522, jholesov
@@ -618,7 +621,8 @@
 sc-change-record-dnd.diff, i#44982, jody
 
 # keybindings: <ctrl> +/‑ to insert/delete a complete line in calc (X#62530)
-calc-new-acceleration.diff, i#67029, michael
+# Disable because it causes NeoOffice bug 2802
+# calc-new-acceleration.diff, i#67029, michael
 
 sc-dp-gridlayout.diff, i#68544 n#190970, jody
 
@@ -752,7 +756,8 @@
 
 [ Misc ]
 # Insert symbol defaults to the Symbol font - not a random one
-gui-sw-insert-symbol.diff, i#15512, michael
+# Disable because it causes NeoOffice bug 3187
+# gui-sw-insert-symbol.diff, i#15512, michael
 # Address data source ordering x#62860
 addrbk-datasrc-ordering.diff, michael
 # Changed the default option in the Templates and Documents dialog.
@@ -1555,7 +1560,8 @@
 sfx2-pre-and-postprocess-during-save-load.diff, i#71939, florian
 sfx2-pre-and-postprocess-crash-fix.diff, n#270544, rodo
 # hack to ignore writerfilter if odf-converter is present
-odf-converter-ignore-writerfilter.diff, n#348471, jholesov
+# Disable because we want to allow user to decide which filter to use
+# odf-converter-ignore-writerfilter.diff, n#348471, jholesov
 
 
 [ msaccess ]
@@ -1591,7 +1597,8 @@
 
 writerfilter-source-dmapper-domainmapper-debug.diff
 
-buildfix-oox-depends-on-unotools.diff
+# Disable because we want to allow user to decide which filter to use
+# buildfix-oox-depends-on-unotools.diff
 
 # fix build dependencies
 #FIXME still necessary? oox-token-deps.diff, pmladek
@@ -1877,10 +1884,6 @@
 
 # fix lpsolve build for OSX
 macosx-lpsolve-makefile.diff
-# fixes build with gcc-3.3; required for MacOSX build
-# it is usable only until anyone uses the disabled code => MacOSXOnly
-# we will move to gcc-4.x, so it does not make sense to push it upstream
-buildfix-boost-gcc33.diff
 
 [ PardusOnly ]
 pardus-default-ExternalMailer.diff
