Index: patches/dev300/apply
===================================================================
--- patches/dev300/apply	(revision 13546)
+++ patches/dev300/apply	(working copy)
@@ -418,7 +418,8 @@
 sd-builddep.diff, i#92662, michael
 
 # use basic (advanced) shapes for rectangle/ellipse on draw/impress's toolbars. make shapes wrap words by default, n#171052, rodo
-sd-toolbar-advanced-shapes.diff, n#171052, rodo
+# Disable because it causes NeoOffice bug 2668
+# sd-toolbar-advanced-shapes.diff, n#171052, rodo
 svx-shapes-default-word-wrap-enable.diff, n#171052, rodo
 svx-honour-customshape-capabilities.diff, n#395372, thorsten
 
@@ -458,7 +459,8 @@
 #FIXME src680: xmlsecurity-verify-usages.diff, i#76871, pmladek
 
 # fix that allows OOo to work with a standard (unpatched) libjpeg,
-jpegc.c.diff, i#80674, n#272574, flr
+# Disable because it causes NeoOffice bug 2613
+# jpegc.c.diff, i#80674, n#272574, flr
 
 # handle hyperlinks to files better
 sw-hyperlinks-relative.diff, n#405712, i#91522, jholesov
@@ -618,7 +620,8 @@
 sc-change-record-dnd.diff, i#44982, jody
 
 # keybindings: <ctrl> +/‑ to insert/delete a complete line in calc (X#62530)
-calc-new-acceleration.diff, i#67029, michael
+# Disable because it causes NeoOffice bug 2802
+# calc-new-acceleration.diff, i#67029, michael
 
 sc-dp-gridlayout.diff, i#68544 n#190970, jody
 
@@ -1877,10 +1880,6 @@
 
 # fix lpsolve build for OSX
 macosx-lpsolve-makefile.diff
-# fixes build with gcc-3.3; required for MacOSX build
-# it is usable only until anyone uses the disabled code => MacOSXOnly
-# we will move to gcc-4.x, so it does not make sense to push it upstream
-buildfix-boost-gcc33.diff
 
 [ PardusOnly ]
 pardus-default-ExternalMailer.diff
