--- configure.in	2007-01-03 21:56:48.000000000 -0800
+++ configure.in	2007-01-03 22:00:09.000000000 -0800
@@ -1283,6 +1283,7 @@
 AC_SUBST(GCCVER)
 
 if test "$GCC" = "yes"; then
+if test "$_os" != "Darwin"; then
    AC_MSG_CHECKING([whether $CC supports -fvisibility=hidden])
    save_CFLAGS=$CFLAGS
    CFLAGS="$CFLAGS -fvisibility=hidden"
@@ -1294,6 +1295,7 @@
       AC_MSG_RESULT([no])
    fi
 fi
+fi
 AC_SUBST(HAVE_GCC_VISIBILITY_FEATURE)
 
 if test "$GCC" = "yes"; then
