--- source/ui/dlg/sdtreelb.cxx	2007-01-21 08:25:30.000000000 -0800
+++ source/ui/dlg/sdtreelb.cxx	2007-01-21 08:56:34.000000000 -0800
@@ -349,17 +349,27 @@
 	SdPage* 	 pPage = NULL;
 	SvLBoxEntry* pEntry = NULL;
 
-	Image aImgPage( BitmapEx( SdResId( BMP_PAGE ) ) );
-	Image aImgPageExcl( BitmapEx( SdResId( BMP_PAGE_EXCLUDED ) ) );
-	Image aImgPageObjsExcl( BitmapEx( SdResId( BMP_PAGEOBJS_EXCLUDED ) ) );
-	Image aImgPageObjs( BitmapEx( SdResId( BMP_PAGEOBJS ) ) );
-	Image aImgObjects( BitmapEx( SdResId( BMP_OBJECTS ) ) );
-
-	Image aImgPageH( BitmapEx( SdResId( BMP_PAGE_H ) ) );
-	Image aImgPageExclH( BitmapEx( SdResId( BMP_PAGE_EXCLUDED_H ) ) );
-	Image aImgPageObjsExclH( BitmapEx( SdResId( BMP_PAGEOBJS_EXCLUDED_H ) ) );
-	Image aImgPageObjsH( BitmapEx( SdResId( BMP_PAGEOBJS_H ) ) );
-	Image aImgObjectsH( BitmapEx( SdResId( BMP_OBJECTS_H ) ) );
+	BitmapEx aBmpPage( SdResId( BMP_PAGE ) );
+	BitmapEx aBmpPageExcl( SdResId( BMP_PAGE_EXCLUDED ) );
+	BitmapEx aBmpPageObjsExcl( SdResId( BMP_PAGEOBJS_EXCLUDED ) );
+	BitmapEx aBmpPageObjs( SdResId( BMP_PAGEOBJS ) );
+	BitmapEx aBmpObjects( SdResId( BMP_OBJECTS ) );
+	Image aImgPage( aBmpPage );
+	Image aImgPageExcl( aBmpPageExcl );
+	Image aImgPageObjsExcl( aBmpPageObjsExcl );
+	Image aImgPageObjs( aBmpPageObjs );
+	Image aImgObjects( aBmpObjects );
+
+	BitmapEx aBmpPageH( SdResId( BMP_PAGE_H ) );
+	BitmapEx aBmpPageExclH( SdResId( BMP_PAGE_EXCLUDED_H ) );
+	BitmapEx aBmpPageObjsExclH( SdResId( BMP_PAGEOBJS_EXCLUDED_H ) );
+	BitmapEx aBmpPageObjsH( SdResId( BMP_PAGEOBJS_H ) );
+	BitmapEx aBmpObjectsH( SdResId( BMP_OBJECTS_H ) );
+	Image aImgPageH( aBmpPageH );
+	Image aImgPageExclH( aBmpPageExclH );
+	Image aImgPageObjsExclH( aBmpPageObjsExclH );
+	Image aImgPageObjsH( aBmpPageObjsH );
+	Image aImgObjectsH( aBmpObjectsH );
 
 	// Zuerst alle Pages incl. Objekte einfuegen
 	USHORT nPage = 0;
@@ -514,10 +524,14 @@
 	SvLBoxEntry* pFileEntry = NULL;
 	SvLBoxEntry* pPageEntry = NULL;
 
-	Image aImgDocOpen( BitmapEx( SdResId( BMP_DOC_OPEN ) ) );
-	Image aImgDocClosed( BitmapEx( SdResId( BMP_DOC_CLOSED ) ) );
-	Image aImgDocOpenH( BitmapEx( SdResId( BMP_DOC_OPEN_H ) ) );
-	Image aImgDocClosedH( BitmapEx( SdResId( BMP_DOC_CLOSED_H ) ) );
+	BitmapEx aBmpDocOpen( SdResId( BMP_DOC_OPEN ) );
+	BitmapEx aBmpDocClosed( SdResId( BMP_DOC_CLOSED ) );
+	BitmapEx aBmpDocOpenH( SdResId( BMP_DOC_OPEN_H ) );
+	BitmapEx aBmpDocClosedH( SdResId( BMP_DOC_CLOSED_H ) );
+	Image aImgDocOpen( aBmpDocOpen );
+	Image aImgDocClosed( aBmpDocClosed );
+	Image aImgDocOpenH( aBmpDocOpenH );
+	Image aImgDocClosedH( aBmpDocClosedH );
 
 	// Dokumentnamen einfuegen
 	pFileEntry = InsertEntry( aDocName,
@@ -713,12 +727,18 @@
 			SdPage* 	 pPage = NULL;
 			SvLBoxEntry* pPageEntry = NULL;
 
-			Image aImgPage( BitmapEx( SdResId( BMP_PAGE ) ) );
-			Image aImgPageObjs( BitmapEx( SdResId( BMP_PAGEOBJS ) ) );
-			Image aImgObjects( BitmapEx( SdResId( BMP_OBJECTS ) ) );
-			Image aImgPageH( BitmapEx( SdResId( BMP_PAGE_H ) ) );
-			Image aImgPageObjsH( BitmapEx( SdResId( BMP_PAGEOBJS_H ) ) );
-			Image aImgObjectsH( BitmapEx( SdResId( BMP_OBJECTS_H ) ) );
+			BitmapEx aBmpPage( SdResId( BMP_PAGE ) );
+			BitmapEx aBmpPageObjs( SdResId( BMP_PAGEOBJS ) );
+			BitmapEx aBmpObjects( SdResId( BMP_OBJECTS ) );
+			BitmapEx aBmpPageH( SdResId( BMP_PAGE_H ) );
+			BitmapEx aBmpPageObjsH( SdResId( BMP_PAGEOBJS_H ) );
+			BitmapEx aBmpObjectsH( SdResId( BMP_OBJECTS_H ) );
+			Image aImgPage( aBmpPage );
+			Image aImgPageObjs( aBmpPageObjs );
+			Image aImgObjects( aBmpObjects );
+			Image aImgPageH( aBmpPageH );
+			Image aImgPageObjsH( aBmpPageObjsH );
+			Image aImgObjectsH( aBmpObjectsH );
 
 			// Dokumentname ist schon eingefuegt
 
