Index: source/dialog/cfg.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/cfg.cxx,v
retrieving revision 1.38
diff -u -r1.38 cfg.cxx
--- source/dialog/cfg.cxx	19 Dec 2006 17:45:00 -0000	1.38
+++ source/dialog/cfg.cxx	22 Jun 2007 17:05:48 -0000
@@ -5457,7 +5457,8 @@
 			// added to the list
 		}
 
-		if ( graphics.getLength() > 0 )
+		// Fix bug 2310 by explicitly checking if the graphics has NULL content
+		if ( graphics.getLength() > 0 && graphics[ 0 ].get() )
 		{
 			Image img = Image( graphics[ 0 ] );
 			aTbSymbol.InsertItem( nId, img, pConstIter->first );
