--- bin/unpack	2009-08-05 05:20:53.000000000 -0700
+++ bin/unpack	2011-08-14 11:32:37.000000000 -0700
@@ -229,8 +229,8 @@
 		echo "	unpacking $pkg..."
 		($OOO_DECOMPRESS_CMD "$SRCDIR/$OOO_SPLIT_PREFIX$pkg.tar.bz2" | $GNUTAR xpf - ) || exit 1
 		echo "	moving it into place..."
-		mv $OOO_SPLIT_PREFIX$pkg/* $OOBUILDDIR
-		rmdir $OOO_SPLIT_PREFIX$pkg
+		( cd $OOO_SPLIT_PREFIX$pkg && $GNUTAR cfp - * ) | ( cd $OOBUILDDIR && cd $GNUTAR xfp - )
+		rm -rf $OOO_SPLIT_PREFIX$pkg
 	done
 	if test "z$PIECE" != "z"; then
 		ln -s "$BUILDDIR/$CVSTAG-$PIECE" $OOBUILDDIR
