--- hsqldb.org/src/org/hsqldb/Parser.java	2007/07/23 12:48:09	1.140
+++ hsqldb.org/src/org/hsqldb/Parser.java	2007/08/10 13:33:19	1.141
@@ -446,7 +446,8 @@
         if (token.equals(Token.T_INTO)) {
             boolean getname = true;
 
-            token = tokenizer.getString();
+            token           = tokenizer.getString();
+            select.intoType = database.getDefaultTableType();
 
             if (tokenizer.wasSimpleToken()) {
                 switch (Token.get(token)) {
@@ -468,20 +469,19 @@
                         break;
 
                     default :
-                        select.intoType = database.getDefaultTableType();
-                        getname         = false;
+                        getname = false;
                         break;
                 }
 
                 if (getname) {
                     token = tokenizer.getName();
-                } else {
-                    if (!tokenizer.wasName()) {
-                        tokenizer.throwUnexpected();
-                    }
                 }
             }
 
+            if (!tokenizer.wasName()) {
+                tokenizer.throwUnexpected();
+            }
+
             select.sIntoTable = database.nameManager.newHsqlName(token,
                     tokenizer.wasQuotedIdentifier());
             select.sIntoTable.schema =
