--- Library_avmediaQuickTime.mk	2015-11-25 04:55:38.000000000 -0800
+++ Library_avmediaQuickTime.mk	2017-02-18 08:03:25.000000000 -0800
@@ -35,6 +35,7 @@
 	QTKit \
 ))
 
+ifneq ($(shell if test $(MACOSX_SDK_VERSION) -lt 101200; then echo $(MACOSX_SDK_VERSION); fi),)
 $(eval $(call gb_Library_add_objcxxobjects,avmediaQuickTime,\
 	avmedia/source/quicktime/framegrabber \
 	avmedia/source/quicktime/manager \
@@ -42,5 +43,10 @@
 	avmedia/source/quicktime/quicktimeuno \
 	avmedia/source/quicktime/window \
 ))
+else	# MACOSX_SDK_VERSION -lt 101200
+$(eval $(call gb_Library_add_objcxxobjects,avmediaQuickTime,\
+	avmedia/source/quicktime/quicktimeuno \
+))
+endif	# MACOSX_SDK_VERSION -lt 101200
 
 # vim: set noet sw=4 ts=4:
--- source/quicktime/quicktimecommon.hxx	2015-11-25 04:55:38.000000000 -0800
+++ source/quicktime/quicktimecommon.hxx	2017-02-18 08:15:18.000000000 -0800
@@ -23,7 +23,9 @@
 #ifdef MACOSX
 #include <premac.h>
 #import <Cocoa/Cocoa.h>
+#if MACOSX_SDK_VERSION < 101200
 #import <QTKit/QTKit.h>
+#endif	// MACOSX_SDK_VERSION < 101200
 #include <postmac.h>
 #endif
 #include <osl/mutex.hxx>
--- source/quicktime/quicktimeuno.mm	2015-11-25 04:55:38.000000000 -0800
+++ source/quicktime/quicktimeuno.mm	2017-02-18 08:14:28.000000000 -0800
@@ -17,7 +17,9 @@
  *   the License at http://www.apache.org/licenses/LICENSE-2.0 .
  */
 
+#if MACOSX_SDK_VERSION < 101200
 #include "quicktimecommon.hxx"
+#endif	// MACOSX_SDK_VERSION < 101200
 #include "manager.hxx"
 
 using namespace ::com::sun::star;
@@ -28,7 +30,11 @@
 
 static uno::Reference< uno::XInterface > SAL_CALL create_MediaPlayer( const uno::Reference< lang::XMultiServiceFactory >& rxFact )
 {
+#if MACOSX_SDK_VERSION < 101200
     return uno::Reference< uno::XInterface >( *new ::avmedia::quicktime::Manager( rxFact ) );
+#else	// MACOSX_SDK_VERSION < 101200
+    return NULL;
+#endif	// MACOSX_SDK_VERSION < 101200
 }
 
 extern "C" SAL_DLLPUBLIC_EXPORT void* SAL_CALL avmediaQuickTime_component_getFactory( const sal_Char* pImplName, void* pServiceManager, void* /* pRegistryKey */ )
