--- source/commontools/dbexception.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/commontools/dbexception.cxx	2021-05-09 13:33:47.000000000 -0400
@@ -79,13 +79,6 @@
 }
 
 
-SQLExceptionInfo::SQLExceptionInfo(const SQLExceptionInfo& _rCopySource)
-    :m_aContent(_rCopySource.m_aContent)
-    ,m_eType(_rCopySource.m_eType)
-{
-}
-
-
 const SQLExceptionInfo& SQLExceptionInfo::operator=(const ::com::sun::star::sdbc::SQLException& _rError)
 {
     m_aContent <<= _rError;
