--- source/migration/migration_impl.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/migration/migration_impl.hxx	2021-05-11 16:51:57.000000000 -0400
@@ -104,16 +104,6 @@
         m_xPopupMenu      = xPopupMenu;
     }
 
-    MigrationItem& operator=(const MigrationItem& aMigrationItem)
-    {
-        m_sParentNodeName = aMigrationItem.m_sParentNodeName;
-        m_sPrevSibling    = aMigrationItem.m_sPrevSibling;
-        m_sCommandURL     = aMigrationItem.m_sCommandURL;
-        m_xPopupMenu      = aMigrationItem.m_xPopupMenu;
-
-        return *this;
-    }
-
     bool operator==(const MigrationItem& aMigrationItem)
     {
         return ( aMigrationItem.m_sParentNodeName == m_sParentNodeName &&
