--- download.lst	2015-11-25 04:55:38.000000000 -0800
+++ download.lst	2017-06-25 18:17:09.000000000 -0700
@@ -36,7 +36,8 @@
 export EPM_TARBALL := 3ade8cfe7e59ca8e65052644fed9fca4-epm-3.7.tar.gz
 export ETONYEK_MD5SUM := 805f941b06448212a988cb65f0691a7a
 export ETONYEK_TARBALL := libetonyek-0.1.1.tar.bz2
-export EXPAT_TARBALL := dd7dab7a5fea97d2a6a43f511449b7cd-expat-2.1.0.tar.gz
+export EXPAT_SHA256SUM := 1868cadae4c82a018e361e2b2091de103cd820aaacb0d6cfa49bd2cd83978885
+export EXPAT_TARBALL := expat-2.2.1.tar.bz2
 export FIREBIRD_MD5SUM := 21154d2004e025c8a3666625b0357bb5
 export FIREBIRD_TARBALL := Firebird-2.5.2.26540-0.tar.bz2
 # FIREBIRD_MD5SUM := b259c2d1c60a03bd104108405ae990a7
