--- hsqldb/UnpackedTarball_hsqldb.mk	2015-11-25 04:55:38.000000000 -0800
+++ hsqldb/UnpackedTarball_hsqldb.mk	2016-11-13 13:52:01.000000000 -0800
@@ -27,6 +27,7 @@
 	$(if $(HSQLDB_USE_JDBC_4_1),\
 		external/hsqldb/patches/jdbc-4.1.patch \
 	) \
+	external/hsqldb/patches/hsqldb.patch \
 ))
 
 # vim: set noet sw=4 ts=4:
--- redland/ExternalProject_raptor.mk	2015-11-25 04:55:38.000000000 -0800
+++ redland/ExternalProject_raptor.mk	2017-02-25 15:18:56.000000000 -0800
@@ -45,6 +45,7 @@
 			 --enable-parsers="rdfxml ntriples turtle trig guess rss-tag-soup" \
 			--with-www=xml \
 			--without-xslt-config \
+			$(if $(filter MACOSX,$(OS)),--with-xml2-config=/usr/bin/xml2-config) \
 			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM)) \
 			$(if $(filter MACOSX,$(OS)),--prefix=/@.__________________________________________________OOO) \
 			$(if $(filter IOS ANDROID,$(OS)),--disable-shared,--disable-static) \
--- opencollada/patches/opencollada.libxml.patch.0	2015-11-25 05:55:38.000000000 -0700
+++ opencollada/patches/opencollada.libxml.patch.0	2018-10-15 19:04:44.000000000 -0600
@@ -10,3 +10,14 @@
  #include <string>
  
  #include "COLLADABUPlatform.h"
+--- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:32:01.000000000 -0600
++++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:59:21.000000000 -0600
+@@ -9,7 +9,7 @@
+ */
+ 
+ #include "GeneratedSaxParserUtils.h"
+-#include <math.h>
++#include <cmath>
+ #include <memory>
+ #include <string.h>
+ #include <limits>
--- libvisio/UnpackedTarball_libvisio.mk	2015-11-25 04:55:38.000000000 -0800
+++ libvisio/UnpackedTarball_libvisio.mk	2019-08-07 19:37:41.000000000 -0700
@@ -13,6 +13,7 @@
 
 $(eval $(call gb_UnpackedTarball_add_patches,libvisio,\
        external/libvisio/vsd-msvc-max.patch.1 \
+       external/libvisio/libvisio.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
--- libabw/UnpackedTarball_libabw.mk	2015-11-25 04:55:38.000000000 -0800
+++ libabw/UnpackedTarball_libabw.mk	2019-08-16 10:44:25.000000000 -0700
@@ -11,4 +11,8 @@
 
 $(eval $(call gb_UnpackedTarball_set_tarball,libabw,$(ABW_TARBALL)))
 
+$(eval $(call gb_UnpackedTarball_add_patches,libabw,\
+       external/libabw/libabw.patch.1 \
+))
+
 # vim: set noet sw=4 ts=4:
--- expat/ExternalProject_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/ExternalProject_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -15,8 +15,9 @@
 
 $(call gb_ExternalProject_get_state_target,expat,configure) :
 	$(call gb_ExternalProject_run,configure,\
-		./configure \
-			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM)) \
+		./configure --without-docbook \
+			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM) \
+			$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
 	,,expat_configure.log)
 
 # vim: set noet sw=4 ts=4:
--- expat/Module_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/Module_expat.mk	2018-05-09 03:27:29.000000000 -0700
@@ -14,14 +14,7 @@
 	StaticLibrary_expat \
 ))
 
-# Actually it wasn't that clear if we should
-# run configure step on mingw or not:
-# CONFIGURATION_ACTION wasn't set, but
-# CROSS_COMPILE case was handled in original makefile
-
-# tested it on mingw and it doesn't hurt ;-)
-#ifneq ($(OS),WNT)
-ifneq ($(OS)$(COM),WNTMSC)
+ifneq ($(OS),WNT)
 $(eval $(call gb_Module_add_targets,expat,\
 	ExternalProject_expat \
 ))
--- expat/README	2015-11-25 04:55:38.000000000 -0800
+++ expat/README	2018-05-09 03:27:29.000000000 -0700
@@ -1,4 +1,4 @@
 Simple SAX parser library with added UTF-16 support.
 
 From:
-[http://expat.sourceforge.net/]
+[https://github.com/libexpat/libexpat]
--- expat/StaticLibrary_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/StaticLibrary_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -9,13 +9,13 @@
 
 $(eval $(call gb_StaticLibrary_StaticLibrary,expat))
 
-$(eval $(call gb_StaticLibrary_set_warnings_not_errors,expat))
+$(eval $(call gb_StaticLibrary_set_warnings_disabled,expat))
 
 $(eval $(call gb_StaticLibrary_use_unpacked,expat,expat))
 
 # no configure step on windows, no dependency
-ifneq ($(OS)$(COM),WNTMSC)
-$(eval $(call gb_StaticLibrary_use_external_project,expat,expat))
+ifneq ($(OS),WNT)
+$(eval $(call gb_StaticLibrary_use_external_project,expat,expat,full))
 endif
 
 $(eval $(call gb_StaticLibrary_set_include,expat,\
@@ -42,11 +42,6 @@
 ))
 endif
 
-# suppress warning spam
-$(eval $(call gb_StaticLibrary_add_cflags,expat,\
-	-w \
-))
-
 $(eval $(call gb_StaticLibrary_add_generated_cobjects,expat,\
 	UnpackedTarball/expat/lib/xmlparse \
 	UnpackedTarball/expat/lib/xmlrole \
--- expat/StaticLibrary_expat_x64.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/StaticLibrary_expat_x64.mk	2019-09-22 19:18:51.000000000 -0700
@@ -11,7 +11,7 @@
 
 $(eval $(call gb_StaticLibrary_set_x64,expat_x64,YES))
 
-$(eval $(call gb_StaticLibrary_set_warnings_not_errors,expat_x64))
+$(eval $(call gb_StaticLibrary_set_warnings_disabled,expat_x64))
 
 $(eval $(call gb_StaticLibrary_use_unpacked,expat_x64,expat))
 
--- expat/UnpackedTarball_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/UnpackedTarball_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -11,8 +11,10 @@
 
 $(eval $(call gb_UnpackedTarball_set_tarball,expat,$(EXPAT_TARBALL)))
 
+$(eval $(call gb_UnpackedTarball_update_autoconf_configs,expat,conftools))
+
 $(eval $(call gb_UnpackedTarball_add_patches,expat,\
-	external/expat/expat-2.1.0.patch \
+	external/expat/expat-winapi.patch \
 ))
 
 # This is a bit hackish.
--- expat/expat-winapi.patch	2015-11-25 04:55:38.000000000 -0800
+++ 
@@ -1,13 +1,13 @@
 --- misc/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:17.375000000 +0000
 +++ misc/build/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:34.703125000 +0000
-@@ -7,10 +7,6 @@
+@@ -81,10 +81,6 @@
+ #  ifndef XML_BUILDING_EXPAT
+ /* using Expat from an application */
  
- /* External API definitions */
- 
--#if defined(_MSC_EXTENSIONS) && !defined(__BEOS__) && !defined(__CYGWIN__)
--#define XML_USE_MSC_EXTENSIONS 1
--#endif
+-#    if defined(_MSC_EXTENSIONS) && ! defined(__BEOS__) && ! defined(__CYGWIN__)
+-#      define XMLIMPORT __declspec(dllimport)
+-#    endif
 -
- /* Expat tries very hard to make the API boundary very specifically
-    defined.  There are two macros defined to control this boundary;
-    each of these can be defined before including this header to
+ #  endif
+ #endif /* not defined XML_STATIC */
+ 
--- Module_external.mk	2015-11-25 07:55:38.000000000 -0500
+++ Module_external.mk	2020-12-18 15:40:25.000000000 -0500
@@ -64,6 +64,7 @@
 	$(call gb_Helper_optional,LIBXML2,libxml2) \
 	$(call gb_Helper_optional,LIBXSLT,libxslt) \
 	$(call gb_Helper_optional,LPSOLVE,lpsolve) \
+	$(call gb_Helper_optional,LIBTOMMATH,libtommath) \
 	$(call gb_Helper_optional,MARIADB,libmariadb) \
 	$(call gb_Helper_optional,MDDS,mdds) \
 	$(if $(WITH_EXTRA_EXTENSIONS),misc_extensions) \
