--- hsqldb/UnpackedTarball_hsqldb.mk	2018-05-09 05:27:29.000000000 -0500
+++ hsqldb/UnpackedTarball_hsqldb.mk	2018-12-30 15:38:46.000000000 -0600
@@ -27,6 +27,7 @@
 	$(if $(HSQLDB_USE_JDBC_4_1),\
 		external/hsqldb/patches/jdbc-4.1.patch \
 	) \
+	external/hsqldb/patches/hsqldb.patch \
 ))
 
 # vim: set noet sw=4 ts=4:
--- redland/ExternalProject_raptor.mk	2018-05-09 05:27:29.000000000 -0500
+++ redland/ExternalProject_raptor.mk	2018-12-30 15:35:04.000000000 -0600
@@ -28,6 +28,7 @@
 			 --enable-parsers="rdfxml ntriples turtle trig guess rss-tag-soup" \
 			--with-www=xml \
 			--without-xslt-config \
+			$(if $(filter MACOSX,$(OS)),--with-xml2-config=/usr/bin/xml2-config) \
 			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM) \
 			$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
 			$(if $(filter MACOSX,$(OS)),--prefix=/@.__________________________________________________OOO) \
--- opencollada/patches/opencollada.libxml.patch.0	2015-11-25 05:55:38.000000000 -0700
+++ opencollada/patches/opencollada.libxml.patch.0	2018-10-15 19:04:44.000000000 -0600
@@ -10,3 +10,14 @@
  #include <string>
  
  #include "COLLADABUPlatform.h"
+--- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:32:01.000000000 -0600
++++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:59:21.000000000 -0600
+@@ -9,7 +9,7 @@
+ */
+ 
+ #include "GeneratedSaxParserUtils.h"
+-#include <math.h>
++#include <cmath>
+ #include <memory>
+ #include <string.h>
+ #include <limits>
--- curl/curl-osx.patch.1	2018-05-09 05:27:29.000000000 -0500
+++ curl/curl-osx.patch.1	2018-12-30 18:53:54.000000000 -0600
@@ -283,3 +283,14 @@
  /* Define to 1 if compiler supports C99 variadic macro style. */
  #cmakedefine HAVE_VARIADIC_MACROS_C99 1
  
+--- a/acinclude.m4	2016-12-19 01:27:56.000000000 -0600
++++ b/acinclude.m4	2018-12-30 18:52:22.000000000 -0600
+@@ -3224,7 +3224,7 @@
+     AC_MSG_CHECKING([for *version-min in CFLAGS])
+     min=""
+     if test -z "$(echo $CFLAGS | grep m.*os.*-version-min)"; then
+-      min="-mmacosx-version-min=10.8"
++      min="-mmacosx-version-min=10.14"
+       CFLAGS="$CFLAGS $min"
+     fi
+     if test -z "$min"; then
--- curl/UnpackedTarball_curl.mk	2018-05-09 05:27:29.000000000 -0500
+++ curl/UnpackedTarball_curl.mk	2018-12-30 19:12:02.000000000 -0600
@@ -35,11 +35,9 @@
 endif
 
 ifeq ($(OS),MACOSX)
-ifneq ($(filter 1090 101000,$(MAC_OS_X_VERSION_MIN_REQUIRED)),)
 $(eval $(call gb_UnpackedTarball_add_patches,curl,\
 	external/curl/curl-osx.patch.1 \
 ))
 endif
-endif
 
 # vim: set noet sw=4 ts=4:
--- opencollada/patches/libc++.patch.0	2018-05-09 05:27:29.000000000 -0500
+++ opencollada/patches/libc++.patch.0	2018-12-30 19:58:11.000000000 -0600
@@ -9,13 +9,3 @@
      #include <unordered_map>
      #include <unordered_set>
      #define COLLADABU_HASH_MAP std::unordered_map
---- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
-+++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp
-@@ -10,6 +10,7 @@
- 
- #include "GeneratedSaxParserUtils.h"
- #include <math.h>
-+#include <cmath>
- #include <memory>
- #include <string.h>
- #include <limits>
--- hunspell/ExternalProject_hunspell.mk	2018-05-09 05:27:29.000000000 -0500
+++ hunspell/ExternalProject_hunspell.mk	2018-12-31 07:34:38.000000000 -0600
@@ -27,6 +27,7 @@
 		LIBS="$(gb_STDLIBS) $(LIBS)" \
 		autoreconf && \
 		$(SHELL) ./configure --disable-shared --disable-nls --with-pic \
+			--disable-dependency-tracking \
 			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM))\
 			$(if $(filter AIX,$(OS)),CFLAGS="-D_LINUX_SOURCE_COMPAT") \
 			$(if $(hunspell_CPPFLAGS),CPPFLAGS='$(hunspell_CPPFLAGS)') \
