--- hsqldb/UnpackedTarball_hsqldb.mk	2015-11-25 04:55:38.000000000 -0800
+++ hsqldb/UnpackedTarball_hsqldb.mk	2016-11-13 13:52:01.000000000 -0800
@@ -27,6 +27,7 @@
 	$(if $(HSQLDB_USE_JDBC_4_1),\
 		external/hsqldb/patches/jdbc-4.1.patch \
 	) \
+	external/hsqldb/patches/hsqldb.patch \
 ))
 
 # vim: set noet sw=4 ts=4:
--- redland/ExternalProject_raptor.mk	2015-11-25 04:55:38.000000000 -0800
+++ redland/ExternalProject_raptor.mk	2017-02-25 15:18:56.000000000 -0800
@@ -45,6 +45,7 @@
 			 --enable-parsers="rdfxml ntriples turtle trig guess rss-tag-soup" \
 			--with-www=xml \
 			--without-xslt-config \
+			$(if $(filter MACOSX,$(OS)),--with-xml2-config=/usr/bin/xml2-config) \
 			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM)) \
 			$(if $(filter MACOSX,$(OS)),--prefix=/@.__________________________________________________OOO) \
 			$(if $(filter IOS ANDROID,$(OS)),--disable-shared,--disable-static) \
--- opencollada/patches/opencollada.libxml.patch.0	2015-11-25 05:55:38.000000000 -0700
+++ opencollada/patches/opencollada.libxml.patch.0	2018-10-15 19:04:44.000000000 -0600
@@ -10,3 +10,14 @@
  #include <string>
  
  #include "COLLADABUPlatform.h"
+--- GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:32:01.000000000 -0600
++++ GeneratedSaxParser/src/GeneratedSaxParserUtils.cpp	2018-10-15 18:59:21.000000000 -0600
+@@ -9,7 +9,7 @@
+ */
+ 
+ #include "GeneratedSaxParserUtils.h"
+-#include <math.h>
++#include <cmath>
+ #include <memory>
+ #include <string.h>
+ #include <limits>
--- firebird/firebird-macosx.patch.1	2015-11-25 05:55:38.000000000 -0700
+++ firebird/firebird-macosx.patch.1	2018-10-15 21:40:42.000000000 -0600
@@ -109,3 +109,19 @@
  EMBED_UTIL_TARGETS=gstat gds_relay gsec fbguard nbackup fb_lock_print fbsvcmgr fbtracemgr
  CLIENT_UTIL_TARGETS=gds_relay gstat gsec fbguard fbmgr_bin nbackup fb_lock_print fbsvcmgr \
  	fbtracemgr
+--- firebird/src/common/classes/FpeControl.h	2013-03-19 02:38:21.000000000 -0600
++++ firebird/src/common/classes/FpeControl.h	2018-10-15 21:31:47.000000000 -0600
+@@ -208,11 +208,13 @@
+ }
+ #else
+ #ifndef isinf
++#ifndef __APPLE__
+ template <typename F>
+ inline bool isinf(F x)
+ {
+ 	return !isnan(x) && isnan(x - x);
+ }
++#endif // __APPLE__
+ #endif // isinf
+ #endif // WIN_NT
+ 
--- libvisio/UnpackedTarball_libvisio.mk	2015-11-25 04:55:38.000000000 -0800
+++ libvisio/UnpackedTarball_libvisio.mk	2019-08-07 19:37:41.000000000 -0700
@@ -13,6 +13,7 @@
 
 $(eval $(call gb_UnpackedTarball_add_patches,libvisio,\
        external/libvisio/vsd-msvc-max.patch.1 \
+       external/libvisio/libvisio.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
--- libabw/UnpackedTarball_libabw.mk	2015-11-25 04:55:38.000000000 -0800
+++ libabw/UnpackedTarball_libabw.mk	2019-08-16 10:44:25.000000000 -0700
@@ -11,4 +11,8 @@
 
 $(eval $(call gb_UnpackedTarball_set_tarball,libabw,$(ABW_TARBALL)))
 
+$(eval $(call gb_UnpackedTarball_add_patches,libabw,\
+       external/libabw/libabw.patch.1 \
+))
+
 # vim: set noet sw=4 ts=4:
--- expat/ExternalProject_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/ExternalProject_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -15,8 +15,9 @@
 
 $(call gb_ExternalProject_get_state_target,expat,configure) :
 	$(call gb_ExternalProject_run,configure,\
-		./configure \
-			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM)) \
+		./configure --without-docbook \
+			$(if $(CROSS_COMPILING),--build=$(BUILD_PLATFORM) --host=$(HOST_PLATFORM) \
+			$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
 	,,expat_configure.log)
 
 # vim: set noet sw=4 ts=4:
--- expat/Module_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/Module_expat.mk	2018-05-09 03:27:29.000000000 -0700
@@ -14,14 +14,7 @@
 	StaticLibrary_expat \
 ))
 
-# Actually it wasn't that clear if we should
-# run configure step on mingw or not:
-# CONFIGURATION_ACTION wasn't set, but
-# CROSS_COMPILE case was handled in original makefile
-
-# tested it on mingw and it doesn't hurt ;-)
-#ifneq ($(OS),WNT)
-ifneq ($(OS)$(COM),WNTMSC)
+ifneq ($(OS),WNT)
 $(eval $(call gb_Module_add_targets,expat,\
 	ExternalProject_expat \
 ))
--- expat/README	2015-11-25 04:55:38.000000000 -0800
+++ expat/README	2018-05-09 03:27:29.000000000 -0700
@@ -1,4 +1,4 @@
 Simple SAX parser library with added UTF-16 support.
 
 From:
-[http://expat.sourceforge.net/]
+[https://github.com/libexpat/libexpat]
--- expat/StaticLibrary_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/StaticLibrary_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -9,13 +9,13 @@
 
 $(eval $(call gb_StaticLibrary_StaticLibrary,expat))
 
-$(eval $(call gb_StaticLibrary_set_warnings_not_errors,expat))
+$(eval $(call gb_StaticLibrary_set_warnings_disabled,expat))
 
 $(eval $(call gb_StaticLibrary_use_unpacked,expat,expat))
 
 # no configure step on windows, no dependency
-ifneq ($(OS)$(COM),WNTMSC)
-$(eval $(call gb_StaticLibrary_use_external_project,expat,expat))
+ifneq ($(OS),WNT)
+$(eval $(call gb_StaticLibrary_use_external_project,expat,expat,full))
 endif
 
 $(eval $(call gb_StaticLibrary_set_include,expat,\
@@ -42,11 +42,6 @@
 ))
 endif
 
-# suppress warning spam
-$(eval $(call gb_StaticLibrary_add_cflags,expat,\
-	-w \
-))
-
 $(eval $(call gb_StaticLibrary_add_generated_cobjects,expat,\
 	UnpackedTarball/expat/lib/xmlparse \
 	UnpackedTarball/expat/lib/xmlrole \
--- expat/StaticLibrary_expat_x64.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/StaticLibrary_expat_x64.mk	2019-09-22 19:18:51.000000000 -0700
@@ -11,7 +11,7 @@
 
 $(eval $(call gb_StaticLibrary_set_x64,expat_x64,YES))
 
-$(eval $(call gb_StaticLibrary_set_warnings_not_errors,expat_x64))
+$(eval $(call gb_StaticLibrary_set_warnings_disabled,expat_x64))
 
 $(eval $(call gb_StaticLibrary_use_unpacked,expat_x64,expat))
 
--- expat/UnpackedTarball_expat.mk	2015-11-25 04:55:38.000000000 -0800
+++ expat/UnpackedTarball_expat.mk	2019-09-22 19:18:51.000000000 -0700
@@ -11,8 +11,10 @@
 
 $(eval $(call gb_UnpackedTarball_set_tarball,expat,$(EXPAT_TARBALL)))
 
+$(eval $(call gb_UnpackedTarball_update_autoconf_configs,expat,conftools))
+
 $(eval $(call gb_UnpackedTarball_add_patches,expat,\
-	external/expat/expat-2.1.0.patch \
+	external/expat/expat-winapi.patch \
 ))
 
 # This is a bit hackish.
--- expat/expat-winapi.patch	2015-11-25 04:55:38.000000000 -0800
+++ 
@@ -1,13 +1,13 @@
 --- misc/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:17.375000000 +0000
 +++ misc/build/expat-2.1.0/lib/expat_external.h	2009-11-16 08:53:34.703125000 +0000
-@@ -7,10 +7,6 @@
+@@ -81,10 +81,6 @@
+ #  ifndef XML_BUILDING_EXPAT
+ /* using Expat from an application */
  
- /* External API definitions */
- 
--#if defined(_MSC_EXTENSIONS) && !defined(__BEOS__) && !defined(__CYGWIN__)
--#define XML_USE_MSC_EXTENSIONS 1
--#endif
+-#    if defined(_MSC_EXTENSIONS) && ! defined(__BEOS__) && ! defined(__CYGWIN__)
+-#      define XMLIMPORT __declspec(dllimport)
+-#    endif
 -
- /* Expat tries very hard to make the API boundary very specifically
-    defined.  There are two macros defined to control this boundary;
-    each of these can be defined before including this header to
+ #  endif
+ #endif /* not defined XML_STATIC */
+ 
--- redland/UnpackedTarball_raptor.mk	2015-11-25 07:55:38.000000000 -0500
+++ redland/UnpackedTarball_raptor.mk	2021-12-07 12:35:00.000000000 -0500
@@ -20,6 +20,7 @@
 	$(if $(filter WNTGCC,$(OS)$(COM)),external/redland/raptor/raptor-mingw.patch.1) \
 	$(if $(filter-out WNT,$(OS)),external/redland/raptor/raptor-bundled-soname.patch.1) \
 	$(if $(filter ANDROID,$(OS)),external/redland/raptor/raptor-android.patch.1) \
+	external/redland/raptor/0001-CVE-2020-25713-raptor2-malformed-input-file-can-lead.patch.1 \
 ))
 
 # vim: set noet sw=4 ts=4:
--- libxmlsec/xmlsec1-configure.patch	2015-11-25 07:55:38.000000000 -0500
+++ libxmlsec/xmlsec1-configure.patch	2021-12-07 14:29:45.000000000 -0500
@@ -28,6 +28,14 @@
  pkgconfig_DATA = xmlsec1.pc @XMLSEC_CRYPTO_PC_FILES_LIST@
 --- misc/xmlsec1-1.2.14/configure.in	2009-06-25 22:53:18.000000000 +0200
 +++ misc/build/xmlsec1-1.2.14/configure.in	2009-10-01 10:28:50.990755126 +0200
+@@ -29,6 +29,7 @@
+ dnl Check the environment
+ dnl
+ AC_PROG_CC
++AC_PROG_CPP
+ AC_PROG_INSTALL
+ AC_HEADER_STDC
+ 
 @@ -192,8 +192,8 @@
  dnl ==========================================================================
  LIBXML_MIN_VERSION="2.7.4"
--- expat/ExternalProject_expat.mk	2022-02-19 14:34:37.000000000 -0500
+++ expat/ExternalProject_expat.mk	2022-02-19 19:13:33.000000000 -0500
@@ -17,7 +17,7 @@
 	$(call gb_Trace_StartRange,expat,EXTERNAL)
 	$(call gb_ExternalProject_run,configure,\
 		$(gb_RUN_CONFIGURE) ./configure --without-docbook \
-			$(gb_CONFIGURE_PLATFORMS) \
+			--host=$(HOST_PLATFORM) \
 			$(if $(CROSS_COMPILING),$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
 	,,expat_configure.log)
 	$(call gb_Trace_EndRange,expat,EXTERNAL)
