--- source/aqua/SalAquaFilePicker.mm.bak	2015-11-25 07:55:38.000000000 -0500
+++ source/aqua/SalAquaFilePicker.mm	2020-12-21 07:54:43.000000000 -0500
@@ -209,7 +209,7 @@
     // So instead of:
     // [m_pDialog setDelegate:m_pDelegate];
     // do:
-    objc_msgSend(m_pDialog, @selector(setDelegate:), m_pDelegate);
+    m_pDialog setDelegate:m_pDelegate];
 
     int nStatus = runandwaitforresult();
 
