--- source/services/pathsettings.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/services/pathsettings.cxx	2021-05-11 09:40:33.000000000 -0400
@@ -102,11 +102,6 @@
                 , bIsReadonly   (false)
             {}
 
-            PathInfo(const PathInfo& rCopy)
-            {
-                takeOver(rCopy);
-            }
-
             void takeOver(const PathInfo& rCopy)
             {
                 sPathName      = rCopy.sPathName;
--- inc/uielement/uielement.hxx	2015-11-25 07:55:38.000000000 -0500
+++ inc/uielement/uielement.hxx	2021-05-11 09:41:48.000000000 -0400
@@ -92,7 +92,6 @@
                    m_nStyle( BUTTON_SYMBOL ) {}
 
     bool operator< ( const UIElement& aUIElement ) const;
-    UIElement& operator=( const UIElement& rUIElement );
 
     OUString                                                      m_aType;
     OUString                                                      m_aName;
--- source/layoutmanager/uielement.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/layoutmanager/uielement.cxx	2021-05-11 09:43:03.000000000 -0400
@@ -100,32 +100,6 @@
     }
 }
 
-UIElement& UIElement::operator= ( const UIElement& rUIElement )
-{
-    if (&rUIElement != this)
-    {
-        m_aType             = rUIElement.m_aType;
-        m_aName             = rUIElement.m_aName;
-        m_aUIName           = rUIElement.m_aUIName;
-        m_xUIElement        = rUIElement.m_xUIElement;
-        m_bFloating         = rUIElement.m_bFloating;
-        m_bVisible          = rUIElement.m_bVisible;
-        m_bUserActive       = rUIElement.m_bUserActive;
-        m_bCreateNewRowCol0 = rUIElement.m_bCreateNewRowCol0;
-        m_bDeactiveHide     = rUIElement.m_bDeactiveHide;
-        m_bMasterHide       = rUIElement.m_bMasterHide;
-        m_bContextSensitive = rUIElement.m_bContextSensitive;
-        m_bContextActive    = rUIElement.m_bContextActive;
-        m_bNoClose          = rUIElement.m_bNoClose;
-        m_bSoftClose        = rUIElement.m_bSoftClose;
-        m_bStateRead        = rUIElement.m_bStateRead;
-        m_nStyle            = rUIElement.m_nStyle;
-        m_aDockedData       = rUIElement.m_aDockedData;
-        m_aFloatingData     = rUIElement.m_aFloatingData;
-    }
-    return *this;
-}
-
 } // namespace framework
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
