--- inc/database.hxx	2015-11-25 07:55:38.000000000 -0500
+++ inc/database.hxx	2021-05-11 16:20:27.000000000 -0400
@@ -45,12 +45,6 @@
     void SetText( const OString& rT ) { aText = rT; }
     bool            IsError() const { return nLine != 0; }
     void            Clear() { nLine = nColumn = 0; }
-    SvIdlError &    operator = ( const SvIdlError & rRef )
-    { aText   = rRef.aText;
-      nLine   = rRef.nLine;
-      nColumn = rRef.nColumn;
-      return *this;
-    }
 };
 
 class SvIdlDataBase
