--- svx/tbcontrl.hxx	2016-11-24 18:45:24.000000000 -0800
+++ svx/tbcontrl.hxx	2016-11-24 19:25:29.000000000 -0800
@@ -138,10 +138,10 @@
 #include <boost/scoped_ptr.hpp>
 #include <com/sun/star/awt/FontDescriptor.hpp>
 #include <svx/PaletteManager.hxx>
+#include <editeng/boxitem.hxx>
 
 // important im tbxctrls.hxx created HeDaBu !!!
 class SvxLineItem;
-class SvxBoxInfoItem;
 class SvxFontItem;
 class SfxStyleControllerItem_Impl;
 class SfxStyleSheetBasePool;
@@ -256,17 +256,25 @@
 
 // class SvxFrameToolBoxControl ------------------------------------------
 
+class SvxFrameWindowState_Impl;
+
 class SVX_DLLPUBLIC SvxFrameToolBoxControl : public SfxToolBoxControl
 {
+    SvxBoxItem                  aCurBorderOuter;
+    SvxBoxInfoItem              aCurBorderInner;
+
 public:
     SFX_DECL_TOOLBOX_CONTROL();
     SvxFrameToolBoxControl( sal_uInt16 nSlotId, sal_uInt16 nId, ToolBox& rTbx );
+    virtual ~SvxFrameToolBoxControl() {}
 
     virtual SfxPopupWindowType  GetPopupWindowType() const SAL_OVERRIDE;
     virtual SfxPopupWindow*     CreatePopupWindow() SAL_OVERRIDE;
     virtual void                StateChanged( sal_uInt16 nSID, SfxItemState eState,
                                               const SfxPoolItem* pState ) SAL_OVERRIDE;
 
+   virtual void                 Select(sal_uInt16 nSelectModifier) SAL_OVERRIDE;
+   DECL_LINK( BorderChangedHdl, SvxFrameWindowState_Impl* );
 };
 
 
--- vcl/svapp.hxx	2015-11-25 04:55:38.000000000 -0800
+++ vcl/svapp.hxx	2016-10-27 15:17:19.000000000 -0700
@@ -629,6 +629,8 @@
     */
     static bool                 IsInExecute();
 
+    static sal_Bool             IsShutDown();
+
     /** Queries whether application has a modal dialog active.
 
      @returns   true if a modal dialog is active, false if not
--- svx/svdoole2.hxx	2015-11-25 04:55:38.000000000 -0800
+++ svx/svdoole2.hxx	2016-12-08 13:26:42.000000000 -0800
@@ -22,6 +22,7 @@
 
 #include <svx/svdorect.hxx>
 #include <svx/svxdllapi.h>
+#include <sfx2/linkmgr.hxx>
 
 #include <com/sun/star/uno/Reference.h>
 
@@ -179,6 +180,21 @@
     virtual SdrObject* DoConvertToPolyObj(bool bBezier, bool bAddText) const SAL_OVERRIDE;
 };
 
+class SVX_DLLPUBLIC SdrEmbedObjectLink : public sfx2::SvBaseLink
+{
+    SdrOle2Obj*         pObj;
+
+public:
+    explicit            SdrEmbedObjectLink(SdrOle2Obj* pObj);
+    virtual             ~SdrEmbedObjectLink();
+
+    virtual void        Closed() override;
+    virtual ::sfx2::SvBaseLink::UpdateResult DataChanged(
+        const OUString& rMimeType, const css::uno::Any & rValue ) override;
+
+    bool                Connect() { return GetRealObject() != nullptr; }
+};
+
 #endif // INCLUDED_SVX_SVDOOLE2_HXX
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
--- cppuhelper/propshlp.hxx	2015-11-25 07:55:38.000000000 -0500
+++ cppuhelper/propshlp.hxx	2020-12-19 10:59:19.000000000 -0500
@@ -649,7 +649,7 @@
 
     /** reserved for future use. finally, the future has arrived...
      */
-    const std::auto_ptr<Impl> m_pReserved;
+    const std::unique_ptr<Impl> m_pReserved;
 
 private:
     OPropertySetHelper( const OPropertySetHelper & );
--- cppuhelper/unourl.hxx	2015-11-25 07:55:38.000000000 -0500
+++ cppuhelper/unourl.hxx	2020-12-19 11:00:05.000000000 -0500
@@ -50,7 +50,7 @@
     explicit UnoUrlDescriptor(rtl::OUString const & rDescriptor);
 
     /// @cond INTERNAL
-    explicit UnoUrlDescriptor(std::auto_ptr< Impl > & rImpl);
+    explicit UnoUrlDescriptor(std::unique_ptr< Impl > & rImpl);
     /// @endcond
 
     UnoUrlDescriptor(UnoUrlDescriptor const & rOther);
@@ -98,7 +98,7 @@
     rtl::OUString getParameter(rtl::OUString const & rKey) const;
 
 private:
-    std::auto_ptr< Impl > m_xImpl;
+    std::unique_ptr< Impl > m_xImpl;
 };
 
 /** Parse UNO URLs into their components.
@@ -175,7 +175,7 @@
 private:
     class Impl;
 
-    std::auto_ptr< Impl > m_xImpl;
+    std::unique_ptr< Impl > m_xImpl;
 };
 
 }
