--- svx/tbcontrl.hxx	2016-11-24 18:45:24.000000000 -0800
+++ svx/tbcontrl.hxx	2016-11-24 19:25:29.000000000 -0800
@@ -138,10 +138,10 @@
 #include <boost/scoped_ptr.hpp>
 #include <com/sun/star/awt/FontDescriptor.hpp>
 #include <svx/PaletteManager.hxx>
+#include <editeng/boxitem.hxx>
 
 // important im tbxctrls.hxx created HeDaBu !!!
 class SvxLineItem;
-class SvxBoxInfoItem;
 class SvxFontItem;
 class SfxStyleControllerItem_Impl;
 class SfxStyleSheetBasePool;
@@ -256,17 +256,25 @@
 
 // class SvxFrameToolBoxControl ------------------------------------------
 
+class SvxFrameWindowState_Impl;
+
 class SVX_DLLPUBLIC SvxFrameToolBoxControl : public SfxToolBoxControl
 {
+    SvxBoxItem                  aCurBorderOuter;
+    SvxBoxInfoItem              aCurBorderInner;
+
 public:
     SFX_DECL_TOOLBOX_CONTROL();
     SvxFrameToolBoxControl( sal_uInt16 nSlotId, sal_uInt16 nId, ToolBox& rTbx );
+    virtual ~SvxFrameToolBoxControl() {}
 
     virtual SfxPopupWindowType  GetPopupWindowType() const SAL_OVERRIDE;
     virtual SfxPopupWindow*     CreatePopupWindow() SAL_OVERRIDE;
     virtual void                StateChanged( sal_uInt16 nSID, SfxItemState eState,
                                               const SfxPoolItem* pState ) SAL_OVERRIDE;
 
+   virtual void                 Select(sal_uInt16 nSelectModifier) SAL_OVERRIDE;
+   DECL_LINK( BorderChangedHdl, SvxFrameWindowState_Impl* );
 };
 
 
--- vcl/svapp.hxx	2015-11-25 04:55:38.000000000 -0800
+++ vcl/svapp.hxx	2016-10-27 15:17:19.000000000 -0700
@@ -629,6 +629,8 @@
     */
     static bool                 IsInExecute();
 
+    static sal_Bool             IsShutDown();
+
     /** Queries whether application has a modal dialog active.
 
      @returns   true if a modal dialog is active, false if not
