--- source/filter/lwpborderstuff.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/lwpborderstuff.hxx	2021-05-11 13:50:44.000000000 -0400
@@ -86,7 +86,6 @@
     sal_uInt16  GetSideType(sal_uInt16 side);
     LwpColor    GetSideColor(sal_uInt16 side);
     float       GetSideWidth(sal_uInt16 side);
-    LwpBorderStuff& operator = (const LwpBorderStuff& rOther);
     friend class LwpParaBorderOverride;
 private:
     sal_uInt16      m_nSides;
--- source/filter/lwpborderstuff.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/lwpborderstuff.cxx	2021-05-11 13:52:07.000000000 -0400
@@ -215,29 +215,4 @@
     return 0;
 }
 
-LwpBorderStuff& LwpBorderStuff::operator = (const LwpBorderStuff& rOther)
-{
-    m_nSides = rOther.m_nSides;
-    m_nValid = rOther.m_nValid;
-
-    m_nBoderGroupIDLeft = rOther.m_nBoderGroupIDLeft;
-    m_nBoderGroupIDRight = rOther.m_nBoderGroupIDRight;
-    m_nBoderGroupIDTop = rOther.m_nBoderGroupIDTop;
-    m_nBoderGroupIDBottom = rOther.m_nBoderGroupIDBottom;
-
-    m_nGroupIndent = rOther.m_nGroupIndent;
-
-    m_nWidthLeft = rOther.m_nWidthLeft;
-    m_nWidthTop = rOther.m_nWidthTop;
-    m_nWidthRight = rOther.m_nWidthRight;
-    m_nWidthBottom = rOther.m_nWidthBottom;
-
-    m_aColorLeft = rOther.m_aColorLeft;
-    m_aColorRight = rOther.m_aColorRight;
-    m_aColorTop = rOther.m_aColorTop;
-    m_aColorBottom = rOther.m_aColorBottom;
-
-    return *this;
-}
-
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
--- source/filter/lwpmargins.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/lwpmargins.hxx	2021-05-11 13:53:09.000000000 -0400
@@ -78,7 +78,6 @@
         m_nBottom = pStrm->QuickReadInt32();
         pStrm->SkipExtra();
     }
-    inline LwpMargins& operator = (const LwpMargins& rOther);
     inline double GetMarginsValue(const sal_uInt8& nWhichSide);
 private:
     sal_Int32       m_nLeft;
@@ -87,15 +86,6 @@
     sal_Int32       m_nBottom;
 };
 
-inline LwpMargins& LwpMargins::operator = (const LwpMargins& rOther)
-{
-    m_nLeft = rOther.m_nLeft;
-    m_nTop = rOther.m_nTop;
-    m_nRight = rOther.m_nRight;
-    m_nBottom = rOther.m_nBottom;
-    return *this;
-}
-
 inline double LwpMargins::GetMarginsValue(const sal_uInt8& nWhichSide)
 {
     switch (nWhichSide)
--- source/filter/xfilter/xfmargins.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/xfilter/xfmargins.hxx	2021-05-11 13:54:03.000000000 -0400
@@ -87,7 +87,6 @@
 
     friend bool operator==(XFMargins& indent1, XFMargins& indent2);
     friend bool operator!=(XFMargins& indent1, XFMargins& indent2);
-    XFMargins& operator=(const XFMargins& other);
 private:
     double  m_fLeft;
     double  m_fRight;
--- source/filter/lwpshadow.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/lwpshadow.hxx	2021-05-11 14:03:27.000000000 -0400
@@ -82,10 +82,6 @@
 
     LwpColor GetColor();
 
-    //add by , 01/26/2005
-    inline LwpShadow& operator = (const LwpShadow& rOther);
-    //end add
-
 private:
     LwpColor        m_aColor;
     sal_Int32       m_nDirX;
@@ -108,13 +104,6 @@
     return m_aColor;
 }
 
-inline LwpShadow& LwpShadow::operator = (const LwpShadow& rOther)
-{
-    m_aColor = rOther.m_aColor;
-    m_nDirX = rOther.m_nDirX;
-    m_nDirY = rOther.m_nDirY;
-    return *this;
-}
 #endif
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
