--- Makefile.gbuild	2022-10-10 06:15:14.000000000 -0400
+++ Makefile.gbuild	2022-10-14 19:10:48.000000000 -0400
@@ -18,7 +18,13 @@
 
 include $(SRCDIR)/solenv/gbuild/gbuild.mk
 
+# If cross-compiling to a DESKTOP platform, just use RepositoryModule_host.mk on the build platform, too
+ifeq ($(gb_Side)-$(filter DESKTOP,$(BUILD_TYPE_FOR_HOST)),build-DESKTOP)
+$(eval $(call gb_Module_make_global_targets,$(SRCDIR)/RepositoryModule_host.mk))
+else
 $(eval $(call gb_Module_make_global_targets,$(SRCDIR)/RepositoryModule_$(gb_Side).mk))
+endif
+
 
 ifeq ($(DISABLE_DYNLOADING),TRUE)
 $(if $(gb_LinkTarget__Lock),$(shell rm -f $(gb_LinkTarget__Lock)))
