--- source/module/pyuno_impl.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/module/pyuno_impl.hxx	2020-12-24 07:20:13.000000000 -0500
@@ -63,7 +63,7 @@
     return PyUnicode_FromString(string);
 }
 
-inline char * PyStr_AsString(PyObject *object)
+inline char const * PyStr_AsString(PyObject *object)
 {
     return PyUnicode_AsUTF8(object);
 }
--- source/module/pyuno_type.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/module/pyuno_type.cxx	2020-12-24 07:22:06.000000000 -0500
@@ -157,7 +157,7 @@
     }
 
     OUString strTypeName( OUString::createFromAscii( PyStr_AsString( typeName.get() ) ) );
-    char *stringValue = PyStr_AsString( value.get() );
+    char const *stringValue = PyStr_AsString( value.get() );
 
     TypeDescription desc( strTypeName );
     if( desc.is() )
--- source/module/pyuno_util.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/module/pyuno_util.cxx	2020-12-24 07:23:06.000000000 -0500
@@ -78,7 +78,7 @@
 #else
 #if PY_MAJOR_VERSION >= 3
     Py_ssize_t size(0);
-    char *pUtf8(PyUnicode_AsUTF8AndSize(pystr, &size));
+    char const *pUtf8(PyUnicode_AsUTF8AndSize(pystr, &size));
     ret = OUString(pUtf8, size, RTL_TEXTENCODING_UTF8);
 #else
     PyObject* pUtf8 = PyUnicode_AsUTF8String(pystr);
