--- RepositoryExternal.mk	2015-11-25 07:55:38.000000000 -0500
+++ RepositoryExternal.mk	2021-10-26 13:07:28.000000000 -0400
@@ -630,7 +630,7 @@
 
 ifneq ($(SYSTEM_BOOST),)
 
-define gb_LinkTarget__use_boostdatetime
+define gb_LinkTarget__use_boost_lib
 $(call gb_LinkTarget_set_include,$(1),\
 	$$(INCLUDE) \
 	$(BOOST_CPPFLAGS) \
@@ -640,43 +640,36 @@
 	$(BOOST_LDFLAGS) \
 )
 
-$(call gb_LinkTarget_add_libs,$(1),\
-	$(BOOST_DATE_TIME_LIB) \
-)
+$(call gb_LinkTarget_add_libs,$(1),$(2))
 
 endef
 
-define gb_LinkTarget__use_boost_iostreams
-$(call gb_LinkTarget_set_include,$(1),\
-	$$(INCLUDE) \
-	$(BOOST_CPPFLAGS) \
-)
+define gb_LinkTarget__use_boost_locale
+$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_LOCALE_LIB))
 
-$(call gb_LinkTarget_add_ldflags,$(1),\
-	$(BOOST_LDFLAGS) \
-)
+endef
 
-$(call gb_LinkTarget_add_libs,$(1),\
-	$(BOOST_IOSTREAMS_LIB) \
-)
+define gb_LinkTarget__use_boostdatetime
+$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_DATE_TIME_LIB))
 
 endef
 
-gb_ExternalProject__use_boost_iostreams :=
+define gb_LinkTarget__use_boost_filesystem
+$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_FILESYSTEM_LIB))
 
-define gb_LinkTarget__use_boost_system
-$(call gb_LinkTarget_set_include,$(1),\
-	$$(INCLUDE) \
-	$(BOOST_CPPFLAGS) \
-)
+endef
 
-$(call gb_LinkTarget_add_ldflags,$(1),\
-	$(BOOST_LDFLAGS) \
-)
+gb_ExternalProject__use_boost_filesystem :=
 
-$(call gb_LinkTarget_add_libs,$(1),\
-	$(BOOST_SYSTEM_LIB) \
-)
+define gb_LinkTarget__use_boost_iostreams
+$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_IOSTREAMS_LIB))
+
+endef
+
+gb_ExternalProject__use_boost_iostreams :=
+
+define gb_LinkTarget__use_boost_system
+$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_SYSTEM_LIB))
 
 endef
 
@@ -694,37 +687,44 @@
 
 else # !SYSTEM_BOOST
 
-ifeq ($(OS),WNT)
-define gb_LinkTarget__use_boostthread
+define gb_LinkTarget__use_boost_lib
 $(call gb_LinkTarget_add_defs,$(1),\
 	-DBOOST_ALL_NO_LIB \
 )
 
-$(call gb_LinkTarget_use_static_libraries,$(1),\
-	boostthread \
+$(call gb_LinkTarget_use_static_libraries,$(1),$(2))
+
+endef
+
+define gb_LinkTarget__use_boost_locale
+$(call gb_LinkTarget__use_boost_lib,$(1),boost_locale)
+$(call gb_LinkTarget_add_libs,$(1),\
+	$(if $(filter $(OS),MACOSX),-liconv) \
 )
+
+endef
+
+define gb_LinkTarget__use_boostthread
+$(call gb_LinkTarget__use_boost_lib,$(1),boostthread)
+
 endef
-endif
 
 define gb_LinkTarget__use_boostdatetime
-$(call gb_LinkTarget_add_defs,$(1),\
-	-DBOOST_ALL_NO_LIB \
-)
+$(call gb_LinkTarget__use_boost_lib,$(1),boost_date_time)
 
-$(call gb_LinkTarget_use_static_libraries,$(1),\
-	boostdatetime \
-)
+endef
+
+define gb_LinkTarget__use_boost_filesystem
+$(call gb_LinkTarget__use_boost_lib,$(1),boost_filesystem)
 
 endef
 
-define gb_LinkTarget__use_boost_iostreams
-$(call gb_LinkTarget_add_defs,$(1),\
-	-DBOOST_ALL_NO_LIB \
-)
+define gb_ExternalProject__use_boost_filesystem
+$(call gb_ExternalProject_use_static_libraries,$(1),boost_filesystem)
+endef
 
-$(call gb_LinkTarget_use_static_libraries,$(1),\
-	boost_iostreams \
-)
+define gb_LinkTarget__use_boost_iostreams
+$(call gb_LinkTarget__use_boost_lib,$(1),boost_iostreams)
 
 endef
 
@@ -733,13 +733,7 @@
 endef
 
 define gb_LinkTarget__use_boost_system
-$(call gb_LinkTarget_add_defs,$(1),\
-	-DBOOST_ALL_NO_LIB \
-)
-
-$(call gb_LinkTarget_use_static_libraries,$(1),\
-	boost_system \
-)
+$(call gb_LinkTarget__use_boost_lib,$(1),boost_system)
 
 endef
 
@@ -750,7 +744,7 @@
 define gb_LinkTarget__use_boost_headers
 $(call gb_LinkTarget_use_unpacked,$(1),boost)
 $(call gb_LinkTarget_set_include,$(1),\
-	-I$(call gb_UnpackedTarball_get_dir,boost) \
+	$(BOOST_CPPFLAGS) \
 	$$(INCLUDE) \
 )
 
@@ -760,6 +754,7 @@
 $(call gb_ExternalProject_get_preparation_target,$(1)) :| $(call gb_UnpackedTarball_get_final_target,boost)
 
 endef
+
 endif # SYSTEM_BOOST
 
 
@@ -778,11 +773,11 @@
 
 define gb_LinkTarget__use_cmis
 $(call gb_LinkTarget_set_include,$(1),\
-	-I$(call gb_UnpackedTarball_get_dir,cmis)/src \
+	-I$(call gb_UnpackedTarball_get_dir,libcmis)/inc \
 	$$(INCLUDE) \
 )
 $(call gb_LinkTarget_use_static_libraries,$(1),\
-	cmislib \
+	libcmis \
 )
 
 endef
@@ -1223,8 +1218,9 @@
 $(call gb_LinkTarget_use_packages,$(1),raptor rasqal redland)
 
 $(call gb_LinkTarget_add_libs,$(1),\
-	-L$(call gb_UnpackedTarball_get_dir,raptor)/src/.libs -lraptor2 \
 	-L$(call gb_UnpackedTarball_get_dir,redland)/src/.libs -lrdf \
+	-L$(call gb_UnpackedTarball_get_dir,raptor)/src/.libs -lraptor2 \
+	-L$(call gb_UnpackedTarball_get_dir,rasqal)/src/.libs -lrasqal \
 )
 endif
 
@@ -2487,6 +2483,8 @@
 
 endef
 
+gb_ExternalProject__use_png :=
+
 else # !SYSTEM_LIBPNG
 
 define gb_LinkTarget__use_png
@@ -2495,12 +2493,19 @@
 	$$(INCLUDE) \
 )
 $(call gb_LinkTarget_use_static_libraries,$(1),\
-	png \
+	libpng \
 )
 $(call gb_LinkTarget__use_zlib,$(1))
 
 endef
 
+define gb_ExternalProject__use_png
+$(call gb_ExternalProject_use_static_libraries,$(1),\
+	libpng \
+)
+
+endef
+
 endif # !SYSTEM_LIBPNG
 
 
@@ -2727,6 +2732,37 @@
 endif # SYSTEM_OPENLDAP
 
 
+ifneq ($(SYSTEM_LIBTOMMATH),)
+
+define gb_LinkTarget__use_libtommath
+$(call gb_LinkTarget_set_include,$(1),\
+	$(LIBTOMMATH_CFLAGS) \
+	$$(INCLUDE) \
+)
+$(call gb_LinkTarget_add_libs,$(1),$(LIBTOMMATH_LIBS))
+
+endef
+
+else # !SYSTEM_LIBTOMMATH
+define gb_LinkTarget__use_libtommath
+$(call gb_LinkTarget_set_include,$(1),\
+	-I${WORKDIR}/UnpackedTarball/libtommath \
+	$$(INCLUDE) \
+)
+$(call gb_LinkTarget_add_libs,$(1),\
+	$(call gb_UnpackedTarball_get_dir,libtommath)/libtommath$(gb_StaticLibrary_PLAINEXT) \
+)
+$(call gb_LinkTarget_use_external_project,$(1),libtommath)
+
+endef
+
+endif # SYSTEM_LIBTOMMATH
+
+define gb_ExternalProject__use_libtommath
+$(call gb_ExternalProject_use_external_project,$(1),libtommath)
+
+endef
+
 ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
 
 ifneq ($(SYSTEM_FIREBIRD),)
@@ -2740,38 +2776,33 @@
 
 endef
 
-# gb_LinkTarget__use_atomic_ops :=
-# gb_LinkTarget__use_tommath :=
-
 else # !SYSTEM_FIREBIRD
 
+$(eval $(call gb_Helper_register_packages_for_install,firebirdsdbc,\
+	firebird \
+))
+
 #$(call gb_LinkTarget__use_libatomic_ops,$(1))
+#$(call gb_LinkTarget__use_libtommath,$(1))
 
 define gb_LinkTarget__use_libfbembed
 $(call gb_LinkTarget_use_package,$(1),firebird)
 $(call gb_LinkTarget_set_include,$(1),\
+	-I$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/include \
 	$$(INCLUDE) \
-	-I$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/include \
 )
 ifeq ($(COM),MSC)
 $(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/bin/ifbembed.lib \
+	$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/bin/ifbclient.lib \
 )
 else
 $(call gb_LinkTarget_add_libs,$(1),\
-	-L$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/lib -lfbembed \
+	-L$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/lib -lfbclient \
 )
 endif
 
 endef
 
-# define gb_LinkTarget__use_tommath
-# $(call gb_LinkTarget_set_include,$(1),\
-# 	$(TOMMATH_CFLAGS) \
-# 	$$(INCLUDE) \
-# )
-
-# $(call gb_LinkTarget_add_libs,$(1),$(TOMMATH_LIBS))
 
 # endef
 
@@ -2781,7 +2812,7 @@
 
 gb_LinkTarget__use_firebird :=
 # gb_LinkTarget__use_atomic_ops :=
-# gb_LinkTarget__use_tommath :=
+# gb_LinkTarget__use_libtommath :=
 
 endif # ENABLE_FIREBIRD_SDBC
 
@@ -3042,7 +3073,7 @@
 
 define gb_LinkTarget__use_python_headers
 $(call gb_LinkTarget_add_defs,$(1),\
-	$(filter-out -I% -isystem%,$(subst -isystem /,-isystem/,$(PYHTON_CFLAGS)))) \
+	$(filter-out -I% -isystem%,$(subst -isystem /,-isystem/,$(PYTHON_CFLAGS)))) \
 )
 
 $(call gb_LinkTarget_set_include,$(1),\
@@ -3065,11 +3096,11 @@
 
 $(eval $(call gb_Helper_register_packages_for_install,python,\
 	python3 \
+    $(if $(filter WNT,$(OS)),libffi) \
 ))
 
-# depend on external project because on MACOSX the Package is disabled...
 define gb_LinkTarget__use_python_headers
-$(call gb_LinkTarget_use_external_project,$(1),python3)
+$(call gb_LinkTarget_use_external_project,$(1),python3,full)
 $(call gb_LinkTarget_set_include,$(1),\
 	-I$(call gb_UnpackedTarball_get_dir,python3) \
 	-I$(call gb_UnpackedTarball_get_dir,python3)/PC \
@@ -3088,8 +3119,15 @@
 endif
 
 ifeq ($(OS),WNT)
+ifeq ($(CPUNAME),X86_64)
+python_arch_subdir=amd64
+else ifeq ($(CPUNAME),AARCH64)
+python_arch_subdir=arm64
+else
+python_arch_subdir=win32
+endif
 $(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,python3)/PCbuild$(if $(filter X86_64,$(CPUNAME)),/amd64)/python$(PYTHON_VERSION_MAJOR)$(PYTHON_VERSION_MINOR)$(if $(MSVC_USE_DEBUG_RUNTIME),_d).lib \
+	$(call gb_UnpackedTarball_get_dir,python3)/PCbuild/$(python_arch_subdir)/python$(PYTHON_VERSION_MAJOR)$(PYTHON_VERSION_MINOR)$(if $(MSVC_USE_DEBUG_RUNTIME),_d).lib \
 )
 else ifeq ($(OS),MACOSX)
 $(call gb_LinkTarget_add_libs,$(1),\
@@ -3098,12 +3136,18 @@
 else
 $(call gb_LinkTarget_add_libs,$(1),\
 	-L$(call gb_UnpackedTarball_get_dir,python3) \
-	-lpython$(PYTHON_VERSION_MAJOR).$(PYTHON_VERSION_MINOR)m \
+	-lpython$(PYTHON_VERSION_MAJOR).$(PYTHON_VERSION_MINOR)$(if $(ENABLE_DBGUTIL),d) \
 )
 endif
 
 endef
 
+# this is only used by python currently
+define gb_ExternalProject__use_libffi
+$(call gb_ExternalProject_use_external_project,$(1),libffi)
+
+endef
+
 endif # SYSTEM_PYTHON
 
 # ORCUS
@@ -3885,7 +3929,7 @@
 
 else
 
-$(call gb_ExternalExecutable_set_internal,python,$(INSTROOT)/$(LIBO_BIN_FOLDER)/$(if $(filter WNT,$(OS)),python-core-$(PYTHON_VERSION)/bin/python.exe,python.bin))
+$(call gb_ExternalExecutable_set_internal,python,$(INSTROOT_FOR_BUILD)/$(LIBO_BIN_FOLDER)/$(if $(filter WNT,$(OS)),python-core-$(PYTHON_VERSION)/bin/python.exe,python.bin))
 $(call gb_ExternalExecutable_set_precommand,python,$(subst $$,$$$$,$(gb_Python_PRECOMMAND)))
 $(call gb_ExternalExecutable_add_dependencies,python,$(call gb_Package_get_target_for_build,python3))
 
