--- RepositoryExternal.mk	2015-11-25 07:55:38.000000000 -0500
+++ RepositoryExternal.mk	2021-02-15 18:07:10.000000000 -0500
@@ -712,7 +712,7 @@
 )
 
 $(call gb_LinkTarget_use_static_libraries,$(1),\
-	boostdatetime \
+	boost_date_time \
 )
 
 endef
@@ -778,11 +778,11 @@
 
 define gb_LinkTarget__use_cmis
 $(call gb_LinkTarget_set_include,$(1),\
-	-I$(call gb_UnpackedTarball_get_dir,cmis)/src \
+	-I$(call gb_UnpackedTarball_get_dir,libcmis)/inc \
 	$$(INCLUDE) \
 )
 $(call gb_LinkTarget_use_static_libraries,$(1),\
-	cmislib \
+	libcmis \
 )
 
 endef
@@ -1223,8 +1223,9 @@
 $(call gb_LinkTarget_use_packages,$(1),raptor rasqal redland)
 
 $(call gb_LinkTarget_add_libs,$(1),\
-	-L$(call gb_UnpackedTarball_get_dir,raptor)/src/.libs -lraptor2 \
 	-L$(call gb_UnpackedTarball_get_dir,redland)/src/.libs -lrdf \
+	-L$(call gb_UnpackedTarball_get_dir,raptor)/src/.libs -lraptor2 \
+	-L$(call gb_UnpackedTarball_get_dir,rasqal)/src/.libs -lrasqal \
 )
 endif
 
@@ -2487,6 +2488,8 @@
 
 endef
 
+gb_ExternalProject__use_png :=
+
 else # !SYSTEM_LIBPNG
 
 define gb_LinkTarget__use_png
@@ -2495,12 +2498,19 @@
 	$$(INCLUDE) \
 )
 $(call gb_LinkTarget_use_static_libraries,$(1),\
-	png \
+	libpng \
 )
 $(call gb_LinkTarget__use_zlib,$(1))
 
 endef
 
+define gb_ExternalProject__use_png
+$(call gb_ExternalProject_use_static_libraries,$(1),\
+	libpng \
+)
+
+endef
+
 endif # !SYSTEM_LIBPNG
 
 
@@ -2727,6 +2737,38 @@
 endif # SYSTEM_OPENLDAP
 
 
+ifneq ($(SYSTEM_LIBTOMMATH),)
+
+define gb_LinkTarget__use_libtommath
+$(call gb_LinkTarget_set_include,$(1),\
+	$(LIBTOMMATH_CFLAGS) \
+	$$(INCLUDE) \
+)
+$(call gb_LinkTarget_add_libs,$(1),$(LIBTOMMATH_LIBS))
+
+endef
+
+else # !SYSTEM_LIBTOMMATH
+define gb_LinkTarget__use_libtommath
+$(call gb_LinkTarget_set_include,$(1),\
+	-I${WORKDIR}/UnpackedTarball/libtommath \
+	$$(INCLUDE) \
+)
+$(call gb_LinkTarget_add_libs,$(1),\
+	$(call gb_UnpackedTarball_get_dir,libtommath)/libtommath$(gb_StaticLibrary_PLAINEXT) \
+)
+$(call gb_LinkTarget_use_external_project,$(1),libtommath)
+
+endef
+
+endif # SYSTEM_LIBTOMMATH
+
+define gb_ExternalProject__use_libtommath
+$(call gb_ExternalProject_use_external_project,$(1),libtommath)
+
+endef
+
+
 ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
 
 ifneq ($(SYSTEM_FIREBIRD),)
@@ -2740,38 +2782,33 @@
 
 endef
 
-# gb_LinkTarget__use_atomic_ops :=
-# gb_LinkTarget__use_tommath :=
-
 else # !SYSTEM_FIREBIRD
 
+$(eval $(call gb_Helper_register_packages_for_install,firebirdsdbc,\
+	firebird \
+))
+
 #$(call gb_LinkTarget__use_libatomic_ops,$(1))
+#$(call gb_LinkTarget__use_libtommath,$(1))
 
 define gb_LinkTarget__use_libfbembed
 $(call gb_LinkTarget_use_package,$(1),firebird)
 $(call gb_LinkTarget_set_include,$(1),\
+	-I$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/include \
 	$$(INCLUDE) \
-	-I$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/include \
 )
 ifeq ($(COM),MSC)
 $(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/bin/ifbembed.lib \
+	$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/bin/ifbclient.lib \
 )
 else
 $(call gb_LinkTarget_add_libs,$(1),\
-	-L$(call gb_UnpackedTarball_get_dir,firebird)/gen/firebird/lib -lfbembed \
+	-L$(call gb_UnpackedTarball_get_dir,firebird)/gen/$(if $(ENABLE_DEBUG),Debug,Release)/firebird/lib -lfbclient \
 )
 endif
 
 endef
 
-# define gb_LinkTarget__use_tommath
-# $(call gb_LinkTarget_set_include,$(1),\
-# 	$(TOMMATH_CFLAGS) \
-# 	$$(INCLUDE) \
-# )
-
-# $(call gb_LinkTarget_add_libs,$(1),$(TOMMATH_LIBS))
 
 # endef
 
@@ -2781,7 +2818,7 @@
 
 gb_LinkTarget__use_firebird :=
 # gb_LinkTarget__use_atomic_ops :=
-# gb_LinkTarget__use_tommath :=
+# gb_LinkTarget__use_libtommath :=
 
 endif # ENABLE_FIREBIRD_SDBC
 
@@ -3042,7 +3079,7 @@
 
 define gb_LinkTarget__use_python_headers
 $(call gb_LinkTarget_add_defs,$(1),\
-	$(filter-out -I% -isystem%,$(subst -isystem /,-isystem/,$(PYHTON_CFLAGS)))) \
+	$(filter-out -I% -isystem%,$(subst -isystem /,-isystem/,$(PYTHON_CFLAGS)))) \
 )
 
 $(call gb_LinkTarget_set_include,$(1),\
@@ -3067,9 +3104,8 @@
 	python3 \
 ))
 
-# depend on external project because on MACOSX the Package is disabled...
 define gb_LinkTarget__use_python_headers
-$(call gb_LinkTarget_use_external_project,$(1),python3)
+$(call gb_LinkTarget_use_external_project,$(1),python3,full)
 $(call gb_LinkTarget_set_include,$(1),\
 	-I$(call gb_UnpackedTarball_get_dir,python3) \
 	-I$(call gb_UnpackedTarball_get_dir,python3)/PC \
@@ -3089,7 +3125,7 @@
 
 ifeq ($(OS),WNT)
 $(call gb_LinkTarget_add_libs,$(1),\
-	$(call gb_UnpackedTarball_get_dir,python3)/PCbuild$(if $(filter X86_64,$(CPUNAME)),/amd64)/python$(PYTHON_VERSION_MAJOR)$(PYTHON_VERSION_MINOR)$(if $(MSVC_USE_DEBUG_RUNTIME),_d).lib \
+	$(call gb_UnpackedTarball_get_dir,python3)/PCbuild/$(if $(filter X86_64,$(CPUNAME)),amd64,win32)/python$(PYTHON_VERSION_MAJOR)$(PYTHON_VERSION_MINOR)$(if $(MSVC_USE_DEBUG_RUNTIME),_d).lib \
 )
 else ifeq ($(OS),MACOSX)
 $(call gb_LinkTarget_add_libs,$(1),\
@@ -3104,6 +3140,12 @@
 
 endef
 
+# this is only used by python currently
+define gb_ExternalProject__use_libffi
+$(call gb_ExternalProject_use_external_project,$(1),libffi)
+
+endef
+
 endif # SYSTEM_PYTHON
 
 # ORCUS
