--- gbuild/Deliver.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/Deliver.mk	2016-10-21 13:35:45.000000000 -0700
@@ -50,6 +50,11 @@
 
 endef
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+define gb_Deliver__deliver
+echo The deliver command has been disabled
+endef
+else
 define gb_Deliver__deliver
 $(if $(gb_Deliver_CLEARONDELIVER),rm -f $(2) &&) $(if $(gb_Deliver_HARDLINK),ln,cp -P -f) $(1) $(2) && touch -hr $(1) $(2)
 endef
@@ -61,6 +66,7 @@
 endef
 endif
 endif
+endif
 
 define gb_Deliver_deliver
 $(if $(1),$(call gb_Deliver__deliver,$(1),$(2)),\
--- gbuild/LinkTarget.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/LinkTarget.mk	2016-10-23 15:07:56.000000000 -0700
@@ -17,6 +17,11 @@
 #   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 #
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+OLDSRCDIR:=$(SRCDIR)
+SRCDIR:=$(SRC_ROOT)
+endif
+
 #the following user-defined variables are supported:
 # YACCFLAGS
 # LEXFLAGS
@@ -1395,4 +1400,9 @@
 
 endef
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+SRCDIR:=$(OLDSRCDIR)
+OLDSRCDIR:=
+endif
+
 # vim: set noet sw=4:
--- gbuild/gbuild.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/gbuild.mk	2016-10-21 13:26:05.000000000 -0700
@@ -266,6 +266,10 @@
 		ENABLE_LTO \
 	)
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+gb_GLOBALDEFS += -DUSE_JAVA
+endif
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))
 
 include $(GBUILDDIR)/Deliver.mk
