--- gbuild/AllLangResTarget.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/AllLangResTarget.mk	2016-10-21 13:14:42.000000000 -0700
@@ -378,7 +378,11 @@
 
 # ResTarget
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+gb_ResTarget_DEFIMAGESLOCATION := $(SRC_ROOT)/icon-themes/galaxy/
+else
 gb_ResTarget_DEFIMAGESLOCATION := $(SRCDIR)/icon-themes/galaxy/
+endif
 
 $(call gb_ResTarget_get_clean_target,%) :
 	$(call gb_Output_announce,$*,$(false),RES,2)
--- gbuild/Deliver.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/Deliver.mk	2016-10-21 13:35:45.000000000 -0700
@@ -50,6 +50,11 @@
 
 endef
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+define gb_Deliver__deliver
+echo The deliver command has been disabled
+endef
+else
 define gb_Deliver__deliver
 $(if $(gb_Deliver_CLEARONDELIVER),rm -f $(2) &&) $(if $(gb_Deliver_HARDLINK),ln,cp -P -f) $(1) $(2) && touch -hr $(1) $(2)
 endef
@@ -61,6 +66,7 @@
 endef
 endif
 endif
+endif
 
 define gb_Deliver_deliver
 $(if $(1),$(call gb_Deliver__deliver,$(1),$(2)),\
--- gbuild/gbuild.mk	2015-11-25 04:55:38.000000000 -0800
+++ gbuild/gbuild.mk	2016-10-21 13:26:05.000000000 -0700
@@ -266,6 +266,10 @@
 		ENABLE_LTO \
 	)
 
+ifeq ($(strip $(PRODUCT_BUILD_TYPE)),java)
+gb_GLOBALDEFS += -DUSE_JAVA
+endif
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))
 
 include $(GBUILDDIR)/Deliver.mk
