--- source/control/ctrlbox.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/control/ctrlbox.cxx	2021-05-03 08:47:37.000000000 -0400
@@ -291,15 +291,6 @@
 {
 }
 
-BorderWidthImpl& BorderWidthImpl::operator= ( const BorderWidthImpl& r )
-{
-    m_nFlags = r.m_nFlags;
-    m_nRate1 = r.m_nRate1;
-    m_nRate2 = r.m_nRate2;
-    m_nRateGap = r.m_nRateGap;
-    return *this;
-}
-
 bool BorderWidthImpl::operator== ( const BorderWidthImpl& r ) const
 {
     return ( m_nFlags == r.m_nFlags ) &&
--- source/control/scriptedtext.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/control/scriptedtext.cxx	2021-05-03 08:43:13.000000000 -0400
@@ -48,9 +48,6 @@
     vector< sal_Int32 >         maWidthVec;         /// The output width of each text portion.
     Size                        maTextSize;         /// The size the text will take in the current output device.
 
-                                /** Assignment operator not implemented to prevent usage. */
-    SvtScriptedTextHelper_Impl& operator=( const SvtScriptedTextHelper_Impl& ) SAL_DELETED_FUNCTION;
-
                                 /** Gets the font of the given script type. */
     const vcl::Font&            GetFont( sal_uInt16 _nScript ) const;
                                 /** Sets a font on the output device depending on the script type. */
@@ -71,9 +68,6 @@
                                     vcl::Font* _pLatinFont,
                                     vcl::Font* _pAsianFont,
                                     vcl::Font* _pCmplxFont );
-                                /** Copy constructor. */
-                                SvtScriptedTextHelper_Impl(
-                                    const SvtScriptedTextHelper_Impl& _rCopy );
                                 /** Destructor. */
                                 ~SvtScriptedTextHelper_Impl();
 
@@ -103,20 +97,6 @@
 {
 }
 
-SvtScriptedTextHelper_Impl::SvtScriptedTextHelper_Impl( const SvtScriptedTextHelper_Impl& _rCopy ) :
-    mrOutDevice( _rCopy.mrOutDevice ),
-    maLatinFont( _rCopy.maLatinFont ),
-    maAsianFont( _rCopy.maAsianFont ),
-    maCmplxFont( _rCopy.maCmplxFont ),
-    maDefltFont( _rCopy.maDefltFont ),
-    maText( _rCopy.maText ),
-    maPosVec( _rCopy.maPosVec ),
-    maScriptVec( _rCopy.maScriptVec ),
-    maWidthVec( _rCopy.maWidthVec ),
-    maTextSize( _rCopy.maTextSize )
-{
-}
-
 SvtScriptedTextHelper_Impl::~SvtScriptedTextHelper_Impl()
 {
 }
--- source/control/inettbc.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/control/inettbc.cxx	2021-10-26 09:16:27.000000000 -0400
@@ -71,6 +71,7 @@
 using namespace ::com::sun::star::task;
 using namespace ::com::sun::star::ucb;
 using namespace ::com::sun::star::uno;
+using namespace ::boost::placeholders;
 
 class SvtURLBox_Impl
 {
