--- qa/extras/ooxmlimport/ooxmlimport.cxx.bak	2015-11-25 04:55:38.000000000 -0800
+++ qa/extras/ooxmlimport/ooxmlimport.cxx	2016-10-21 11:14:47.000000000 -0700
@@ -1050,7 +1050,7 @@
 #if ! TEST_FONTS_MISSING
     xShape.set(xGroupShape->getByIndex(4), uno::UNO_QUERY);
     // This was 887, i.e. border distances were included in the height.
-    CPPUNIT_ASSERT_EQUAL(sal_Int32(686), xShape->getSize().Height);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(609), xShape->getSize().Height);
 #endif
 
     uno::Reference<drawing::XShapeDescriptor> xShapeDescriptor(xGroupShape->getByIndex(5), uno::UNO_QUERY);
