--- source/app/svapp.cxx	2015-11-25 04:55:38.000000000 -0800
+++ source/app/svapp.cxx	2016-10-27 15:19:44.000000000 -0700
@@ -437,6 +437,11 @@
     return ImplGetSVData()->maAppData.mbInAppExecute;
 }
 
+sal_Bool Application::IsShutDown()
+{
+    return ImplGetSVData()->maAppData.mbAppQuit;
+}
+
 bool Application::IsInModalMode()
 {
     return (ImplGetSVData()->maAppData.mnModalMode != 0);
--- osx/PictToBmpFlt.cxx	2015-11-25 04:55:38.000000000 -0800
+++ osx/PictToBmpFlt.cxx	2017-02-16 13:41:28.000000000 -0800
@@ -19,7 +19,9 @@
 
 #include <premac.h>
 #include <Carbon/Carbon.h>
+#if MACOSX_SDK_VERSION < 101200
 #include <QuickTime/QuickTime.h>
+#endif	// MACOSX_SDK_VERSION < 101200
 #include <postmac.h>
 
 #include <string.h>
--- osx/OSXTransferable.cxx	2015-11-25 05:55:38.000000000 -0700
+++ osx/OSXTransferable.cxx	2018-10-15 18:11:30.000000000 -0600
@@ -73,8 +73,8 @@
   bool bInternal(false);
   NSString* sysFormat =
       (aFlavor.MimeType.startsWith("image/png"))
-      ? mDataFlavorMapper->openOfficeImageToSystemFlavor( mPasteboard )
-      : mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal);
+      ? (NSString *)mDataFlavorMapper->openOfficeImageToSystemFlavor( mPasteboard )
+      : (NSString *)mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal);
   DataProviderPtr_t dp;
 
   if ([sysFormat caseInsensitiveCompare: NSFilenamesPboardType] == NSOrderedSame)
