--- source/app/svapp.cxx	2015-11-25 04:55:38.000000000 -0800
+++ source/app/svapp.cxx	2016-10-27 15:19:44.000000000 -0700
@@ -622,6 +622,11 @@
     return ImplGetSVData()->maAppData.mbInAppExecute;
 }
 
+sal_Bool Application::IsShutDown()
+{
+    return ImplGetSVData()->maAppData.mbAppQuit;
+}
+
 bool Application::IsInModalMode()
 {
     return (ImplGetSVData()->maAppData.mnModalMode != 0);
--- osx/OSXTransferable.cxx	2018-05-09 05:27:29.000000000 -0500
+++ osx/OSXTransferable.cxx	2018-12-30 17:25:10.000000000 -0600
@@ -109,7 +109,7 @@
   NSString const * sysFormat =
       (aFlavor.MimeType.startsWith("image/png"))
       ? DataFlavorMapper::openOfficeImageToSystemFlavor( mPasteboard )
-      : mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal);
+      : static_cast<NSString*>(mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal));
   DataProviderPtr_t dp;
 
   if ([sysFormat caseInsensitiveCompare: NSFilenamesPboardType] == NSOrderedSame)
