--- source/app/svapp.cxx	2022-10-10 06:15:14.000000000 -0400
+++ source/app/svapp.cxx	2022-10-15 13:30:58.000000000 -0400
@@ -609,6 +609,11 @@
     return ImplGetSVData()->maAppData.mbAppQuit;
 }
 
+sal_Bool Application::IsShutDown()
+{
+    return ImplGetSVData()->maAppData.mbAppQuit;
+}
+
 bool Application::IsInModalMode()
 {
     return (ImplGetSVData()->maAppData.mnModalMode != 0);
--- osx/saldata.cxx	2022-10-10 06:15:14.000000000 -0400
+++ osx/saldata.cxx	2022-10-15 13:33:17.000000000 -0400
@@ -83,8 +83,10 @@
         NSAutoreleasePool* pPool = reinterpret_cast<NSAutoreleasePool*>( osl_getThreadKeyData( s_aAutoReleaseKey ) );
         if( pPool )
         {
-            osl_setThreadKeyData( s_aAutoReleaseKey, nullptr );
-            [pPool release];
+            // Avoid double release of pool by only releasing when setting the
+            // thread key data fails
+            if ( !osl_setThreadKeyData( s_aAutoReleaseKey, nullptr ) )
+                [pPool release];
         }
 
         osl_destroyThreadKey( s_aAutoReleaseKey );
