--- source/app/svapp.cxx	2015-11-25 04:55:38.000000000 -0800
+++ source/app/svapp.cxx	2016-10-27 15:19:44.000000000 -0700
@@ -437,6 +437,11 @@
     return ImplGetSVData()->maAppData.mbInAppExecute;
 }
 
+sal_Bool Application::IsShutDown()
+{
+    return ImplGetSVData()->maAppData.mbAppQuit;
+}
+
 bool Application::IsInModalMode()
 {
     return (ImplGetSVData()->maAppData.mnModalMode != 0);
--- osx/PictToBmpFlt.cxx	2015-11-25 04:55:38.000000000 -0800
+++ osx/PictToBmpFlt.cxx	2017-02-16 13:41:28.000000000 -0800
@@ -19,7 +19,9 @@
 
 #include <premac.h>
 #include <Carbon/Carbon.h>
+#if MACOSX_SDK_VERSION < 101200
 #include <QuickTime/QuickTime.h>
+#endif	// MACOSX_SDK_VERSION < 101200
 #include <postmac.h>
 
 #include <string.h>
--- osx/OSXTransferable.cxx	2015-11-25 05:55:38.000000000 -0700
+++ osx/OSXTransferable.cxx	2018-10-15 18:11:30.000000000 -0600
@@ -73,8 +73,8 @@
   bool bInternal(false);
   NSString* sysFormat =
       (aFlavor.MimeType.startsWith("image/png"))
-      ? mDataFlavorMapper->openOfficeImageToSystemFlavor( mPasteboard )
-      : mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal);
+      ? (NSString *)mDataFlavorMapper->openOfficeImageToSystemFlavor( mPasteboard )
+      : (NSString *)mDataFlavorMapper->openOfficeToSystemFlavor(aFlavor, bInternal);
   DataProviderPtr_t dp;
 
   if ([sysFormat caseInsensitiveCompare: NSFilenamesPboardType] == NSOrderedSame)
--- osx/salframe.cxx	2015-11-25 07:55:38.000000000 -0500
+++ osx/salframe.cxx	2020-12-17 21:24:36.000000000 -0500
@@ -197,7 +197,7 @@
 
     if( [mpNSWindow respondsToSelector: @selector(setRestorable:)])
     {
-        objc_msgSend(mpNSWindow, @selector(setRestorable:), NO);
+        [mpNSWindow setRestorable:NO];
     }
     const NSRect aRect = { NSZeroPoint, NSMakeSize( maGeometry.nWidth, maGeometry.nHeight )};
     mnTrackingRectTag = [mpNSView addTrackingRect: aRect owner: mpNSView userData: nil assumeInside: NO];
--- Library_vcl.mk	2015-11-25 07:55:38.000000000 -0500
+++ Library_vcl.mk	2020-12-27 22:26:31.000000000 -0500
@@ -499,7 +499,6 @@
     vcl/osx/salobj \
 ))
 $(eval $(call gb_Library_use_system_darwin_frameworks,vcl,\
-    $(if $(filter X86_64,$(CPUNAME)),,QuickTime) \
     Cocoa \
     Carbon \
     CoreFoundation \
--- source/app/IconThemeScanner.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/app/IconThemeScanner.cxx	2021-04-29 17:10:53.000000000 -0400
@@ -170,9 +170,6 @@
     return url;
 }
 
-IconThemeScanner::~IconThemeScanner()
-{;}
-
 namespace
 {
     class SameTheme :
--- source/app/settings.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/app/settings.cxx	2021-04-29 17:10:02.000000000 -0400
@@ -483,10 +483,6 @@
 {
 }
 
-MouseSettings::~MouseSettings()
-{
-}
-
 void MouseSettings::CopyData()
 {
     // copy if other references exist
@@ -771,10 +767,6 @@
 {
 }
 
-StyleSettings::~StyleSettings()
-{
-}
-
 void
 StyleSettings::SetFaceColor( const Color& rColor )
 {
@@ -2409,10 +2401,6 @@
 {
 }
 
-MiscSettings::~MiscSettings()
-{
-}
-
 void MiscSettings::CopyData()
 {
     // copy if other references exist
@@ -2613,10 +2601,6 @@
 {
 }
 
-HelpSettings::~HelpSettings()
-{
-}
-
 void HelpSettings::CopyData()
 {
     // copy if other references exist
@@ -2715,15 +2699,6 @@
 {
 }
 
-AllSettings::AllSettings( const AllSettings& rSet )
-{
-    mpData = rSet.mpData;
-}
-
-AllSettings::~AllSettings()
-{
-}
-
 void AllSettings::CopyData()
 {
     // copy if other references exist
--- source/gdi/print.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/gdi/print.cxx	2021-04-29 17:06:53.000000000 -0400
@@ -115,10 +115,6 @@
 {
 }
 
-PrinterOptions::~PrinterOptions()
-{
-}
-
 #define PROPERTYNAME_REDUCETRANSPARENCY                 OUString("ReduceTransparency")
 #define PROPERTYNAME_REDUCEDTRANSPARENCYMODE            OUString("ReducedTransparencyMode")
 #define PROPERTYNAME_REDUCEGRADIENTS                    OUString("ReduceGradients")
@@ -362,20 +358,6 @@
     mnJobs      = 0;
 }
 
-QueueInfo::QueueInfo( const QueueInfo& rInfo ) :
-    maPrinterName( rInfo.maPrinterName ),
-    maDriver( rInfo.maDriver ),
-    maLocation( rInfo.maLocation ),
-    maComment( rInfo.maComment ),
-    mnStatus( rInfo.mnStatus ),
-    mnJobs( rInfo.mnJobs )
-{
-}
-
-QueueInfo::~QueueInfo()
-{
-}
-
 bool QueueInfo::operator==( const QueueInfo& rInfo ) const
 {
     return
--- source/window/keyevent.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/window/keyevent.cxx	2021-04-29 17:04:44.000000000 -0400
@@ -22,12 +22,6 @@
 #include <tools/debug.hxx>
 #include <vcl/event.hxx>
 
-KeyEvent::KeyEvent (const KeyEvent& rKeyEvent) :
-    maKeyCode (rKeyEvent.maKeyCode),
-    mnRepeat  (rKeyEvent.mnRepeat),
-    mnCharCode(rKeyEvent.mnCharCode)
-{}
-
 KeyEvent KeyEvent::LogicalTextDirectionality (TextDirectionality eMode) const
 {
     KeyEvent aClone(*this);
--- source/gdi/pdfwriter_impl.hxx	2021-05-02 13:04:41.000000000 -0400
+++ source/gdi/pdfwriter_impl.hxx	2021-05-03 06:45:53.000000000 -0400
@@ -182,6 +182,14 @@
 
         BitmapID() : m_nSize( 0 ), m_nChecksum( 0 ), m_nMaskChecksum( 0 ) {}
 
+        BitmapID( const BitmapID& rID ) :
+                m_aPixelSize( rID.m_aPixelSize ),
+                m_nSize( rID.m_nSize ),
+                m_nChecksum( rID.m_nChecksum ),
+                m_nMaskChecksum( rID.m_nMaskChecksum )
+        {
+        }
+
         BitmapID& operator=( const BitmapID& rCopy )
         {
             m_aPixelSize    = rCopy.m_aPixelSize;
@@ -819,14 +827,14 @@
     sal_Int32                               m_nEncryptionBufferSize;
 
     /* check and reallocate the buffer for encryption */
-    bool checkEncryptionBufferSize( register sal_Int32 newSize );
+    bool checkEncryptionBufferSize( sal_Int32 newSize );
     /* this function implements part of the PDF spec algorithm 3.1 in encryption, the rest (the actual encryption) is in PDFWriterImpl::writeBuffer */
-    void checkAndEnableStreamEncryption( register sal_Int32 nObject );
+    void checkAndEnableStreamEncryption( sal_Int32 nObject );
 
     void disableStreamEncryption() { m_bEncryptThisStream = false; };
 
     /* */
-    void enableStringEncryption( register sal_Int32 nObject );
+    void enableStringEncryption( sal_Int32 nObject );
 
 // test if the encryption is active, if yes than encrypt the unicode string  and add to the OStringBuffer parameter
     void appendUnicodeTextStringEncrypt( const OUString& rInString, const sal_Int32 nInObjectNumber, OStringBuffer& rOutBuffer );
--- source/gdi/pdfwriter_impl2.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/gdi/pdfwriter_impl2.cxx	2021-05-02 13:00:54.000000000 -0400
@@ -1133,7 +1133,7 @@
     return pResult;
 }
 
-bool PDFWriterImpl::checkEncryptionBufferSize( register sal_Int32 newSize )
+bool PDFWriterImpl::checkEncryptionBufferSize( sal_Int32 newSize )
 {
     if( m_nEncryptionBufferSize < newSize )
     {
@@ -1148,7 +1148,7 @@
     return ( m_nEncryptionBufferSize != 0 );
 }
 
-void PDFWriterImpl::checkAndEnableStreamEncryption( register sal_Int32 nObject )
+void PDFWriterImpl::checkAndEnableStreamEncryption( sal_Int32 nObject )
 {
     if( m_aContext.Encryption.Encrypt() )
     {
@@ -1168,7 +1168,7 @@
     }
 }
 
-void PDFWriterImpl::enableStringEncryption( register sal_Int32 nObject )
+void PDFWriterImpl::enableStringEncryption( sal_Int32 nObject )
 {
     if( m_aContext.Encryption.Encrypt() )
     {
--- source/filter/wmf/winmtf.hxx	2015-11-25 07:55:38.000000000 -0500
+++ source/filter/wmf/winmtf.hxx	2021-05-11 08:01:01.000000000 -0400
@@ -382,15 +382,6 @@
             && aType == pStyle->aType;
     }
 
-    WinMtfFillStyle& operator=(const WinMtfFillStyle& rStyle)
-    {
-        aFillColor = rStyle.aFillColor;
-        bTransparent = rStyle.bTransparent;
-        aBmp = rStyle.aBmp;
-        aType = rStyle.aType;
-        return *this;
-    }
-
     WinMtfFillStyle& operator=(WinMtfFillStyle* pStyle)
     {
         aFillColor = pStyle->aFillColor;
--- source/control/combobox.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/control/combobox.cxx	2021-10-25 16:35:54.000000000 -0400
@@ -32,6 +32,8 @@
 #include <ilstbox.hxx>
 #include <controldata.hxx>
 
+using namespace ::boost::placeholders;
+
 static void lcl_GetSelectedEntries( ::std::set< sal_Int32 >& rSelectedPos, const OUString& rText, sal_Unicode cTokenSep, const ImplEntryList* pEntryList )
 {
     for (sal_Int32 n = comphelper::string::getTokenCount(rText, cTokenSep); n;)
--- source/control/lstbox.cxx	2015-11-25 07:55:38.000000000 -0500
+++ source/control/lstbox.cxx	2021-10-25 16:35:43.000000000 -0400
@@ -37,6 +37,8 @@
 
 #include <com/sun/star/datatransfer/dnd/XDropTarget.hpp>
 
+using namespace ::boost::placeholders;
+
 void ListBox::EnableQuickSelection( const bool& b )
 {
     mpImplLB->GetMainWindow().EnableQuickSelection(b);
--- osx/saldata.cxx	2015-11-25 07:55:38.000000000 -0500
+++ osx/saldata.cxx	2021-10-29 14:01:26.000000000 -0400
@@ -74,8 +74,10 @@
         pPool = reinterpret_cast<NSAutoreleasePool*>( osl_getThreadKeyData( s_aAutoReleaseKey ) );
         if( pPool )
         {
-            osl_setThreadKeyData( s_aAutoReleaseKey, NULL );
-            [pPool release];
+            // Avoid double release of pool by only releasing when setting the
+            // thread key data fails
+            if ( !osl_setThreadKeyData( s_aAutoReleaseKey, NULL ) )
+                [pPool release];
         }
 
         osl_destroyThreadKey( s_aAutoReleaseKey );
