--- qa/unit/WpftWriterFilterTest.cxx.bak	2015-11-25 05:55:38.000000000 -0700
+++ qa/unit/WpftWriterFilterTest.cxx	2018-10-17 19:45:45.000000000 -0600
@@ -34,7 +34,9 @@
     doTest("com.sun.star.comp.Writer.AbiWordImportFilter", "/writerperfect/qa/unit/data/writer/libabw/");
     doTest("org.libreoffice.comp.Writer.EBookImportFilter", "/writerperfect/qa/unit/data/writer/libe-book/");
     doTest("com.sun.star.comp.Writer.MSWorksImportFilter", "/writerperfect/qa/unit/data/writer/libwps/");
+#if defined USE_JAVA || !defined MACOSX
     doTest("com.sun.star.comp.Writer.MWAWImportFilter", "/writerperfect/qa/unit/data/writer/libmwaw/");
+#endif	// USE_JAVA || !MACOSX
     doTest("com.sun.star.comp.Writer.WordPerfectImportFilter", "/writerperfect/qa/unit/data/writer/libwpd/");
 }
 
