--- util/compact.xsl	2018-05-09 05:27:29.000000000 -0500
+++ util/compact.xsl	2018-12-30 17:39:43.000000000 -0600
@@ -29,23 +29,31 @@
             <xsl:apply-templates select="@*[normalize-space()]|node()|text()"/>
         </xsl:copy>
     </xsl:template>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="@localize"/>
+    -->
     <xsl:template match="@xml-lang"/>
     <xsl:template match="alt"/>
     <xsl:template match="bookmark_value"/>
     <xsl:template match="comment()"/>       <!-- Remove all XML comments -->
     <xsl:template match="comment"/>
     <xsl:template match="history"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="image/@id"/>
+    -->
     <xsl:template match="image/@width"/>
     <xsl:template match="image/@height"/>
     <xsl:template match="link/@name"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="paragraph/@id"/>
+    -->
     <xsl:template match="paragraph/@l10n"/>
     <xsl:template match="paragraph/@oldref"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="table/@id"/>
     <xsl:template match="title/@id"/>
     <xsl:template match="topic/@id"/>
+    -->
     <xsl:template match="topic/@indexer"/>
     <xsl:template match="topic/@status"/>
 
