--- util/compact.xsl	2022-10-10 06:15:14.000000000 -0400
+++ util/compact.xsl	2022-10-15 15:09:30.000000000 -0400
@@ -29,21 +29,27 @@
             <xsl:apply-templates select="@*[normalize-space()]|node()|text()"/>
         </xsl:copy>
     </xsl:template>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="@localize"/>
+    -->
     <xsl:template match="@xml-lang"/>
     <xsl:template match="alt"/>
     <xsl:template match="bookmark_value"/>
     <xsl:template match="comment()"/>       <!-- Remove all XML comments -->
     <xsl:template match="comment"/>
     <xsl:template match="history"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="image/@id"/>
+    -->
     <xsl:template match="image/@width"/>
     <xsl:template match="image/@height"/>
     <xsl:template match="link/@name"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="paragraph/@id"/>
     <xsl:template match="table/@id"/>
     <xsl:template match="title/@id"/>
     <xsl:template match="topic/@id"/>
+    -->
     <xsl:template match="topic/@indexer"/>
     <xsl:template match="topic/@status"/>
 
