--- util/compact.xsl	2015-11-25 06:55:38.000000000 -0600
+++ util/compact.xsl	2017-04-12 15:07:07.000000000 -0500
@@ -24,23 +33,31 @@
     <!-- To remove attributes or nodes, 
          simply write a matching template that doesn't do anything. 
          Therefore, it is removed -->
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="@localize"/>
+    -->
     <xsl:template match="@xml-lang"/>
     <xsl:template match="alt"/>
     <xsl:template match="bookmark_value"/>
     <xsl:template match="comment()"/>       <!-- Remove all XML comments -->
     <xsl:template match="comment"/>
     <xsl:template match="history"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="image/@id"/>
+    -->
     <xsl:template match="image/@width"/>
     <xsl:template match="image/@height"/>
     <xsl:template match="link/@name"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="paragraph/@id"/>
+    -->
     <xsl:template match="paragraph/@l10n"/>
     <xsl:template match="paragraph/@oldref"/>
+    <!-- Do not remove attributes used in main_transform.xsl
     <xsl:template match="table/@id"/>
     <xsl:template match="title/@id"/>
     <xsl:template match="topic/@id"/>
+    -->
     <xsl:template match="topic/@indexer"/>
     <xsl:template match="topic/@status"/>
 
